{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:47.732036Z",
     "start_time": "2025-03-26T19:58:47.729774Z"
    }
   },
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n",
    "from sklearn.utils import shuffle"
   ],
   "outputs": [],
   "execution_count": 197
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:47.885787Z",
     "start_time": "2025-03-26T19:58:47.870980Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df = pd.read_csv('data/toxins1.tsv', sep='\\t')\n",
    "df"
   ],
   "id": "8a325e392f034880",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                                       Protein families  \\\n",
       "0                               Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family, P-III s...   \n",
       "2                                        Insulin family   \n",
       "3                                        Insulin family   \n",
       "4     Arthropod phospholipase D family, Class II sub...   \n",
       "...                                                 ...   \n",
       "6423                                                NaN   \n",
       "6424                                                NaN   \n",
       "6425                                                NaN   \n",
       "6426                                                NaN   \n",
       "6427                                                NaN   \n",
       "\n",
       "                                               Sequence  Fragment  \n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment  \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN  \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN  \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN  \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment  \n",
       "...                                                 ...       ...  \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN  \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN  \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN  \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment  \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment  \n",
       "\n",
       "[6428 rows x 6 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family, P-III s...</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family, Class II sub...</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 6 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 198
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:48.024508Z",
     "start_time": "2025-03-26T19:58:48.019200Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df['Protein families'] = df['Protein families'].str.split(',').str[0]\n",
    "df['Protein families'] = df['Protein families'].str.split(';').str[0]"
   ],
   "id": "614d78cba278e28f",
   "outputs": [],
   "execution_count": 199
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:48.187973Z",
     "start_time": "2025-03-26T19:58:48.183559Z"
    }
   },
   "cell_type": "code",
   "source": [
    "filtered = df[df['Protein families'].str.contains('superfamily', na=False)]['Protein families'].unique()\n",
    "filtered"
   ],
   "id": "e6959ed706b8aaa2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Conotoxin A superfamily', 'Conotoxin O1 superfamily',\n",
       "       'Conotoxin M superfamily',\n",
       "       'Long (4 C-C) scorpion toxin superfamily',\n",
       "       'Short scorpion toxin superfamily', 'Conotoxin B superfamily',\n",
       "       'Formicidae venom precursor-01 superfamily',\n",
       "       'Conotoxin D superfamily', 'O2 superfamily',\n",
       "       'AB hydrolase superfamily', 'Conotoxin T superfamily',\n",
       "       'Non-disulfide-bridged peptide (NDBP) superfamily',\n",
       "       'Long (3 C-C) scorpion toxin superfamily',\n",
       "       'Conotoxin J superfamily', 'Conotoxin O2 superfamily',\n",
       "       'Conotoxin S superfamily', 'Conotoxin C superfamily',\n",
       "       'Conotoxin L superfamily', 'Conotoxin I2 superfamily',\n",
       "       'Conotoxin I1 superfamily', 'Conotoxin P superfamily',\n",
       "       'Conotoxin U superfamily', 'Conotoxin V superfamily',\n",
       "       'Conotoxin Y superfamily', 'Conotoxin K superfamily',\n",
       "       'Conotoxin I3 superfamily', 'Pg turripeptide superfamily',\n",
       "       'Conopeptide P-like superfamily', 'O1 superfamily',\n",
       "       'Conotoxin H superfamily', 'Conotoxin NSf-1 superfamily',\n",
       "       'Conotoxin G superfamily', 'Conotoxin R superfamily',\n",
       "       'Conotoxin Q superfamily', 'Conotoxin E superfamily',\n",
       "       'Conotoxin N superfamily', 'Conopeptide I2-like superfamily',\n",
       "       'E superfamily', 'F superfamily', 'Teretoxin C (TC) superfamily',\n",
       "       'Teretoxin A (TA) superfamily', 'Teretoxin M (TM) superfamily',\n",
       "       'Teretoxin H (TH) superfamily', 'Teretoxin N (TN) superfamily',\n",
       "       'Teretoxin J (TJ) superfamily', 'Conotoxin O3 superfamily',\n",
       "       'I1 superfamily'], dtype=object)"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 200
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:48.624442Z",
     "start_time": "2025-03-26T19:58:48.621236Z"
    }
   },
   "cell_type": "code",
   "source": "df[\"Protein families\"].value_counts()",
   "id": "abaec70fd5b54e72",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Snake three-finger toxin family            532\n",
       "Long (4 C-C) scorpion toxin superfamily    418\n",
       "Phospholipase A2 family                    414\n",
       "Neurotoxin 10 (Hwtx-1) family              289\n",
       "Venom metalloproteinase (M12B) family      281\n",
       "                                          ... \n",
       "Scoloptoxin family                           1\n",
       "Scolopendra neurotoxin 8 family              1\n",
       "Scolopendra toxin 6 family                   1\n",
       "Scolopendra toxin 7 family                   1\n",
       "SLPTX(15) family                             1\n",
       "Name: count, Length: 235, dtype: int64"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 201
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:48.816392Z",
     "start_time": "2025-03-26T19:58:48.811847Z"
    }
   },
   "cell_type": "code",
   "source": "df[df[\"Protein families\"].str.contains(\"phospholipase\", case=False, na=False)][\"Protein families\"].value_counts()",
   "id": "305c9c3ea1cf9f5b",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Phospholipase A2 family             414\n",
       "Arthropod phospholipase D family    217\n",
       "Phospholipase B-like family           2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 202
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:49.734214Z",
     "start_time": "2025-03-26T19:58:49.726723Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df['Protein families'] = df['Protein families'].replace('I1 superfamily', 'Conotoxin I1 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('O1 superfamily', 'Conotoxin O1 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('O2 superfamily', 'Conotoxin O2 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('E superfamily', 'Conotoxin E superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('F superfamily', 'Conotoxin F superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin M family', 'Conotoxin M superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin B2 family', 'Conotoxin B2 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin O1 family', 'Conotoxin O1 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin O2 family', 'Conotoxin O2 superfamily')"
   ],
   "id": "d97faf51efd2f8f6",
   "outputs": [],
   "execution_count": 203
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:50.186469Z",
     "start_time": "2025-03-26T19:58:50.161995Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mapping = {\n",
    "    r'Conotoxin.*': 'Conotoxin family',\n",
    "    r'Neurotoxin.*': 'Neurotoxin family',\n",
    "    r'Scoloptoxin.*|Scolopendra.*': 'Scoloptoxin family',\n",
    "    r'Caterpillar.*': 'Caterpillar family',\n",
    "    r'Teretoxin.*': 'Teretoxin family',\n",
    "    r'Limacoditoxin.*': 'Limacoditoxin family',\n",
    "    r'Scutigerotoxin.*': 'Scutigerotoxin family',\n",
    "    r'Cationic peptide.*': 'Cationic peptide family',\n",
    "    r'Formicidae venom.*': 'Formicidae venom family',\n",
    "    r'Bradykinin-potentiating peptide family|Natriuretic peptide family': 'Natriuretic, Bradykinin potentiating peptide family',\n",
    "    r'.*phospholipase.*|.*Phospholipase.*': 'Phospholipase family'\n",
    "}\n",
    "\n",
    "# Apply mapping\n",
    "for pattern, replacement in mapping.items():\n",
    "    df['Protein families'] = df['Protein families'].str.replace(pattern, replacement, regex=True)\n",
    "\n",
    "df['Protein families'].value_counts()"
   ],
   "id": "8f13fa5913864de4",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                           1131\n",
       "Neurotoxin family                          1015\n",
       "Phospholipase family                        633\n",
       "Snake three-finger toxin family             532\n",
       "Long (4 C-C) scorpion toxin superfamily     418\n",
       "                                           ... \n",
       "DNase II family                               1\n",
       "5'-nucleotidase family                        1\n",
       "Diuretic hormone class 2 family               1\n",
       "CART family                                   1\n",
       "SLPTX(15) family                              1\n",
       "Name: count, Length: 97, dtype: int64"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 204
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:50.953579Z",
     "start_time": "2025-03-26T19:58:50.951255Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# just some MSA shenanigans\n",
    "# filtered_df = df[df[\"Protein families\"].str.contains(\"Neurotoxin\", na=False, case=False)]\n",
    "#\n",
    "# with open(\"/Users/selin/Desktop/filtered_neurotoxins.fasta\", \"w\") as fasta_file:\n",
    "#     for _, row in filtered_df.iterrows():\n",
    "#         fasta_file.write(f\">{row['Entry']}\\t{row[\"Protein families\"]}\\n{row['Sequence']}\\n\")\n",
    "#\n",
    "#\n",
    "# filtered_df = df[df[\"Protein families\"].str.contains(\"Conotoxin\", na=False, case=False)]\n",
    "#\n",
    "# with open(\"/Users/selin/Desktop/filtered_conotoxins.fasta\", \"w\") as fasta_file:\n",
    "#     for _, row in filtered_df.iterrows():\n",
    "#         fasta_file.write(f\">{row['Entry']}\\t{row[\"Protein families\"]}\\n{row['Sequence']}\\n\")"
   ],
   "id": "c9b1bafb1a3a288b",
   "outputs": [],
   "execution_count": 205
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:58:51.610537Z",
     "start_time": "2025-03-26T19:58:51.607785Z"
    }
   },
   "cell_type": "code",
   "source": "df[\"Protein families\"]",
   "id": "7918b6efed4306c4",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0                            Conotoxin family\n",
       "1       Venom metalloproteinase (M12B) family\n",
       "2                              Insulin family\n",
       "3                              Insulin family\n",
       "4                        Phospholipase family\n",
       "                        ...                  \n",
       "6423                                      NaN\n",
       "6424                                      NaN\n",
       "6425                                      NaN\n",
       "6426                                      NaN\n",
       "6427                                      NaN\n",
       "Name: Protein families, Length: 6428, dtype: object"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 206
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T11:02:52.067525Z",
     "start_time": "2025-03-26T11:02:52.062348Z"
    }
   },
   "cell_type": "code",
   "source": [
    "scolop = df[df[\"Entry\"].isin([\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"])]\n",
    "scolop"
   ],
   "id": "de40e54c5b02400d",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Entry                                           Organism  \\\n",
       "2440  I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "5337  P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "5339  P0DPX7  Scolopendra morsitans (Tanzanian blue ringleg ...   \n",
       "5340  P0DPX8            Ethmostigmus rubripes (Giant centipede)   \n",
       "5341  P0DPX9            Ethmostigmus rubripes (Giant centipede)   \n",
       "5342  P0DPY0            Ethmostigmus rubripes (Giant centipede)   \n",
       "5343  P0DPY1  Cormocephalus westwoodi (Westwood's green cent...   \n",
       "\n",
       "                                          Protein names    Protein families  \\\n",
       "2440  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...  Scoloptoxin family   \n",
       "5337          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)  Scoloptoxin family   \n",
       "5339          U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)  Scoloptoxin family   \n",
       "5340          U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)  Scoloptoxin family   \n",
       "5341          U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)  Scoloptoxin family   \n",
       "5342          U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)  Scoloptoxin family   \n",
       "5343          U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)  Scoloptoxin family   \n",
       "\n",
       "                                               Sequence Fragment  \n",
       "2440  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...      NaN  \n",
       "5337  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN  \n",
       "5339  MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...      NaN  \n",
       "5340  MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...      NaN  \n",
       "5341  MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...      NaN  \n",
       "5342  MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...      NaN  \n",
       "5343  MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...      NaN  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2440</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>P0DPX7</td>\n",
       "      <td>Scolopendra morsitans (Tanzanian blue ringleg ...</td>\n",
       "      <td>U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5340</th>\n",
       "      <td>P0DPX8</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5341</th>\n",
       "      <td>P0DPX9</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5342</th>\n",
       "      <td>P0DPY0</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5343</th>\n",
       "      <td>P0DPY1</td>\n",
       "      <td>Cormocephalus westwoodi (Westwood's green cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 151
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:04.135785Z",
     "start_time": "2025-03-26T19:59:04.129216Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df = df.loc[df['Fragment'] != \"fragment\"]\n",
    "df = df.loc[df['Protein families'].notnull()]\n",
    "df"
   ],
   "id": "38bd4e2986b1ff91",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "6     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "6405      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "6406      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "6407      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "6408      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "6409      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "5     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "6     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "6405                                       Toxin ICK-13   \n",
       "6406                                        Toxin ICK-8   \n",
       "6407   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "6408                                       Contryphan-G   \n",
       "6409  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "1     Venom metalloproteinase (M12B) family   \n",
       "2                            Insulin family   \n",
       "3                            Insulin family   \n",
       "5                      Phospholipase family   \n",
       "6                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "6405                      Neurotoxin family   \n",
       "6406                      Neurotoxin family   \n",
       "6407                      Neurotoxin family   \n",
       "6408                       Conotoxin family   \n",
       "6409                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN  \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN  \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN  \n",
       "5     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN  \n",
       "6     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN  \n",
       "...                                                 ...      ...  \n",
       "6405  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN  \n",
       "6406  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN  \n",
       "6407  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN  \n",
       "6408  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN  \n",
       "6409  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN  \n",
       "\n",
       "[5039 rows x 6 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6405</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6406</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6407</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6408</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6409</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows Ã— 6 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 207
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:05.383873Z",
     "start_time": "2025-03-26T19:59:05.380528Z"
    }
   },
   "cell_type": "code",
   "source": "df[\"Protein families\"].value_counts()",
   "id": "db4c776d8ffb6929",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                           989\n",
       "Neurotoxin family                          959\n",
       "Snake three-finger toxin family            485\n",
       "Long (4 C-C) scorpion toxin superfamily    337\n",
       "Phospholipase family                       319\n",
       "                                          ... \n",
       "DNase II family                              1\n",
       "Type-B carboxylesterase/lipase family        1\n",
       "Diuretic hormone class 2 family              1\n",
       "CART family                                  1\n",
       "SLPTX(15) family                             1\n",
       "Name: count, Length: 95, dtype: int64"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 208
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T11:04:12.124819Z",
     "start_time": "2025-03-26T11:04:12.122739Z"
    }
   },
   "cell_type": "code",
   "source": "#df.to_csv('data/foldseek/toxins.csv', index=False)",
   "id": "78a8d03042788329",
   "outputs": [],
   "execution_count": 157
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### back to clustering and redundancy reduction",
   "id": "b4c1a4625d4b10da"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T09:56:44.902752Z",
     "start_time": "2025-03-23T09:56:44.841958Z"
    }
   },
   "cell_type": "code",
   "source": [
    "def write_fasta(df, filename):\n",
    "    \"\"\"Writes a DataFrame to a FASTA file.\"\"\"\n",
    "    with open(filename, \"w\") as f:\n",
    "        for _, row in df.iterrows():\n",
    "            f.write(f\">{row['Entry']}\\n{row['Sequence']}\\n\")\n",
    "\n",
    "write_fasta(df, \"data/toxins.fasta\")"
   ],
   "id": "50bc1a6313d9e4ac",
   "outputs": [],
   "execution_count": 38
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T09:56:47.016731Z",
     "start_time": "2025-03-23T09:56:44.969898Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins.fasta data/mmseqs_0.3/toxins_cluster data/mmseqs_0.3/tmp --min-seq-id 0.3",
   "id": "e5c8c71cdab57aaa",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins.fasta data/mmseqs_0.3/toxins_cluster data/mmseqs_0.3/tmp --min-seq-id 0.3 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.3\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins.fasta data/mmseqs_0.3/tmp/9711347851748926998/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[4950] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "Create directory data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp\r\n",
      "cluster data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.3 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 5.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 5039 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 5.04K 0s 13ms     \r\n",
      "Sort kmer 0h 0m 0s 1ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 21ms\r\n",
      "rescorediagonal data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 5.04K 0s 8ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms=================> ] 97.26% 4.90K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "clust data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_rescore1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 5.04K 0s 6ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 8428 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 5.04K 0s 6ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 5.04K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 15ms\r\n",
      "Total time: 0h 0m 0s 16ms\r\n",
      "\r\n",
      "Size of the sequence database: 5039\r\n",
      "Size of the alignment database: 5039\r\n",
      "Number of clusters: 1915\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 17ms\r\n",
      "createsubdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/order_redundancy data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/order_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_filter1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_filter2 --filter-file data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 1.91K 0s 5ms      .99% 1.11K eta 0s       \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rescorediagonal data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_filter2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.91K 0s 5ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 1ms==============>    ] 92.63% 1.77K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "align data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_rescore2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1915 type: Aminoacid\r\n",
      "Target database size: 1915 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.91K 0s 25ms     \r\n",
      "Time for merging to aln: 0h 0m 0s 1ms\r\n",
      "2401 alignments calculated\r\n",
      "2345 sequence pairs passed the thresholds (0.976676 of overall calculated)\r\n",
      "1.224543 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 37ms\r\n",
      "clust data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/aln data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.91K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 430 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.91K 0s 4ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.91K 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 9ms\r\n",
      "Total time: 0h 0m 0s 10ms\r\n",
      "\r\n",
      "Size of the sequence database: 1915\r\n",
      "Size of the alignment database: 1915\r\n",
      "Number of clusters: 1624\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "mergeclusters data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pre_clust data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 1.91K 0s 4ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 1.62K 0s 7ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 5.04K 0s 12ms     \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 15ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/linclust/15767041414715427829/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_redundancy data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1624 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1624 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.62K 0s 9ms      \r\n",
      "Index table: Masked residues: 2872\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.62K 0s 7ms      \r\n",
      "Index statistics\r\n",
      "Entries:          73814\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.001153\r\n",
      "Top 10 k-mers\r\n",
      "    TICDYT\t32\r\n",
      "    KCNKCY\t23\r\n",
      "    CYKWHR\t14\r\n",
      "    PYGSLE\t13\r\n",
      "    RGGPKP\t13\r\n",
      "    YSDRTT\t13\r\n",
      "    EDMYKV\t13\r\n",
      "    MRMTLL\t12\r\n",
      "    TMHLNL\t12\r\n",
      "    GLCCLP\t12\r\n",
      "Time for index table init: 0h 0m 0s 133ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1624\r\n",
      "Target db start 1 to 1624\r\n",
      "[=================================================================] 100.00% 1.62K 0s 11ms     \r\n",
      "[================================================================>] 99.94% 1.62K eta 0s       \r\n",
      "7.716461 k-mers per position\r\n",
      "85 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "5 sequences passed prefiltering per query sequence\r\n",
      "3 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 320ms\r\n",
      "align data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step0 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1624 type: Aminoacid\r\n",
      "Target database size: 1624 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.62K 0s 30ms     \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\r\n",
      "6942 alignments calculated\r\n",
      "6152 sequence pairs passed the thresholds (0.886200 of overall calculated)\r\n",
      "3.788177 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 36ms\r\n",
      "clust data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step0 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.62K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 298 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.62K 0s 4ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.62K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 10ms\r\n",
      "Total time: 0h 0m 0s 11ms\r\n",
      "\r\n",
      "Size of the sequence database: 1624\r\n",
      "Size of the alignment database: 1624\r\n",
      "Number of clusters: 945\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 12ms\r\n",
      "createsubdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step0 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 945 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 945 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 945 0s 5ms      \r\n",
      "Index table: Masked residues: 2231\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 945 0s 5ms      \r\n",
      "Index statistics\r\n",
      "Entries:          57417\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000897\r\n",
      "Top 10 k-mers\r\n",
      "    MKTVVA\t12\r\n",
      "    VALLCQ\t10\r\n",
      "    KLCLAV\t7\r\n",
      "    LDISGL\t6\r\n",
      "    YGCGNN\t6\r\n",
      "    LLLLKP\t6\r\n",
      "    FGKDGS\t6\r\n",
      "    MKLLVV\t6\r\n",
      "    LLLFLI\t4\r\n",
      "    TCLVLI\t4\r\n",
      "Time for index table init: 0h 0m 0s 111ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 945\r\n",
      "Target db start 1 to 945\r\n",
      "[=================================================================] 100.00% 945 0s 12ms     \r\n",
      "[==========================================================>      ] 90.47% 855 eta 0s       \r\n",
      "32.177335 k-mers per position\r\n",
      "69 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 301ms\r\n",
      "align data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 945 type: Aminoacid\r\n",
      "Target database size: 945 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 945 0s 6ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 1ms\r\n",
      "1424 alignments calculated\r\n",
      "1156 sequence pairs passed the thresholds (0.811798 of overall calculated)\r\n",
      "1.223280 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "clust data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 945 0s 2ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 55 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 945 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 945 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 7ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 945\r\n",
      "Size of the alignment database: 945\r\n",
      "Number of clusters: 843\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "createsubdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 843 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 843 type: Aminoacid\r\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 843 0s 5ms        ] 16.03% 136 eta 0s       \r\n",
      "Index table: Masked residues: 2180\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 843 0s 5ms      \r\n",
      "Index statistics\r\n",
      "Entries:          53722\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000839\r\n",
      "Top 10 k-mers\r\n",
      "    GCGKRI\t9\r\n",
      "    RISSGC\t6\r\n",
      "    KLRGSG\t6\r\n",
      "    LDISGL\t6\r\n",
      "    LTVIVL\t6\r\n",
      "    FGKDGS\t6\r\n",
      "    IGVGCN\t4\r\n",
      "    YGCGNN\t4\r\n",
      "    YCGSCP\t4\r\n",
      "    LVILFP\t4\r\n",
      "Time for index table init: 0h 0m 0s 108ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 118\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 843\r\n",
      "Target db start 1 to 843\r\n",
      "[=================================================================] 100.00% 843 0s 23ms     \r\n",
      "\r\n",
      "192.255094 k-mers per position\r\n",
      "100 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "3 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 308ms\r\n",
      "align data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 843 type: Aminoacid\r\n",
      "Target database size: 843 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 843 0s 7ms      ====================================>       ] 89.19% 752 eta 0s       \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "1640 alignments calculated\r\n",
      "1017 sequence pairs passed the thresholds (0.620122 of overall calculated)\r\n",
      "1.206406 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "clust data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step2 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 843 0s 2ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 42 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 843 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 843 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 7ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 843\r\n",
      "Size of the alignment database: 843\r\n",
      "Number of clusters: 752\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "mergeclusters data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_redundancy data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step0 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step1 data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 1.62K 0s 3ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 945 0s 6ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 843 0s 9ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 752 0s 12ms     \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 5.04K 0s 16ms     \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 19ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_tmp/938995366834650638/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu data/mmseqs_0.3/tmp/9711347851748926998/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2repseq data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu data/mmseqs_0.3/tmp/9711347851748926998/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 752 0s 3ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "result2flat data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_rep data/mmseqs_0.3/tmp/9711347851748926998/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createseqfiledb data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu data/mmseqs_0.3/tmp/9711347851748926998/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 752 0s 3ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "result2flat data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/input data/mmseqs_0.3/tmp/9711347851748926998/clu_seqs data/mmseqs_0.3/tmp/9711347851748926998/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.3/tmp/9711347851748926998/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 39
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:32.234787Z",
     "start_time": "2025-03-26T19:59:32.225473Z"
    }
   },
   "cell_type": "code",
   "source": [
    "rep_data = []\n",
    "with open(\"data/mmseqs_0.3/toxins_cluster_rep_seq.fasta\") as f:\n",
    "    for record in SeqIO.parse(f, \"fasta\"):\n",
    "        rep_data.append([record.id, str(record.seq)])\n",
    "\n",
    "toxins_rep = pd.DataFrame(rep_data, columns=[\"Entry\", \"Sequence\"])\n",
    "\n",
    "toxins_rep = toxins_rep.merge(df[[\"Entry\", \"Protein families\"]], on=\"Entry\", how=\"left\")\n",
    "toxins_rep"
   ],
   "id": "489dcf49987c9f15",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "          Entry                                           Sequence  \\\n",
       "0        Q2I2R5  MKLSVMFIVFLMLTMPMTCAGISRSATNGGEADVRAHDKAANLMAL...   \n",
       "1        Q3BER1  MIPVLLVTICLAVFPFQGSSIILESGNINDYEIVYPKKVAVLPTGA...   \n",
       "2        B1P1B4  MTRKILAVLLVFTLVACNNAEKYSETDVEDSPMIQERRCEPSGKPC...   \n",
       "3        B1P1H0  MYDEILSAFFEVNDELQSEARCGEKNDRCKTNQDCCSGFRCTKFRR...   \n",
       "4        B1P1H8  MIFLLPPVIFVMLLAESVLILGDSEDADLMEMVQMSRPFFNPIIPA...   \n",
       "..          ...                                                ...   \n",
       "747  A0A7S8RGC8  MYNMVSLFIVAVLLLTYANVEGSDVTGGFPVNSNNCIYPCYSTQDE...   \n",
       "748      A5X2W8  MKTLLLILGVVAFVYLEPGYTTNCFTCTTWTLSCREFEKCPPDKGT...   \n",
       "749      A6YR42  MKLTCVLVVLLLVLPFGDLITTSNTEDNKRGATPWQNSLKARGVCS...   \n",
       "750      A7X3W6  MGQFTVVSLGLLAMFLSLSGAKGDNCPASWISRNGVCNKLFPDRKT...   \n",
       "751      A9QLM3  MNKVCFVVVLVLFVALAAYVSPIEGVPTGGCPLSDSLCAKYCKSHK...   \n",
       "\n",
       "                            Protein families  \n",
       "0                           Conotoxin family  \n",
       "1                         Disintegrin family  \n",
       "2                          Neurotoxin family  \n",
       "3                          Neurotoxin family  \n",
       "4                          Neurotoxin family  \n",
       "..                                       ...  \n",
       "747  Long (4 C-C) scorpion toxin superfamily  \n",
       "748          Snake three-finger toxin family  \n",
       "749                         Conotoxin family  \n",
       "750                 True venom lectin family  \n",
       "751         Short scorpion toxin superfamily  \n",
       "\n",
       "[752 rows x 3 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Protein families</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q2I2R5</td>\n",
       "      <td>MKLSVMFIVFLMLTMPMTCAGISRSATNGGEADVRAHDKAANLMAL...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q3BER1</td>\n",
       "      <td>MIPVLLVTICLAVFPFQGSSIILESGNINDYEIVYPKKVAVLPTGA...</td>\n",
       "      <td>Disintegrin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B1P1B4</td>\n",
       "      <td>MTRKILAVLLVFTLVACNNAEKYSETDVEDSPMIQERRCEPSGKPC...</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B1P1H0</td>\n",
       "      <td>MYDEILSAFFEVNDELQSEARCGEKNDRCKTNQDCCSGFRCTKFRR...</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>B1P1H8</td>\n",
       "      <td>MIFLLPPVIFVMLLAESVLILGDSEDADLMEMVQMSRPFFNPIIPA...</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>747</th>\n",
       "      <td>A0A7S8RGC8</td>\n",
       "      <td>MYNMVSLFIVAVLLLTYANVEGSDVTGGFPVNSNNCIYPCYSTQDE...</td>\n",
       "      <td>Long (4 C-C) scorpion toxin superfamily</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>748</th>\n",
       "      <td>A5X2W8</td>\n",
       "      <td>MKTLLLILGVVAFVYLEPGYTTNCFTCTTWTLSCREFEKCPPDKGT...</td>\n",
       "      <td>Snake three-finger toxin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>A6YR42</td>\n",
       "      <td>MKLTCVLVVLLLVLPFGDLITTSNTEDNKRGATPWQNSLKARGVCS...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>A7X3W6</td>\n",
       "      <td>MGQFTVVSLGLLAMFLSLSGAKGDNCPASWISRNGVCNKLFPDRKT...</td>\n",
       "      <td>True venom lectin family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>751</th>\n",
       "      <td>A9QLM3</td>\n",
       "      <td>MNKVCFVVVLVLFVALAAYVSPIEGVPTGGCPLSDSLCAKYCKSHK...</td>\n",
       "      <td>Short scorpion toxin superfamily</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>752 rows Ã— 3 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 209
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:34.805143Z",
     "start_time": "2025-03-26T19:59:34.803425Z"
    }
   },
   "cell_type": "code",
   "source": "# toxins_rep[\"Protein families\"].value_counts()",
   "id": "56f9fa13754606f4",
   "outputs": [],
   "execution_count": 210
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:34.939162Z",
     "start_time": "2025-03-26T19:59:34.933465Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cluster_df = pd.read_csv(\"data/mmseqs_0.3/toxins_cluster_cluster.tsv\", sep='\\t', header=None, names=['Entry', 'cluster_member'])\n",
    "cluster_df"
   ],
   "id": "4c6b68af0330835b",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry cluster_member\n",
       "0     A0A0D4WV12     A0A0D4WV12\n",
       "1     A0A0D4WV12     A0A1L4BJ98\n",
       "2     A0A0D4WV12         E5D3Z8\n",
       "3     A0A0D4WV12         Q8I914\n",
       "4     A0A0D4WV12         Q1KY79\n",
       "...          ...            ...\n",
       "5034      Q9BP83         Q9BH84\n",
       "5035      Q9BP83         Q9XYZ0\n",
       "5036      Q9BP83         Q9BP99\n",
       "5037      Q9BP83         Q9XZL2\n",
       "5038      Q9BP83         Q9U654\n",
       "\n",
       "[5039 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>cluster_member</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>A0A0D4WV12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>E5D3Z8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Q8I914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Q1KY79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9BH84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9XYZ0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9BP99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9XZL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9U654</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows Ã— 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 211
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### noise",
   "id": "1a4a5d6acf552c25"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:37.561182Z",
     "start_time": "2025-03-26T19:59:37.558255Z"
    }
   },
   "cell_type": "code",
   "source": "print((cluster_df['Entry'].value_counts() != 1).sum())",
   "id": "d63aed4259f44e3f",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "403\n"
     ]
    }
   ],
   "execution_count": 212
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:38.620503Z",
     "start_time": "2025-03-26T19:59:38.612984Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cluster_df['Cluster'] = cluster_df['Entry'].rank(method='dense').astype(int) - 1\n",
    "cluster_df"
   ],
   "id": "617014e66017d591",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry cluster_member  Cluster\n",
       "0     A0A0D4WV12     A0A0D4WV12        8\n",
       "1     A0A0D4WV12     A0A1L4BJ98        8\n",
       "2     A0A0D4WV12         E5D3Z8        8\n",
       "3     A0A0D4WV12         Q8I914        8\n",
       "4     A0A0D4WV12         Q1KY79        8\n",
       "...          ...            ...      ...\n",
       "5034      Q9BP83         Q9BH84      731\n",
       "5035      Q9BP83         Q9XYZ0      731\n",
       "5036      Q9BP83         Q9BP99      731\n",
       "5037      Q9BP83         Q9XZL2      731\n",
       "5038      Q9BP83         Q9U654      731\n",
       "\n",
       "[5039 rows x 3 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>cluster_member</th>\n",
       "      <th>Cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>E5D3Z8</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Q8I914</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Q1KY79</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9BH84</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9XYZ0</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9BP99</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9XZL2</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>Q9BP83</td>\n",
       "      <td>Q9U654</td>\n",
       "      <td>731</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows Ã— 3 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 213
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:50.178780Z",
     "start_time": "2025-03-26T19:59:50.170426Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df = df.merge(cluster_df[['cluster_member', 'Cluster']], left_on='Entry', right_on='cluster_member', how='left').drop(columns='cluster_member')\n",
    "df"
   ],
   "id": "2d40abcc6e1a0868",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "1     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "2     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "5034      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5035      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5037      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5038      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "1     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "2     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "5034                                       Toxin ICK-13   \n",
       "5035                                        Toxin ICK-8   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "5037                                       Contryphan-G   \n",
       "5038  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "0     Venom metalloproteinase (M12B) family   \n",
       "1                            Insulin family   \n",
       "2                            Insulin family   \n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "5034                      Neurotoxin family   \n",
       "5035                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "5037                       Conotoxin family   \n",
       "5038                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \n",
       "0     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN       60  \n",
       "1     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN        7  \n",
       "2     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7  \n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8  \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12  \n",
       "...                                                 ...      ...      ...  \n",
       "5034  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN      748  \n",
       "5035  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN       19  \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581  \n",
       "5037  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN       24  \n",
       "5038  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN      751  \n",
       "\n",
       "[5039 rows x 7 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>751</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows Ã— 7 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 214
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:52.278332Z",
     "start_time": "2025-03-26T19:59:52.270542Z"
    }
   },
   "cell_type": "code",
   "source": [
    "rep_seq_ids = {record.id for record in SeqIO.parse(\"data/mmseqs_0.3/toxins_cluster_rep_seq.fasta\", \"fasta\")}\n",
    "df['Cluster_Rep'] = df['Entry'].isin(rep_seq_ids)\n",
    "df"
   ],
   "id": "d90235f0a5ac4ecd",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "1     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "2     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "5034      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5035      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5037      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5038      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "1     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "2     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "5034                                       Toxin ICK-13   \n",
       "5035                                        Toxin ICK-8   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "5037                                       Contryphan-G   \n",
       "5038  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "0     Venom metalloproteinase (M12B) family   \n",
       "1                            Insulin family   \n",
       "2                            Insulin family   \n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "5034                      Neurotoxin family   \n",
       "5035                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "5037                       Conotoxin family   \n",
       "5038                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN       60   \n",
       "1     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN        7   \n",
       "2     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7   \n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "...                                                 ...      ...      ...   \n",
       "5034  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN      748   \n",
       "5035  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN       19   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "5037  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN       24   \n",
       "5038  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN      751   \n",
       "\n",
       "      Cluster_Rep  \n",
       "0           False  \n",
       "1           False  \n",
       "2           False  \n",
       "3            True  \n",
       "4            True  \n",
       "...           ...  \n",
       "5034        False  \n",
       "5035        False  \n",
       "5036        False  \n",
       "5037        False  \n",
       "5038         True  \n",
       "\n",
       "[5039 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>751</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows Ã— 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 215
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Train-Val-Test sets with 70:15:15 split",
   "id": "4965221a60e0c49"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:55.409476Z",
     "start_time": "2025-03-26T19:59:55.343607Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Get unique clusters and shuffle them\n",
    "clusters = df['Cluster'].unique()\n",
    "np.random.seed(42)\n",
    "np.random.shuffle(clusters)\n",
    "\n",
    "# Calculate total number of entries\n",
    "total_entries = len(df)\n",
    "\n",
    "# Define target sizes\n",
    "train_target = 0.7 * total_entries\n",
    "val_target = 0.15 * total_entries\n",
    "test_target = 0.15 * total_entries\n",
    "\n",
    "# Initialize cluster lists\n",
    "train_clusters, val_clusters, test_clusters = [], [], []\n",
    "\n",
    "# Initialize counters\n",
    "train_count, val_count, test_count = 0, 0, 0\n",
    "\n",
    "# Assign clusters while considering their sizes\n",
    "for cluster in clusters:\n",
    "    cluster_size = len(df[df['Cluster'] == cluster])\n",
    "\n",
    "    if train_count + cluster_size <= train_target:\n",
    "        train_clusters.append(cluster)\n",
    "        train_count += cluster_size\n",
    "    elif val_count + cluster_size <= val_target:\n",
    "        val_clusters.append(cluster)\n",
    "        val_count += cluster_size\n",
    "    else:\n",
    "        test_clusters.append(cluster)\n",
    "        test_count += cluster_size\n",
    "\n",
    "# Filter original df by cluster IDs\n",
    "train_df = df[df['Cluster'].isin(train_clusters)]\n",
    "val_df = df[df['Cluster'].isin(val_clusters)]\n",
    "test_df = df[df['Cluster'].isin(test_clusters)]\n",
    "\n",
    "# Shuffle each dataset\n",
    "train_df = shuffle(train_df, random_state=42).reset_index(drop=True)\n",
    "val_df = shuffle(val_df, random_state=42).reset_index(drop=True)\n",
    "test_df = shuffle(test_df, random_state=42).reset_index(drop=True)\n",
    "\n",
    "# Print results\n",
    "print(f\"Total entries: {total_entries}\")\n",
    "print(f\"Train entries: {len(train_df)} ({len(train_df)/total_entries:.2%})\")\n",
    "print(f\"Validation entries: {len(val_df)} ({len(val_df)/total_entries:.2%})\")\n",
    "print(f\"Test entries: {len(test_df)} ({len(test_df)/total_entries:.2%})\")\n",
    "\n",
    "# Output dataframes\n",
    "train_df, val_df, test_df"
   ],
   "id": "f6c7a3b9113fbb92",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total entries: 5039\n",
      "Train entries: 3527 (69.99%)\n",
      "Validation entries: 755 (14.98%)\n",
      "Test entries: 757 (15.02%)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(           Entry                                           Organism  \\\n",
       " 0         Q4VM07  Macrovipera lebetinus (Levantine viper) (Viper...   \n",
       " 1         B6DD05  Lycosa singoriensis (Wolf spider) (Aranea sing...   \n",
       " 2         P0CI57      Lychas mucronatus (Chinese swimming scorpion)   \n",
       " 3         P10116  Laticauda colubrina (Yellow-lipped sea krait) ...   \n",
       " 4         Q1ELU5                       Lachesana tarabaevi (Spider)   \n",
       " ...          ...                                                ...   \n",
       " 3522  A0A2P1BRQ0          Scorpaena plumieri (Spotted scorpionfish)   \n",
       " 3523      C1IC52        Walterinnesia aegyptia (Desert black snake)   \n",
       " 3524      P82942    Naja kaouthia (Monocled cobra) (Naja siamensis)   \n",
       " 3525      W4VS70              Conus victoriae (Queen Victoria cone)   \n",
       " 3526      Q3YEE2                    Conus capitaneus (Captain cone)   \n",
       " \n",
       "                                           Protein names  \\\n",
       " 0     Zinc metalloproteinase-disintegrin-like VLAIP-...   \n",
       " 1     U10-lycotoxin-Ls1b (Toxin-like structure LSTX-J3)   \n",
       " 2                                 Neurotoxin LmNaTx11.6   \n",
       " 3     Basic phospholipase A2 2 (svPLA2) (EC 3.1.1.4)...   \n",
       " 4     M-zodatoxin-Lt4a (M-ZDTX-Lt4a) [Cleaved into: ...   \n",
       " ...                                                 ...   \n",
       " 3522               Cytolytic toxin-alpha (Sp-CTx-alpha)   \n",
       " 3523  Kunitz-type serine protease inhibitor spermati...   \n",
       " 3524  Hemorrhagic metalloproteinase-disintegrin-like...   \n",
       " 3525                           Conotoxin Vc22.1 (E_Vc1)   \n",
       " 3526                                   Conotoxin CaHr91   \n",
       " \n",
       "                              Protein families  \\\n",
       " 0       Venom metalloproteinase (M12B) family   \n",
       " 1                           Neurotoxin family   \n",
       " 2     Long (4 C-C) scorpion toxin superfamily   \n",
       " 3                        Phospholipase family   \n",
       " 4                     Cationic peptide family   \n",
       " ...                                       ...   \n",
       " 3522                    SNTX/VTX toxin family   \n",
       " 3523                 Venom Kunitz-type family   \n",
       " 3524    Venom metalloproteinase (M12B) family   \n",
       " 3525                         Conotoxin family   \n",
       " 3526                         Conotoxin family   \n",
       " \n",
       "                                                Sequence Fragment  Cluster  \\\n",
       " 0     MMQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKITALPKGA...      NaN       60   \n",
       " 1     MKLIIFTGLVLFAIVSLIEAEEESGRVCIPLNGECSKSPDKCCDNI...      NaN      108   \n",
       " 2     MKIVIIFFIAMMAVGVYSKDGYLVKKNGCAYDCTVLFGDSNCNGEC...      NaN      682   \n",
       " 3     NLIQFSELIQCANKGKRATYYYMDYGCYCGKGGSGTPVDDLDRCCK...      NaN      199   \n",
       " 4     MKFSIIALALAVAFVCVAESRSEEEGYDVSEEIQAEELEEAERGGI...      NaN      684   \n",
       " ...                                                 ...      ...      ...   \n",
       " 3522  MSSDIIMAGLGRPFTLGFLYDARREKLIPGFSLFGDETLQKYATST...      NaN       26   \n",
       " 3523  MSSGCLLLLLGLLTLWAELTPVSGRPRLCELPAESGLCNAYIPSFY...      NaN      163   \n",
       " 3524  TNTPEQDRYLQAEKYIEFYVIVDNRMYRYYNYDKPAIKIRVYEMIN...      NaN      609   \n",
       " 3525  MMTRVFLAMFFLLVLTKGWPRLYDGDCTRGPNMHITCFKDQSCGLI...      NaN       13   \n",
       " 3526  MKLTCALIITVLFLSITADDSRGKQGYRALKSIAGMLNSKTVRECR...      NaN      731   \n",
       " \n",
       "       Cluster_Rep  \n",
       " 0           False  \n",
       " 1           False  \n",
       " 2           False  \n",
       " 3           False  \n",
       " 4           False  \n",
       " ...           ...  \n",
       " 3522         True  \n",
       " 3523        False  \n",
       " 3524         True  \n",
       " 3525        False  \n",
       " 3526        False  \n",
       " \n",
       " [3527 rows x 8 columns],\n",
       "           Entry                                           Organism  \\\n",
       " 0        D2Y2I6  Cyriopagopus hainanus (Chinese bird spider) (H...   \n",
       " 1    A0A7S8RFI7             Tityus serrulatus (Brazilian scorpion)   \n",
       " 2        P0DM21                 Conus textile (Cloth-of-gold cone)   \n",
       " 3        Q7YT39  Grammostola rosea (Chilean rose tarantula) (Gr...   \n",
       " 4        Q8JI38  Laticauda semifasciata (Black-banded sea krait...   \n",
       " ..          ...                                                ...   \n",
       " 750      Q2PQJ3                 Bothrops jararacussu (Jararacussu)   \n",
       " 751      B1P1E6  Chilobrachys guangxiensis (Chinese earth tiger...   \n",
       " 752      D2Y1Z6  Cyriopagopus hainanus (Chinese bird spider) (H...   \n",
       " 753      P82424  Neoponera goeldii (Ponerine ant) (Pachycondyla...   \n",
       " 754      B0FXM3  Crotalus durissus terrificus (South American r...   \n",
       " \n",
       "                                          Protein names  \\\n",
       " 0    Mu-theraphotoxin-Hhn2i (Mu-TRTX-Hhn2i) (Hainan...   \n",
       " 1    Putative sodium channel toxin Ts37 (Tityustoxi...   \n",
       " 2                Alpha-conotoxin TxIA (Conotoxin tx1a)   \n",
       " 3    M-theraphotoxin-Gr1a (M-TRTX-Gr1a) (GsMTx-4) (...   \n",
       " 4         Cysteine-rich venom protein latisemin (CRVP)   \n",
       " ..                                                 ...   \n",
       " 750  Thrombin-like enzyme BjussuSP-1 (SVTLE BjussuS...   \n",
       " 751  U14-theraphotoxin-Cg1a 1 (U14-TRTX-Cg1a) (Jing...   \n",
       " 752  Mu-theraphotoxin-Hhn2j 4 (Mu-TRTX-Hhn2j) (Hain...   \n",
       " 753  U1-poneritoxin-Ng1b (U1-PONTX-Ng1b) (Poneratox...   \n",
       " 754  Thrombin-like enzyme gyroxin B1.7 (SVTLE gyrox...   \n",
       " \n",
       "                                      Protein families  \\\n",
       " 0                                   Neurotoxin family   \n",
       " 1             Long (4 C-C) scorpion toxin superfamily   \n",
       " 2                                    Conotoxin family   \n",
       " 3                                   Neurotoxin family   \n",
       " 4                                        CRISP family   \n",
       " ..                                                ...   \n",
       " 750                               Peptidase S1 family   \n",
       " 751                                 Neurotoxin family   \n",
       " 752                                 Neurotoxin family   \n",
       " 753  Non-disulfide-bridged peptide (NDBP) superfamily   \n",
       " 754                               Peptidase S1 family   \n",
       " \n",
       "                                               Sequence Fragment  Cluster  \\\n",
       " 0    MKASMFLALAGLVLLFVVGYASESEEKEFPRELLSKIFAVDDFKGE...      NaN      338   \n",
       " 1    MAGEWACLLVSLVLLWGAAGSRDGFLLDRNFCRIKCSFLGSNSMCA...      NaN       51   \n",
       " 2    MFTVFLLVVLATAVVSFTSDRASDDGKAAASDLITLTIKGCCSRPP...      NaN       31   \n",
       " 3    MKTSVVFVIAGLALLSVVCYASELKEQSSVNEVLSTIFHFEQPEER...      NaN      338   \n",
       " 4    MIAFIVLLSLAAVLQQSSGTVDFASESSNKRENQKEIVDKHNALRR...      NaN      725   \n",
       " ..                                                 ...      ...      ...   \n",
       " 750  VLGGDECDINEHPFLAFLYSHGYFCGLTLINQEWVVTAAHCDSTNF...      NaN      700   \n",
       " 751  MKTSVLLVILGIAAITVQCTASESVEQDSLRTFVDAVLGWNAEMAS...      NaN       64   \n",
       " 752  MKASMFLALAGLVLLFVVGYASESEEKEFPIELLSKIFAVDVFKGE...      NaN      338   \n",
       " 753                          WLGSALKIGAKLLPSVVGLFQKKKK      NaN      604   \n",
       " 754  MVLIRVLANLLILQLSYAQKSSELVIGGDECNINEHRLLAIVYTNS...      NaN      700   \n",
       " \n",
       "      Cluster_Rep  \n",
       " 0          False  \n",
       " 1           True  \n",
       " 2          False  \n",
       " 3          False  \n",
       " 4          False  \n",
       " ..           ...  \n",
       " 750        False  \n",
       " 751        False  \n",
       " 752        False  \n",
       " 753        False  \n",
       " 754        False  \n",
       " \n",
       " [755 rows x 8 columns],\n",
       "           Entry                                           Organism  \\\n",
       " 0        Q91136                          Naja atra (Chinese cobra)   \n",
       " 1    A0A0B5A7P2  Conus geographus (Geography cone) (Nubecula ge...   \n",
       " 2        P0DUS5            Doratifera vulnerans (Mottled cup moth)   \n",
       " 3        F1CJ67  Hottentotta judaicus (Black scorpion) (Buthotu...   \n",
       " 4    A0A2P1BST7       Micrurus mipartitus (Red-tailed coral snake)   \n",
       " ..          ...                                                ...   \n",
       " 752      P83312                  Parabuthus schlechteri (Scorpion)   \n",
       " 753  A0A0B5AC90  Conus tulipa (Fish-hunting cone snail) (Tulip ...   \n",
       " 754      P0DPU7  Scolopendra dehaani (Thai centipede) (Scolopen...   \n",
       " 755  A0A125S9E4                   Conus imperialis (Imperial cone)   \n",
       " 756  A0A0B5ABD9  Conus geographus (Geography cone) (Nubecula ge...   \n",
       " \n",
       "                                          Protein names  \\\n",
       " 0      Cytotoxin I-like T-15 (Cardiotoxin I-like T-15)   \n",
       " 1    Con-Ins G1c (Insulin 1c) [Cleaved into: Con-In...   \n",
       " 2    DELTA-limacoditoxin(2)-Dv11 (DELTA-LCTX(2)-Dv1...   \n",
       " 3                                 U10-hottentoxin-Hj2a   \n",
       " 4    Mipartoxin-1A (Mipartoxin-IA) (Three-finger to...   \n",
       " ..                                                 ...   \n",
       " 752  Parabutoporin (PP) (Non-disulfide-bridged pept...   \n",
       " 753  Con-Ins T1B (Con-Ins T2) (Insulin 2) [Cleaved ...   \n",
       " 754                                 Scoloptoxin SSD996   \n",
       " 755                Conotoxin Im23a (Conopeptide im010)   \n",
       " 756  Con-Ins G2 (Insulin 2) [Cleaved into: Con-Ins ...   \n",
       " \n",
       "                                      Protein families  \\\n",
       " 0                     Snake three-finger toxin family   \n",
       " 1                                      Insulin family   \n",
       " 2                                Limacoditoxin family   \n",
       " 3                    Short scorpion toxin superfamily   \n",
       " 4                     Snake three-finger toxin family   \n",
       " ..                                                ...   \n",
       " 752  Non-disulfide-bridged peptide (NDBP) superfamily   \n",
       " 753                                    Insulin family   \n",
       " 754                                Scoloptoxin family   \n",
       " 755                                  Conotoxin family   \n",
       " 756                                    Insulin family   \n",
       " \n",
       "                                               Sequence Fragment  Cluster  \\\n",
       " 0    MKTLLLTLVVVTIVCLDLGYTLKCNKLIPIASKTCTAGKNLCYKMF...      NaN      179   \n",
       " 1    MTTSFYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7   \n",
       " 2    MKFAKTFLLLFVVLLLLSIVMAEPKRGFGKLLRKVFKVGRRVAGSA...      NaN      477   \n",
       " 3    MQKLLIILILFCILKFNVDVEGRTAFPCNQSKCQERCKKEIKKGKC...      NaN      178   \n",
       " 4    MKTLLLTLVVVTIVCLDLGNSLKCYVSREGETQTCPEGEKLCEKYA...      NaN      179   \n",
       " ..                                                 ...      ...      ...   \n",
       " 752      FKLGSFLKKAWKSKLAKKLRAKGKEMLKDYAKGLLEGGSEEVPGQ      NaN      618   \n",
       " 753  MTTSFYFLLMALGLLLYVCQSSFGNQHTRNSDTPKYRCGSDIPNSY...      NaN        7   \n",
       " 754  MKVFIAVLSIAIVLIIIVSIQETTANTGEAETTFAVPNKKFTREDC...      NaN      381   \n",
       " 755  MIMRMTLTLFVLVVMTAASASGDALTEAKRIPYCGQTGAECYSWCI...      NaN      147   \n",
       " 756  MTTSSYFLLVALGLLLYVRQSFSTHEHTCQLDDPAHPQGKCGSDLV...      NaN        7   \n",
       " \n",
       "      Cluster_Rep  \n",
       " 0          False  \n",
       " 1          False  \n",
       " 2           True  \n",
       " 3          False  \n",
       " 4          False  \n",
       " ..           ...  \n",
       " 752         True  \n",
       " 753        False  \n",
       " 754         True  \n",
       " 755        False  \n",
       " 756        False  \n",
       " \n",
       " [757 rows x 8 columns])"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 216
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:55.706942Z",
     "start_time": "2025-03-26T19:59:55.651120Z"
    }
   },
   "cell_type": "code",
   "source": [
    "write_fasta(train_df, \"data/toxins_train.fasta\")\n",
    "write_fasta(val_df, \"data/toxins_val.fasta\")\n",
    "write_fasta(test_df, \"data/toxins_test.fasta\")"
   ],
   "id": "7ce4d9c75bcc450d",
   "outputs": [],
   "execution_count": 217
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:02.500269Z",
     "start_time": "2025-03-26T20:00:00.300913Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins_train.fasta data/mmseqs_0.5/train_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "85b55d3f72a81043",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_train.fasta data/mmseqs_0.5/train_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_train.fasta data/mmseqs_0.5/tmp/6398341368778221515/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[3435] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 3527 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 3.53K 0s 12ms     \r\n",
      "Sort kmer 0h 0m 0s 1ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 21ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 3.53K 0s 7ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms================>  ] 96.14% 3.39K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 13ms\r\n",
      "clust data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_rescore1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 3.53K 0s 5ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 5973 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 3.53K 0s 5ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 3.53K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 12ms\r\n",
      "Total time: 0h 0m 0s 13ms\r\n",
      "\r\n",
      "Size of the sequence database: 3527\r\n",
      "Size of the alignment database: 3527\r\n",
      "Number of clusters: 1364\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 15ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/order_redundancy data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/order_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_filter1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_filter2 --filter-file data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 1.36K 0s 4ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_filter2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.36K 0s 4ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms================>  ] 96.70% 1.32K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "align data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_rescore2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1364 type: Aminoacid\r\n",
      "Target database size: 1364 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.36K 0s 14ms     \r\n",
      "Time for merging to aln: 0h 0m 0s 1ms\r\n",
      "1687 alignments calculated\r\n",
      "1590 sequence pairs passed the thresholds (0.942501 of overall calculated)\r\n",
      "1.165689 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 25ms\r\n",
      "clust data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/aln data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.36K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 226 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.36K 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.36K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 9ms\r\n",
      "Total time: 0h 0m 0s 10ms\r\n",
      "\r\n",
      "Size of the sequence database: 1364\r\n",
      "Size of the alignment database: 1364\r\n",
      "Number of clusters: 1195\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pre_clust data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 1.36K 0s 3ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 1.20K 0s 6ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 3.53K 0s 11ms     \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 13ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/linclust/17522866819169927381/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_redundancy data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1195 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1195 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.20K 0s 7ms      \r\n",
      "Index table: Masked residues: 2723\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.20K 0s 7ms                                    ] 3.43% 42 eta 0s       \r\n",
      "Index statistics\r\n",
      "Entries:          58234\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000910\r\n",
      "Top 10 k-mers\r\n",
      "    KCNKCY\t22\r\n",
      "    PYGSLE\t14\r\n",
      "    ICAFQG\t14\r\n",
      "    EDMYKV\t14\r\n",
      "    ISCGGH\t13\r\n",
      "    KVGPLH\t13\r\n",
      "    EDCYRI\t13\r\n",
      "    CVIAFL\t13\r\n",
      "    YEKNPV\t13\r\n",
      "    SEHSGR\t12\r\n",
      "Time for index table init: 0h 0m 0s 125ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1195\r\n",
      "Target db start 1 to 1195\r\n",
      "[=================================================================] 100.00% 1.20K 0s 11ms     \r\n",
      "[================================================================>] 99.33% 1.19K eta 0s       \r\n",
      "8.160284 k-mers per position\r\n",
      "94 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "5 sequences passed prefiltering per query sequence\r\n",
      "3 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 312ms\r\n",
      "align data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step0 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1195 type: Aminoacid\r\n",
      "Target database size: 1195 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.20K 0s 33ms     \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\r\n",
      "5208 alignments calculated\r\n",
      "3359 sequence pairs passed the thresholds (0.644969 of overall calculated)\r\n",
      "2.810879 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 38ms\r\n",
      "clust data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step0 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.20K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 42 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.20K 0s 4ms      ====================>                                        ] 36.93% 442 eta 0s       \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.20K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 10ms\r\n",
      "Total time: 0h 0m 0s 10ms\r\n",
      "\r\n",
      "Size of the sequence database: 1195\r\n",
      "Size of the alignment database: 1195\r\n",
      "Number of clusters: 786\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 12ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step0 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 786 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 786 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 786 0s 4ms      \r\n",
      "Index table: Masked residues: 2304\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 786 0s 5ms      ==============>                        ] 62.17% 489 eta 0s       \r\n",
      "Index statistics\r\n",
      "Entries:          44613\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000697\r\n",
      "Top 10 k-mers\r\n",
      "    VALLCQ\t13\r\n",
      "    KCNKCY\t9\r\n",
      "    GCGKRI\t6\r\n",
      "    YGCGNN\t6\r\n",
      "    FGKDGS\t5\r\n",
      "    CFLLIG\t4\r\n",
      "    MAEGLG\t4\r\n",
      "    KLRGSG\t4\r\n",
      "    EFYGGN\t4\r\n",
      "    HGCNCR\t4\r\n",
      "Time for index table init: 0h 0m 0s 109ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 786\r\n",
      "Target db start 1 to 786\r\n",
      "[=================================================================] 100.00% 786 0s 8ms      \r\n",
      "[============================================================>    ] 92.48% 727 eta 0s       \r\n",
      "13.088940 k-mers per position\r\n",
      "61 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 292ms\r\n",
      "align data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 786 type: Aminoacid\r\n",
      "Target database size: 786 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 786 0s 8ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "1276 alignments calculated\r\n",
      "811 sequence pairs passed the thresholds (0.635580 of overall calculated)\r\n",
      "1.031807 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 12ms\r\n",
      "clust data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 786 0s 2ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 3 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 786 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 786 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 7ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 786\r\n",
      "Size of the alignment database: 786\r\n",
      "Number of clusters: 772\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 772 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 772 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 772 0s 3ms      \r\n",
      "Index table: Masked residues: 2291\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 772 0s 5ms      \r\n",
      "Index statistics\r\n",
      "Entries:          53341\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000833\r\n",
      "Top 10 k-mers\r\n",
      "    MKTVVA\t14\r\n",
      "    VALLCQ\t12\r\n",
      "    KLCLAV\t10\r\n",
      "    KCNKCY\t9\r\n",
      "    FGKDGS\t5\r\n",
      "    VLLAFT\t5\r\n",
      "    MELILV\t5\r\n",
      "    KLRGSG\t4\r\n",
      "    VSGLFL\t4\r\n",
      "    LDISGL\t4\r\n",
      "Time for index table init: 0h 0m 0s 109ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 772\r\n",
      "Target db start 1 to 772\r\n",
      "[=================================================================] 100.00% 772 0s 9ms      \r\n",
      "[============================================================>    ] 93.77% 724 eta 0s       \r\n",
      "34.566337 k-mers per position\r\n",
      "79 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 304ms\r\n",
      "align data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 772 type: Aminoacid\r\n",
      "Target database size: 772 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 772 0s 6ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "1486 alignments calculated\r\n",
      "796 sequence pairs passed the thresholds (0.535666 of overall calculated)\r\n",
      "1.031088 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "clust data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step2 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 772 0s 2ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 6 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 772 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 772 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 7ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 772\r\n",
      "Size of the alignment database: 772\r\n",
      "Number of clusters: 757\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_redundancy data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step0 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step1 data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 1.20K 0s 3ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 786 0s 6ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 772 0s 9ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 757 0s 12ms     \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 3.53K 0s 16ms     \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 19ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_tmp/349779068787583308/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu data/mmseqs_0.5/tmp/6398341368778221515/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu data/mmseqs_0.5/tmp/6398341368778221515/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 757 0s 2ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_rep data/mmseqs_0.5/tmp/6398341368778221515/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu data/mmseqs_0.5/tmp/6398341368778221515/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 757 0s 3ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/input data/mmseqs_0.5/tmp/6398341368778221515/clu_seqs data/mmseqs_0.5/tmp/6398341368778221515/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/6398341368778221515/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 218
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:04.332771Z",
     "start_time": "2025-03-26T20:00:02.505296Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins_val.fasta data/mmseqs_0.5/val_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "4e88e675eaece40c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_val.fasta data/mmseqs_0.5/val_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_val.fasta data/mmseqs_0.5/tmp/8382699393235466370/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[708] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 755 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 755 0s 6ms      \r\n",
      "Sort kmer 0h 0m 0s 0ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 13ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 755 0s 3ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms===============>   ] 95.36% 720 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 755 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1425 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 755 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 755 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 9ms\r\n",
      "\r\n",
      "Size of the sequence database: 755\r\n",
      "Size of the alignment database: 755\r\n",
      "Number of clusters: 266\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/order_redundancy data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/order_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter2 --filter-file data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 266 0s 2ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 266 0s 2ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms==================>] 99.62% 265 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 266 type: Aminoacid\r\n",
      "Target database size: 266 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 266 0s 6ms      \r\n",
      "Time for merging to aln: 0h 0m 0s 0ms\r\n",
      "343 alignments calculated\r\n",
      "337 sequence pairs passed the thresholds (0.982507 of overall calculated)\r\n",
      "1.266917 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/aln data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 266 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 71 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 266 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 266 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 6ms\r\n",
      "\r\n",
      "Size of the sequence database: 266\r\n",
      "Size of the alignment database: 266\r\n",
      "Number of clusters: 220\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pre_clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 266 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 220 0s 2ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 755 0s 6ms      \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 220 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 220 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Index table: Masked residues: 271\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          8662\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000135\r\n",
      "Top 10 k-mers\r\n",
      "    CSPCNN\t7\r\n",
      "    APSPPP\t6\r\n",
      "    GQQIWG\t5\r\n",
      "    GMMFFL\t5\r\n",
      "    GDGPCN\t5\r\n",
      "    SGPIGQ\t5\r\n",
      "    IGVGCN\t4\r\n",
      "    DWQIGN\t4\r\n",
      "    HIPSSN\t4\r\n",
      "    CRMWIT\t4\r\n",
      "Time for index table init: 0h 0m 0s 109ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 220\r\n",
      "Target db start 1 to 220\r\n",
      "[=================================================================] 100.00% 220 0s 3ms      \r\n",
      "[==============================================================>  ] 95.43% 210 eta 0s       \r\n",
      "6.218910 k-mers per position\r\n",
      "47 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 292ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 220 type: Aminoacid\r\n",
      "Target database size: 220 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 220 0s 3ms      \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\r\n",
      "459 alignments calculated\r\n",
      "379 sequence pairs passed the thresholds (0.825708 of overall calculated)\r\n",
      "1.722727 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 220 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 4ms\r\n",
      "\r\n",
      "Size of the sequence database: 220\r\n",
      "Size of the alignment database: 220\r\n",
      "Number of clusters: 171\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 171 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 171 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Index table: Masked residues: 259\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          8201\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000128\r\n",
      "Top 10 k-mers\r\n",
      "    GCGKRI\t3\r\n",
      "    TLVTCL\t3\r\n",
      "    MRMTLL\t3\r\n",
      "    IGVGCN\t3\r\n",
      "    TTPACP\t3\r\n",
      "    SCPNCC\t2\r\n",
      "    GQVPCC\t2\r\n",
      "    GQCDDC\t2\r\n",
      "    AVNHDC\t2\r\n",
      "    GFRPFC\t2\r\n",
      "Time for index table init: 0h 0m 0s 101ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 171\r\n",
      "Target db start 1 to 171\r\n",
      "[=================================================================] 100.00% 171 0s 5ms      \r\n",
      "\r\n",
      "11.075385 k-mers per position\r\n",
      "44 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 280ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 171 type: Aminoacid\r\n",
      "Target database size: 171 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 171 0s 2ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "211 alignments calculated\r\n",
      "181 sequence pairs passed the thresholds (0.857820 of overall calculated)\r\n",
      "1.058480 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 0 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 171 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 4ms\r\n",
      "\r\n",
      "Size of the sequence database: 171\r\n",
      "Size of the alignment database: 171\r\n",
      "Number of clusters: 166\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 166 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 166 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 166 0s 1ms      \r\n",
      "Index table: Masked residues: 259\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 166 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          9822\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000153\r\n",
      "Top 10 k-mers\r\n",
      "    LALLDG\t3\r\n",
      "    VGSLGG\t3\r\n",
      "    GCGKRI\t3\r\n",
      "    ALPAGK\t3\r\n",
      "    TLVTCL\t3\r\n",
      "    MRMTLL\t3\r\n",
      "    MFVLVL\t3\r\n",
      "    IGVGCN\t3\r\n",
      "    LLLLKP\t3\r\n",
      "    VLTVFT\t3\r\n",
      "Time for index table init: 0h 0m 0s 112ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 166\r\n",
      "Target db start 1 to 166\r\n",
      "[=================================================================] 100.00% 166 0s 4ms      \r\n",
      "[================================================================>] 99.39% 165 eta 0s       \r\n",
      "29.999293 k-mers per position\r\n",
      "56 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 290ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 166 type: Aminoacid\r\n",
      "Target database size: 166 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 166 0s 2ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "220 alignments calculated\r\n",
      "169 sequence pairs passed the thresholds (0.768182 of overall calculated)\r\n",
      "1.018072 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 166 0s 0ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 166 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 166 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 4ms\r\n",
      "\r\n",
      "Size of the sequence database: 166\r\n",
      "Size of the alignment database: 166\r\n",
      "Number of clusters: 164\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 171 0s 2ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 166 0s 3ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 164 0s 4ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 755 0s 8ms      \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 164 0s 1ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_rep data/mmseqs_0.5/tmp/8382699393235466370/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 164 0s 1ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_seqs data/mmseqs_0.5/tmp/8382699393235466370/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 219
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.118386Z",
     "start_time": "2025-03-26T20:00:04.355232Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins_test.fasta data/mmseqs_0.5/test_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "37b6452492119dab",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_test.fasta data/mmseqs_0.5/test_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_test.fasta data/mmseqs_0.5/tmp/4163972790619549465/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[708] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 757 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 757 0s 5ms      \r\n",
      "Sort kmer 0h 0m 0s 0ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 757 0s 2ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms==================>] 99.07% 750 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 757 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1197 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 757 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 757 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 757\r\n",
      "Size of the alignment database: 757\r\n",
      "Number of clusters: 275\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/order_redundancy data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/order_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter2 --filter-file data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 275 0s 2ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 275 0s 2ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms================>  ] 96.72% 266 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 275 type: Aminoacid\r\n",
      "Target database size: 275 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 275 0s 2ms      \r\n",
      "Time for merging to aln: 0h 0m 0s 0ms\r\n",
      "321 alignments calculated\r\n",
      "303 sequence pairs passed the thresholds (0.943925 of overall calculated)\r\n",
      "1.101818 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/aln data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 28 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 275\r\n",
      "Size of the alignment database: 275\r\n",
      "Number of clusters: 251\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pre_clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 251 0s 3ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 757 0s 6ms      \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 251 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 251 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Index table: Masked residues: 80\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          9817\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000153\r\n",
      "Top 10 k-mers\r\n",
      "    TICDYT\t32\r\n",
      "    CGPVGI\t13\r\n",
      "    CYKWHR\t13\r\n",
      "    RGGPKP\t12\r\n",
      "    LDGTCY\t12\r\n",
      "    CPGNFK\t9\r\n",
      "    KRCDPK\t9\r\n",
      "    VKGICP\t9\r\n",
      "    CSDCFP\t9\r\n",
      "    VCDGLK\t7\r\n",
      "Time for index table init: 0h 0m 0s 99ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 251\r\n",
      "Target db start 1 to 251\r\n",
      "[=================================================================] 100.00% 251 0s 5ms      \r\n",
      "[===========================================================>     ] 92.00% 231 eta 0s       \r\n",
      "6.921199 k-mers per position\r\n",
      "82 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "8 sequences passed prefiltering per query sequence\r\n",
      "4 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 276ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 251 type: Aminoacid\r\n",
      "Target database size: 251 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 251 0s 3ms      \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\r\n",
      "1675 alignments calculated\r\n",
      "1034 sequence pairs passed the thresholds (0.617313 of overall calculated)\r\n",
      "4.119522 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 49 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 251 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 4ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 251\r\n",
      "Size of the alignment database: 251\r\n",
      "Number of clusters: 132\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 132 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 132 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 132 0s 1ms      \r\n",
      "Index table: Masked residues: 59\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "Index statistics\r\n",
      "Entries:          5718\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000089\r\n",
      "Top 10 k-mers\r\n",
      "    TLVTCL\t9\r\n",
      "    TICDYT\t7\r\n",
      "    LNVLAC\t2\r\n",
      "    WCKECC\t2\r\n",
      "    SSLKCC\t2\r\n",
      "    VKGNCC\t2\r\n",
      "    KCEIGC\t2\r\n",
      "    GTIRGC\t2\r\n",
      "    CLLYIC\t2\r\n",
      "    TECCKC\t2\r\n",
      "Time for index table init: 0h 0m 0s 106ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 132\r\n",
      "Target db start 1 to 132\r\n",
      "[=================================================================] 100.00% 132 0s 3ms      \r\n",
      "[=============================================================>   ] 94.66% 125 eta 0s       \r\n",
      "13.313164 k-mers per position\r\n",
      "43 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 283ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 132 type: Aminoacid\r\n",
      "Target database size: 132 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 132 0s 1ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 1ms\r\n",
      "235 alignments calculated\r\n",
      "142 sequence pairs passed the thresholds (0.604255 of overall calculated)\r\n",
      "1.075758 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 0 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 3ms\r\n",
      "\r\n",
      "Size of the sequence database: 132\r\n",
      "Size of the alignment database: 132\r\n",
      "Number of clusters: 127\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 127 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 127 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 127 0s 1ms      \r\n",
      "Index table: Masked residues: 59\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "Index statistics\r\n",
      "Entries:          6491\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000101\r\n",
      "Top 10 k-mers\r\n",
      "    TLVTCL\t6\r\n",
      "    MKLLVV\t6\r\n",
      "    CSDCFP\t3\r\n",
      "    HGSGYA\t2\r\n",
      "    LNVLAC\t2\r\n",
      "    WCKECC\t2\r\n",
      "    KCEIGC\t2\r\n",
      "    TECCKC\t2\r\n",
      "    KTPGLC\t2\r\n",
      "    GKFPLC\t2\r\n",
      "Time for index table init: 0h 0m 0s 108ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 127\r\n",
      "Target db start 1 to 127\r\n",
      "[=================================================================] 100.00% 127 0s 4ms      \r\n",
      "[=============================================================>   ] 94.44% 120 eta 0s       \r\n",
      "35.515875 k-mers per position\r\n",
      "51 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 292ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 127 type: Aminoacid\r\n",
      "Target database size: 127 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 127 0s 1ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "228 alignments calculated\r\n",
      "132 sequence pairs passed the thresholds (0.578947 of overall calculated)\r\n",
      "1.039370 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 2ms\r\n",
      "Total time: 0h 0m 0s 3ms\r\n",
      "\r\n",
      "Size of the sequence database: 127\r\n",
      "Size of the alignment database: 127\r\n",
      "Number of clusters: 124\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 132 0s 2ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 127 0s 3ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 124 0s 4ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 757 0s 7ms      \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 124 0s 0ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_rep data/mmseqs_0.5/tmp/4163972790619549465/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 124 0s 0ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_seqs data/mmseqs_0.5/tmp/4163972790619549465/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 220
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.164125Z",
     "start_time": "2025-03-26T20:00:06.156660Z"
    }
   },
   "cell_type": "code",
   "source": [
    "rep_seq_files = {\n",
    "    \"Train_Cluster_Rep\": \"data/mmseqs_0.5/train_cluster_rep_seq.fasta\",\n",
    "    \"Val_Cluster_Rep\": \"data/mmseqs_0.5/val_cluster_rep_seq.fasta\",\n",
    "    \"Test_Cluster_Rep\": \"data/mmseqs_0.5/test_cluster_rep_seq.fasta\",\n",
    "}\n",
    "\n",
    "for col, file in rep_seq_files.items():\n",
    "    rep_seq_ids = {record.id for record in SeqIO.parse(file, \"fasta\")}\n",
    "    df[col] = df['Entry'].isin(rep_seq_ids)\n",
    "\n",
    "df"
   ],
   "id": "3b891d1bb61c3473",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "1     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "2     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "5034      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5035      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5037      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5038      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "1     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "2     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "5034                                       Toxin ICK-13   \n",
       "5035                                        Toxin ICK-8   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "5037                                       Contryphan-G   \n",
       "5038  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "0     Venom metalloproteinase (M12B) family   \n",
       "1                            Insulin family   \n",
       "2                            Insulin family   \n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "5034                      Neurotoxin family   \n",
       "5035                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "5037                       Conotoxin family   \n",
       "5038                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN       60   \n",
       "1     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN        7   \n",
       "2     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7   \n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "...                                                 ...      ...      ...   \n",
       "5034  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN      748   \n",
       "5035  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN       19   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "5037  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN       24   \n",
       "5038  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN      751   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "0           False              False            False             False  \n",
       "1           False              False            False             False  \n",
       "2           False              False            False             False  \n",
       "3            True               True            False             False  \n",
       "4            True               True            False             False  \n",
       "...           ...                ...              ...               ...  \n",
       "5034        False              False            False             False  \n",
       "5035        False              False            False             False  \n",
       "5036        False               True            False             False  \n",
       "5037        False              False            False             False  \n",
       "5038         True              False            False             False  \n",
       "\n",
       "[5039 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>751</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 221
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.183955Z",
     "start_time": "2025-03-26T20:00:06.181439Z"
    }
   },
   "cell_type": "code",
   "source": [
    "all_50_pct_seq = (\n",
    "    df[\"Train_Cluster_Rep\"].sum().item() +\n",
    "    df[\"Val_Cluster_Rep\"].sum().item() +\n",
    "    df[\"Test_Cluster_Rep\"].sum().item()\n",
    ")\n",
    "print(\"All representative sequences after 50% redundancy reduction:\", all_50_pct_seq)\n",
    "print(f\"Train entries: {df[\"Train_Cluster_Rep\"].sum().item()} ({df[\"Train_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")\n",
    "\n",
    "print(f\"Validation entries: {df[\"Val_Cluster_Rep\"].sum().item()} ({df[\"Val_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")\n",
    "print(f\"Test entries: {df[\"Test_Cluster_Rep\"].sum().item()} ({df[\"Test_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")"
   ],
   "id": "5c3f48f7dcfebde8",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All representative sequences after 50% redundancy reduction: 1045\n",
      "Train entries: 757 (72.44%)\n",
      "Validation entries: 164 (15.69%)\n",
      "Test entries: 124 (11.87%)\n"
     ]
    }
   ],
   "execution_count": 222
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.213984Z",
     "start_time": "2025-03-26T20:00:06.208695Z"
    }
   },
   "cell_type": "code",
   "source": [
    "train_val_test_data = df[df[[\"Train_Cluster_Rep\", \"Val_Cluster_Rep\", \"Test_Cluster_Rep\"]].any(axis=1)]\n",
    "train_val_test_data"
   ],
   "id": "fb55be7b5bb58242",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "7     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "15    A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "16    A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "5024      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "5026      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5028      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "5033      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "7     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "15    Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "16    Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "5024                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "5026                                        Toxin ICK-9   \n",
       "5028                              Conotoxin Vc1 (H_Vc1)   \n",
       "5033                                       Toxin ICK-18   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "\n",
       "                           Protein families  \\\n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "7                      Phospholipase family   \n",
       "15                     Phospholipase family   \n",
       "16    Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "5024                       Conotoxin family   \n",
       "5026                      Neurotoxin family   \n",
       "5028                       Conotoxin family   \n",
       "5033                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "7     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN        8   \n",
       "15    MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...      NaN       35   \n",
       "16    MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN       36   \n",
       "...                                                 ...      ...      ...   \n",
       "5024  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN      744   \n",
       "5026  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN       19   \n",
       "5028  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN      746   \n",
       "5033  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN      748   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "3            True               True            False             False  \n",
       "4            True               True            False             False  \n",
       "7           False               True            False             False  \n",
       "15          False               True            False             False  \n",
       "16           True              False            False              True  \n",
       "...           ...                ...              ...               ...  \n",
       "5024        False               True            False             False  \n",
       "5026        False               True            False             False  \n",
       "5028         True               True            False             False  \n",
       "5033         True               True            False             False  \n",
       "5036        False               True            False             False  \n",
       "\n",
       "[1045 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5024</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>744</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5026</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5028</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>746</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5033</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1045 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 223
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.258634Z",
     "start_time": "2025-03-26T20:00:06.254668Z"
    }
   },
   "cell_type": "code",
   "source": [
    "no_na = train_val_test_data.dropna(subset=[\"Protein families\"]).copy()\n",
    "\n",
    "print(\"All representative sequences after 50% redundancy reduction:\", len(no_na))\n",
    "print(f\"Train entries: {no_na['Train_Cluster_Rep'].sum()} ({no_na['Train_Cluster_Rep'].sum()/len(no_na):.2%})\")\n",
    "print(f\"Validation entries: {no_na['Val_Cluster_Rep'].sum()} ({no_na['Val_Cluster_Rep'].sum()/len(no_na):.2%})\")\n",
    "print(f\"Test entries: {no_na['Test_Cluster_Rep'].sum()} ({no_na['Test_Cluster_Rep'].sum()/len(no_na):.2%})\")"
   ],
   "id": "bb801831b12fcb5c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All representative sequences after 50% redundancy reduction: 1045\n",
      "Train entries: 757 (72.44%)\n",
      "Validation entries: 164 (15.69%)\n",
      "Test entries: 124 (11.87%)\n"
     ]
    }
   ],
   "execution_count": 224
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.393505Z",
     "start_time": "2025-03-26T20:00:06.390685Z"
    }
   },
   "cell_type": "code",
   "source": "no_na[\"Protein families\"].value_counts()",
   "id": "78840daa8b5029e0",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                                       215\n",
       "Neurotoxin family                                      167\n",
       "Scoloptoxin family                                      83\n",
       "Short scorpion toxin superfamily                        69\n",
       "Natriuretic, Bradykinin potentiating peptide family     52\n",
       "                                                      ... \n",
       "Actinoporin family                                       1\n",
       "ConoGAY family                                           1\n",
       "Opioid neuropeptide precursor family                     1\n",
       "Diuretic hormone class 2 family                          1\n",
       "Cystatin family                                          1\n",
       "Name: count, Length: 93, dtype: int64"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 225
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:10.957297Z",
     "start_time": "2025-03-26T20:00:10.952689Z"
    }
   },
   "cell_type": "code",
   "source": [
    "print(no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() <= 10].sum())\n",
    "no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() <= 10]"
   ],
   "id": "7f3c8089b0715c9d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "199\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Long chain scorpion toxin family           10\n",
       "Bradykinin-related peptide family          10\n",
       "Teretoxin family                            9\n",
       "Vasopressin/oxytocin family                 8\n",
       "FARP (FMRFamide related peptide) family     8\n",
       "                                           ..\n",
       "Actinoporin family                          1\n",
       "ConoGAY family                              1\n",
       "Opioid neuropeptide precursor family        1\n",
       "Diuretic hormone class 2 family             1\n",
       "Cystatin family                             1\n",
       "Name: count, Length: 77, dtype: int64"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 226
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:15.657297Z",
     "start_time": "2025-03-26T20:00:15.652045Z"
    }
   },
   "cell_type": "code",
   "source": [
    "print(no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() > 10].sum())\n",
    "no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() > 10]"
   ],
   "id": "3623adb8c4ef986d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "846\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                                       215\n",
       "Neurotoxin family                                      167\n",
       "Scoloptoxin family                                      83\n",
       "Short scorpion toxin superfamily                        69\n",
       "Natriuretic, Bradykinin potentiating peptide family     52\n",
       "MCD family                                              41\n",
       "Long (4 C-C) scorpion toxin superfamily                 39\n",
       "Snake three-finger toxin family                         38\n",
       "Cationic peptide family                                 31\n",
       "Non-disulfide-bridged peptide (NDBP) superfamily        19\n",
       "Venom Kunitz-type family                                18\n",
       "Formicidae venom family                                 16\n",
       "Phospholipase family                                    16\n",
       "Long (3 C-C) scorpion toxin superfamily                 15\n",
       "Venom metalloproteinase (M12B) family                   15\n",
       "Limacoditoxin family                                    12\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 227
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### some interesting scoloptoxins are still in the data, nice",
   "id": "e0017834632f7562"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T11:05:55.401877Z",
     "start_time": "2025-03-26T11:05:55.396062Z"
    }
   },
   "cell_type": "code",
   "source": "no_na[no_na[\"Entry\"].isin([\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"])]",
   "id": "566e17256a3a9d55",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Entry                                           Organism  \\\n",
       "4187  P0DPX7  Scolopendra morsitans (Tanzanian blue ringleg ...   \n",
       "4188  P0DPX8            Ethmostigmus rubripes (Giant centipede)   \n",
       "4189  P0DPX9            Ethmostigmus rubripes (Giant centipede)   \n",
       "4190  P0DPY0            Ethmostigmus rubripes (Giant centipede)   \n",
       "4191  P0DPY1  Cormocephalus westwoodi (Westwood's green cent...   \n",
       "\n",
       "                                  Protein names    Protein families  \\\n",
       "4187  U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)  Scoloptoxin family   \n",
       "4188  U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)  Scoloptoxin family   \n",
       "4189  U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)  Scoloptoxin family   \n",
       "4190  U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)  Scoloptoxin family   \n",
       "4191  U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)  Scoloptoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "4187  MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...      NaN      393   \n",
       "4188  MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...      NaN      393   \n",
       "4189  MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...      NaN      393   \n",
       "4190  MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...      NaN      393   \n",
       "4191  MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...      NaN      393   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "4187        False              False             True             False  \n",
       "4188        False              False             True             False  \n",
       "4189        False              False             True             False  \n",
       "4190         True              False             True             False  \n",
       "4191        False              False             True             False  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4187</th>\n",
       "      <td>P0DPX7</td>\n",
       "      <td>Scolopendra morsitans (Tanzanian blue ringleg ...</td>\n",
       "      <td>U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4188</th>\n",
       "      <td>P0DPX8</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4189</th>\n",
       "      <td>P0DPX9</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4190</th>\n",
       "      <td>P0DPY0</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191</th>\n",
       "      <td>P0DPY1</td>\n",
       "      <td>Cormocephalus westwoodi (Westwood's green cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 177
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:43.066333Z",
     "start_time": "2025-03-26T20:00:42.965083Z"
    }
   },
   "cell_type": "code",
   "source": [
    "no_na[\"Protein families\"] = no_na[\"Protein families\"].apply(lambda x: \"other\" if no_na[\"Protein families\"].value_counts()[x] <= 10 else x)\n",
    "no_na"
   ],
   "id": "8df294feeae090d5",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "7     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "15    A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "16    A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "5024      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "5026      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5028      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "5033      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "7     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "15    Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "16    Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "5024                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "5026                                        Toxin ICK-9   \n",
       "5028                              Conotoxin Vc1 (H_Vc1)   \n",
       "5033                                       Toxin ICK-18   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "\n",
       "                           Protein families  \\\n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "7                      Phospholipase family   \n",
       "15                     Phospholipase family   \n",
       "16    Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "5024                       Conotoxin family   \n",
       "5026                      Neurotoxin family   \n",
       "5028                       Conotoxin family   \n",
       "5033                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "7     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN        8   \n",
       "15    MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...      NaN       35   \n",
       "16    MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN       36   \n",
       "...                                                 ...      ...      ...   \n",
       "5024  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN      744   \n",
       "5026  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN       19   \n",
       "5028  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN      746   \n",
       "5033  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN      748   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "3            True               True            False             False  \n",
       "4            True               True            False             False  \n",
       "7           False               True            False             False  \n",
       "15          False               True            False             False  \n",
       "16           True              False            False              True  \n",
       "...           ...                ...              ...               ...  \n",
       "5024        False               True            False             False  \n",
       "5026        False               True            False             False  \n",
       "5028         True               True            False             False  \n",
       "5033         True               True            False             False  \n",
       "5036        False               True            False             False  \n",
       "\n",
       "[1045 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5024</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>744</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5026</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5028</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>746</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5033</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1045 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 228
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:47.611959Z",
     "start_time": "2025-03-26T20:00:47.609008Z"
    }
   },
   "cell_type": "code",
   "source": "no_na[\"Protein families\"].value_counts()",
   "id": "a087ce5d75769362",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                                       215\n",
       "other                                                  199\n",
       "Neurotoxin family                                      167\n",
       "Scoloptoxin family                                      83\n",
       "Short scorpion toxin superfamily                        69\n",
       "Natriuretic, Bradykinin potentiating peptide family     52\n",
       "MCD family                                              41\n",
       "Long (4 C-C) scorpion toxin superfamily                 39\n",
       "Snake three-finger toxin family                         38\n",
       "Cationic peptide family                                 31\n",
       "Non-disulfide-bridged peptide (NDBP) superfamily        19\n",
       "Venom Kunitz-type family                                18\n",
       "Phospholipase family                                    16\n",
       "Formicidae venom family                                 16\n",
       "Venom metalloproteinase (M12B) family                   15\n",
       "Long (3 C-C) scorpion toxin superfamily                 15\n",
       "Limacoditoxin family                                    12\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 229
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:01:19.057685Z",
     "start_time": "2025-03-26T20:01:19.052416Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Get all unique protein families\n",
    "unique_protein_families = set(no_na[\"Protein families\"].unique())\n",
    "\n",
    "# Find unique protein families present in each cluster column separately\n",
    "train_families = set(no_na.loc[no_na[\"Train_Cluster_Rep\"] > 0, \"Protein families\"].unique())\n",
    "val_families = set(no_na.loc[no_na[\"Val_Cluster_Rep\"] > 0, \"Protein families\"].unique())\n",
    "test_families = set(no_na.loc[no_na[\"Test_Cluster_Rep\"] > 0, \"Protein families\"].unique())\n",
    "\n",
    "# Find missing families in each set\n",
    "missing_in_train = unique_protein_families - train_families\n",
    "missing_in_val = unique_protein_families - val_families\n",
    "missing_in_test = unique_protein_families - test_families\n",
    "\n",
    "# Check if all unique protein families are present in all three\n",
    "all_present_in_all = not (missing_in_train or missing_in_val or missing_in_test)\n",
    "\n",
    "# Print results\n",
    "print(f\"All unique protein families are present in all three columns: {all_present_in_all}\")\n",
    "\n",
    "if not all_present_in_all:\n",
    "    if missing_in_train:\n",
    "        print(f\"Missing in Train: {missing_in_train}\")\n",
    "    if missing_in_val:\n",
    "        print(f\"Missing in Val: {missing_in_val}\")\n",
    "    if missing_in_test:\n",
    "        print(f\"Missing in Test: {missing_in_test}\")"
   ],
   "id": "a35f8251fc75be06",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All unique protein families are present in all three columns: False\n",
      "Missing in Val: {'Venom Kunitz-type family'}\n",
      "Missing in Test: {'Formicidae venom family', 'Phospholipase family'}\n"
     ]
    }
   ],
   "execution_count": 230
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### added the scolops back for visualization only",
   "id": "6bfe5958481ec89d"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:00:14.888923Z",
     "start_time": "2025-03-23T10:00:14.880976Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# find duplicate entries\n",
    "duplicate_entries = set(no_na['Entry']).intersection(set(scolop['Entry']))\n",
    "\n",
    "# keep only unique scolop entries (not in no_na)\n",
    "unique_scolop = scolop[~scolop['Entry'].isin(duplicate_entries)].copy()\n",
    "\n",
    "# merge no_na with unique scolop entries\n",
    "merged_df = pd.concat([no_na, unique_scolop], ignore_index=True)\n",
    "merged_df"
   ],
   "id": "3c1ddd89cf359ef7",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "1     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "2     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "3     A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "4     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "1042      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1043      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1044      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1045      I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1046      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "1     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "2     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "3     Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "4     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "1042                              Conotoxin Vc1 (H_Vc1)   \n",
       "1043                                       Toxin ICK-18   \n",
       "1044   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "1045  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...   \n",
       "1046          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1                                     other   \n",
       "2                                     other   \n",
       "3                   Phospholipase A2 family   \n",
       "4     Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "1042                                  other   \n",
       "1043                                  other   \n",
       "1044          Neurotoxin 10 (Hwtx-1) family   \n",
       "1045                  Scoloptoxin-05 family   \n",
       "1046                  Scoloptoxin-05 family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN      8.0   \n",
       "1     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN     12.0   \n",
       "2     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN      8.0   \n",
       "3     MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...      NaN     35.0   \n",
       "4     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN     36.0   \n",
       "...                                                 ...      ...      ...   \n",
       "1042  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN    746.0   \n",
       "1043  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN    748.0   \n",
       "1044  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN    581.0   \n",
       "1045  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...      NaN      NaN   \n",
       "1046  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN      NaN   \n",
       "\n",
       "     Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \n",
       "0           True              True           False            False  \n",
       "1           True              True           False            False  \n",
       "2          False              True           False            False  \n",
       "3          False              True           False            False  \n",
       "4           True             False           False             True  \n",
       "...          ...               ...             ...              ...  \n",
       "1042        True              True           False            False  \n",
       "1043        True              True           False            False  \n",
       "1044       False              True           False            False  \n",
       "1045         NaN               NaN             NaN              NaN  \n",
       "1046         NaN               NaN             NaN              NaN  \n",
       "\n",
       "[1047 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>other</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>other</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase A2 family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1042</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>746.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1043</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin 10 (Hwtx-1) family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1045</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1047 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 63
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:01:14.555266Z",
     "start_time": "2025-03-23T10:01:14.549244Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df[merged_df[\"Entry\"].isin([\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"])]",
   "id": "19c912a36f1d52f8",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Entry                                           Organism  \\\n",
       "874   P0DPX7  Scolopendra morsitans (Tanzanian blue ringleg ...   \n",
       "875   P0DPX8            Ethmostigmus rubripes (Giant centipede)   \n",
       "876   P0DPX9            Ethmostigmus rubripes (Giant centipede)   \n",
       "877   P0DPY0            Ethmostigmus rubripes (Giant centipede)   \n",
       "878   P0DPY1  Cormocephalus westwoodi (Westwood's green cent...   \n",
       "1045  I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1046  P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "874           U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)   \n",
       "875           U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)   \n",
       "876           U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)   \n",
       "877           U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)   \n",
       "878           U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)   \n",
       "1045  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...   \n",
       "1046          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "           Protein families  \\\n",
       "874                   other   \n",
       "875                   other   \n",
       "876                   other   \n",
       "877                   other   \n",
       "878                   other   \n",
       "1045  Scoloptoxin-05 family   \n",
       "1046  Scoloptoxin-05 family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "874   MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...      NaN    393.0   \n",
       "875   MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...      NaN    393.0   \n",
       "876   MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...      NaN    393.0   \n",
       "877   MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...      NaN    393.0   \n",
       "878   MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...      NaN    393.0   \n",
       "1045  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...      NaN      NaN   \n",
       "1046  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN      NaN   \n",
       "\n",
       "     Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \n",
       "874        False             False            True            False  \n",
       "875        False             False            True            False  \n",
       "876        False             False            True            False  \n",
       "877         True             False            True            False  \n",
       "878        False             False            True            False  \n",
       "1045         NaN               NaN             NaN              NaN  \n",
       "1046         NaN               NaN             NaN              NaN  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>P0DPX7</td>\n",
       "      <td>Scolopendra morsitans (Tanzanian blue ringleg ...</td>\n",
       "      <td>U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>P0DPX8</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>P0DPX9</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>877</th>\n",
       "      <td>P0DPY0</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>878</th>\n",
       "      <td>P0DPY1</td>\n",
       "      <td>Cormocephalus westwoodi (Westwood's green cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1045</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 64
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:01:46.565107Z",
     "start_time": "2025-03-23T10:01:46.562366Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df.loc[merged_df[\"Protein families\"] == \"Scoloptoxin-05 family\", \"Protein families\"] = \"other\"",
   "id": "60cb67463e138fe5",
   "outputs": [],
   "execution_count": 65
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:01:51.143061Z",
     "start_time": "2025-03-23T10:01:51.139430Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df[\"Protein families\"].value_counts()",
   "id": "53f84dcfd80a0440",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "other                                               456\n",
       "Short scorpion toxin superfamily                     69\n",
       "Neurotoxin 10 (Hwtx-1) family                        42\n",
       "MCD family                                           41\n",
       "Long (4 C-C) scorpion toxin superfamily              39\n",
       "Snake three-finger toxin family                      38\n",
       "Bradykinin-potentiating peptide family               37\n",
       "Conotoxin O1 superfamily                             37\n",
       "Conotoxin M superfamily                              35\n",
       "Conotoxin A superfamily                              30\n",
       "Conotoxin O2 superfamily                             27\n",
       "Conotoxin T superfamily                              26\n",
       "Non-disulfide-bridged peptide (NDBP) superfamily     19\n",
       "Venom Kunitz-type family                             18\n",
       "Neurotoxin 02 (plectoxin) family                     18\n",
       "Cationic peptide 04 (cupiennin) family               18\n",
       "Neurotoxin 14 (magi-1) family                        16\n",
       "Natriuretic peptide family                           15\n",
       "Venom metalloproteinase (M12B) family                15\n",
       "Long (3 C-C) scorpion toxin superfamily              15\n",
       "Conotoxin I1 superfamily                             13\n",
       "Phospholipase A2 family                              12\n",
       "Formicidae venom precursor-01 superfamily            11\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 66
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:02:06.194758Z",
     "start_time": "2025-03-23T10:02:06.164128Z"
    }
   },
   "cell_type": "code",
   "source": [
    "write_fasta(merged_df, \"data/toxins_no_na_scolop.fasta\")\n",
    "write_fasta(no_na, \"data/toxins_no_na.fasta\")"
   ],
   "id": "8d9e91ecd8353902",
   "outputs": [],
   "execution_count": 67
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:02:09.200600Z",
     "start_time": "2025-03-23T10:02:09.197345Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df[\"Fragment\"].value_counts()",
   "id": "9df49c759f0209ab",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: count, dtype: int64)"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 68
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T11:06:21.590928Z",
     "start_time": "2025-03-26T11:06:21.584064Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df",
   "id": "d5342ea48bbcdb3",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "1     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "2     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "3     A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "4     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "1042      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1043      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1044      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1045      I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1046      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "1     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "2     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "3     Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "4     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "1042                              Conotoxin Vc1 (H_Vc1)   \n",
       "1043                                       Toxin ICK-18   \n",
       "1044   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "1045  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...   \n",
       "1046          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1                                     other   \n",
       "2                                     other   \n",
       "3                   Phospholipase A2 family   \n",
       "4     Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "1042                                  other   \n",
       "1043                                  other   \n",
       "1044          Neurotoxin 10 (Hwtx-1) family   \n",
       "1045                                  other   \n",
       "1046                                  other   \n",
       "\n",
       "                                               Sequence            Fragment  \\\n",
       "0     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...  mature_seq (NO_SP)   \n",
       "1     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...       full_seq (SP)   \n",
       "2     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...       full_seq (SP)   \n",
       "3     MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...  mature_seq (NO_SP)   \n",
       "4     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...       full_seq (SP)   \n",
       "...                                                 ...                 ...   \n",
       "1042  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...       full_seq (SP)   \n",
       "1043  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...       full_seq (SP)   \n",
       "1044  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...  mature_seq (NO_SP)   \n",
       "1045  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...       full_seq (SP)   \n",
       "1046  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...       full_seq (SP)   \n",
       "\n",
       "      Cluster Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \\\n",
       "0         8.0        True              True           False            False   \n",
       "1        12.0        True              True           False            False   \n",
       "2         8.0       False              True           False            False   \n",
       "3        35.0       False              True           False            False   \n",
       "4        36.0        True             False           False             True   \n",
       "...       ...         ...               ...             ...              ...   \n",
       "1042    746.0        True              True           False            False   \n",
       "1043    748.0        True              True           False            False   \n",
       "1044    581.0       False              True           False            False   \n",
       "1045      NaN         NaN               NaN             NaN              NaN   \n",
       "1046      NaN         NaN               NaN             NaN              NaN   \n",
       "\n",
       "      Scoloptoxin     protein_category  \n",
       "0           other  phospholipase_other  \n",
       "1           other          scoloptoxin  \n",
       "2           other  phospholipase_other  \n",
       "3           other     phospholipase_a2  \n",
       "4           other    metalloproteinase  \n",
       "...           ...                  ...  \n",
       "1042        other            conotoxin  \n",
       "1043        other           neurotoxin  \n",
       "1044        other           neurotoxin  \n",
       "1045  Scoloptoxin          scoloptoxin  \n",
       "1046  Scoloptoxin          scoloptoxin  \n",
       "\n",
       "[1047 rows x 13 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "      <th>Scoloptoxin</th>\n",
       "      <th>protein_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>other</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>8.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>12.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>other</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>8.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase A2 family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>35.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>36.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>metalloproteinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1042</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>746.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1043</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>748.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin 10 (Hwtx-1) family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>581.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1045</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1047 rows Ã— 13 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 180
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:02:40.667816Z",
     "start_time": "2025-03-23T10:02:40.660714Z"
    }
   },
   "cell_type": "code",
   "source": [
    "merged_df[\"Scoloptoxin\"] = merged_df.apply(\n",
    "    lambda row: \"Scoloptoxin\" if row[\"Entry\"] in [\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"]\n",
    "    else \"3FTX\" if row[\"Protein families\"] == \"Snake three-finger toxin family\"\n",
    "    else \"other\",\n",
    "    axis=1\n",
    ")\n",
    "\n",
    "#merged_df.to_csv(\"data/scolop.csv\", columns=[\"Entry\", \"Protein families\", \"Fragment\", \"Scoloptoxin\"], index=False, header=[\"identifier\", \"Protein families\", \"Fragment\", \"Scoloptoxin\"])"
   ],
   "id": "87994bfc3f14fe6a",
   "outputs": [],
   "execution_count": 70
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### add regex protein_category",
   "id": "f1e8a4e8e4a1dd27"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:01:35.351604Z",
     "start_time": "2025-03-26T20:01:35.331227Z"
    }
   },
   "cell_type": "code",
   "source": [
    "regex_tox = pd.read_csv(\"data/toxins_regex.csv\")\n",
    "regex_tox"
   ],
   "id": "745af5ab2073cdb",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                                       Protein families  \\\n",
       "0                               Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family, P-III s...   \n",
       "2                                        Insulin family   \n",
       "3                                        Insulin family   \n",
       "4     Arthropod phospholipase D family, Class II sub...   \n",
       "...                                                 ...   \n",
       "6423                                                NaN   \n",
       "6424                                                NaN   \n",
       "6425                                                NaN   \n",
       "6426                                                NaN   \n",
       "6427                                                NaN   \n",
       "\n",
       "                                               Sequence  Fragment  \\\n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment   \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN   \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN   \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN   \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment   \n",
       "...                                                 ...       ...   \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN   \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN   \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN   \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment   \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment   \n",
       "\n",
       "         protein_category  \n",
       "0               conotoxin  \n",
       "1       metalloproteinase  \n",
       "2         hormone_related  \n",
       "3         hormone_related  \n",
       "4     phospholipase_other  \n",
       "...                   ...  \n",
       "6423                  NaN  \n",
       "6424                  NaN  \n",
       "6425                  NaN  \n",
       "6426                  NaN  \n",
       "6427                  NaN  \n",
       "\n",
       "[6428 rows x 7 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>protein_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family, P-III s...</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metalloproteinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hormone_related</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hormone_related</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family, Class II sub...</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 7 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 231
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:01:36.422551Z",
     "start_time": "2025-03-26T20:01:36.410629Z"
    }
   },
   "cell_type": "code",
   "source": [
    "merged_df = merged_df.merge(regex_tox[['Entry', 'protein_category']], on='Entry', how='left')\n",
    "no_na = no_na.merge(regex_tox[['Entry', 'protein_category']], on='Entry', how='left')\n",
    "\n",
    "merged_df # supposed to be 1393 rows"
   ],
   "id": "977b1322c6f129ff",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "1     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "2     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "3     A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "4     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "1042      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1043      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1044      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1045      I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1046      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "1     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "2     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "3     Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "4     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "1042                              Conotoxin Vc1 (H_Vc1)   \n",
       "1043                                       Toxin ICK-18   \n",
       "1044   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "1045  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...   \n",
       "1046          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1                                     other   \n",
       "2                                     other   \n",
       "3                   Phospholipase A2 family   \n",
       "4     Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "1042                                  other   \n",
       "1043                                  other   \n",
       "1044          Neurotoxin 10 (Hwtx-1) family   \n",
       "1045                                  other   \n",
       "1046                                  other   \n",
       "\n",
       "                                               Sequence            Fragment  \\\n",
       "0     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...  mature_seq (NO_SP)   \n",
       "1     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...       full_seq (SP)   \n",
       "2     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...       full_seq (SP)   \n",
       "3     MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...  mature_seq (NO_SP)   \n",
       "4     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...       full_seq (SP)   \n",
       "...                                                 ...                 ...   \n",
       "1042  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...       full_seq (SP)   \n",
       "1043  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...       full_seq (SP)   \n",
       "1044  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...  mature_seq (NO_SP)   \n",
       "1045  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...       full_seq (SP)   \n",
       "1046  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...       full_seq (SP)   \n",
       "\n",
       "      Cluster Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \\\n",
       "0         8.0        True              True           False            False   \n",
       "1        12.0        True              True           False            False   \n",
       "2         8.0       False              True           False            False   \n",
       "3        35.0       False              True           False            False   \n",
       "4        36.0        True             False           False             True   \n",
       "...       ...         ...               ...             ...              ...   \n",
       "1042    746.0        True              True           False            False   \n",
       "1043    748.0        True              True           False            False   \n",
       "1044    581.0       False              True           False            False   \n",
       "1045      NaN         NaN               NaN             NaN              NaN   \n",
       "1046      NaN         NaN               NaN             NaN              NaN   \n",
       "\n",
       "      Scoloptoxin   protein_category_x   protein_category_y  \\\n",
       "0           other  phospholipase_other  phospholipase_other   \n",
       "1           other          scoloptoxin          scoloptoxin   \n",
       "2           other  phospholipase_other  phospholipase_other   \n",
       "3           other     phospholipase_a2     phospholipase_a2   \n",
       "4           other    metalloproteinase    metalloproteinase   \n",
       "...           ...                  ...                  ...   \n",
       "1042        other            conotoxin            conotoxin   \n",
       "1043        other           neurotoxin           neurotoxin   \n",
       "1044        other           neurotoxin           neurotoxin   \n",
       "1045  Scoloptoxin          scoloptoxin          scoloptoxin   \n",
       "1046  Scoloptoxin          scoloptoxin          scoloptoxin   \n",
       "\n",
       "         protein_category  \n",
       "0     phospholipase_other  \n",
       "1             scoloptoxin  \n",
       "2     phospholipase_other  \n",
       "3        phospholipase_a2  \n",
       "4       metalloproteinase  \n",
       "...                   ...  \n",
       "1042            conotoxin  \n",
       "1043           neurotoxin  \n",
       "1044           neurotoxin  \n",
       "1045          scoloptoxin  \n",
       "1046          scoloptoxin  \n",
       "\n",
       "[1047 rows x 15 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "      <th>Scoloptoxin</th>\n",
       "      <th>protein_category_x</th>\n",
       "      <th>protein_category_y</th>\n",
       "      <th>protein_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>other</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>8.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>12.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>other</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>8.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase A2 family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>35.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>36.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>metalloproteinase</td>\n",
       "      <td>metalloproteinase</td>\n",
       "      <td>metalloproteinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1042</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>746.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>conotoxin</td>\n",
       "      <td>conotoxin</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1043</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>748.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin 10 (Hwtx-1) family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>581.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1045</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1047 rows Ã— 15 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 232
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:02:49.202061Z",
     "start_time": "2025-03-23T10:02:49.196675Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df.to_csv(\"data/scolop.csv\", columns=[\"Entry\", \"Protein families\", \"Fragment\", \"Scoloptoxin\", \"protein_category\"], index=False, header=[\"identifier\", \"Protein families\", \"Fragment\", \"Scoloptoxin\", \"protein_category\"])",
   "id": "19e2b93a29fb2730",
   "outputs": [],
   "execution_count": 73
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### cut off SP and then see which Entry is full_seq, mature or fragment",
   "id": "9a49267cb1167227"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:03:02.025441Z",
     "start_time": "2025-03-23T10:03:02.007852Z"
    }
   },
   "cell_type": "code",
   "source": "write_fasta(merged_df, \"data/toxins_no_na_scolop.fasta\")",
   "id": "c1bb5d14e430b550",
   "outputs": [],
   "execution_count": 74
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:01:51.987803Z",
     "start_time": "2025-03-26T20:01:51.981338Z"
    }
   },
   "cell_type": "code",
   "source": [
    "sp6_results = pd.read_csv(\"data/sp6/output.gff3\", sep=\"\\t\", usecols=[0]).rename(columns={\"## gff-version 3\": \"Entry\"})\n",
    "sp6_results"
   ],
   "id": "5d276eba84c1e33c",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "          Entry\n",
       "0    A0A0N7CSQ4\n",
       "1    A0A1L4BJ98\n",
       "2    A0A6B7FMR5\n",
       "3        A0S864\n",
       "4        A8YPR6\n",
       "..          ...\n",
       "590      W4VSB6\n",
       "591      W4VSB9\n",
       "592      W4VSG7\n",
       "593      W4VSI6\n",
       "594      P0DPX5\n",
       "\n",
       "[595 rows x 1 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0S864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A8YPR6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>W4VSB6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>W4VSB9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>W4VSG7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>W4VSI6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>594</th>\n",
       "      <td>P0DPX5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>595 rows Ã— 1 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 233
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:01:52.202403Z",
     "start_time": "2025-03-26T20:01:52.191804Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Merge to check which entries are in sp6_results\n",
    "merged_df = merged_df.merge(sp6_results[['Entry']], on='Entry', how='left', indicator=True)\n",
    "no_na = no_na.merge(sp6_results[['Entry']], on='Entry', how='left', indicator=True)\n",
    "\n",
    "# Replace NaN values in \"Fragment\" based on presence in sp6_results\n",
    "merged_df['Fragment'] = np.where(\n",
    "    merged_df['Fragment'].isna(),\n",
    "    np.where(merged_df['_merge'] == 'both', 'full_seq (SP)', 'mature_seq (NO_SP)'),\n",
    "    merged_df['Fragment']\n",
    ")\n",
    "no_na['Fragment'] = np.where(\n",
    "    no_na['Fragment'].isna(),\n",
    "    np.where(no_na['_merge'] == 'both', 'full_seq (SP)', 'mature_seq (NO_SP)'),\n",
    "    no_na['Fragment']\n",
    ")\n",
    "\n",
    "# Drop the '_merge' column as it's no longer needed\n",
    "merged_df.drop(columns=['_merge'], inplace=True)\n",
    "no_na.drop(columns=['_merge'], inplace=True)\n",
    "merged_df"
   ],
   "id": "7c4ae678c6d4e031",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "1     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "2     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "3     A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "4     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "1042      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1043      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1044      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1045      I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1046      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "1     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "2     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "3     Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "4     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "1042                              Conotoxin Vc1 (H_Vc1)   \n",
       "1043                                       Toxin ICK-18   \n",
       "1044   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "1045  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...   \n",
       "1046          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1                                     other   \n",
       "2                                     other   \n",
       "3                   Phospholipase A2 family   \n",
       "4     Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "1042                                  other   \n",
       "1043                                  other   \n",
       "1044          Neurotoxin 10 (Hwtx-1) family   \n",
       "1045                                  other   \n",
       "1046                                  other   \n",
       "\n",
       "                                               Sequence            Fragment  \\\n",
       "0     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...  mature_seq (NO_SP)   \n",
       "1     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...       full_seq (SP)   \n",
       "2     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...       full_seq (SP)   \n",
       "3     MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...  mature_seq (NO_SP)   \n",
       "4     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...       full_seq (SP)   \n",
       "...                                                 ...                 ...   \n",
       "1042  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...       full_seq (SP)   \n",
       "1043  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...       full_seq (SP)   \n",
       "1044  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...  mature_seq (NO_SP)   \n",
       "1045  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...       full_seq (SP)   \n",
       "1046  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...       full_seq (SP)   \n",
       "\n",
       "      Cluster Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \\\n",
       "0         8.0        True              True           False            False   \n",
       "1        12.0        True              True           False            False   \n",
       "2         8.0       False              True           False            False   \n",
       "3        35.0       False              True           False            False   \n",
       "4        36.0        True             False           False             True   \n",
       "...       ...         ...               ...             ...              ...   \n",
       "1042    746.0        True              True           False            False   \n",
       "1043    748.0        True              True           False            False   \n",
       "1044    581.0       False              True           False            False   \n",
       "1045      NaN         NaN               NaN             NaN              NaN   \n",
       "1046      NaN         NaN               NaN             NaN              NaN   \n",
       "\n",
       "      Scoloptoxin   protein_category_x   protein_category_y  \\\n",
       "0           other  phospholipase_other  phospholipase_other   \n",
       "1           other          scoloptoxin          scoloptoxin   \n",
       "2           other  phospholipase_other  phospholipase_other   \n",
       "3           other     phospholipase_a2     phospholipase_a2   \n",
       "4           other    metalloproteinase    metalloproteinase   \n",
       "...           ...                  ...                  ...   \n",
       "1042        other            conotoxin            conotoxin   \n",
       "1043        other           neurotoxin           neurotoxin   \n",
       "1044        other           neurotoxin           neurotoxin   \n",
       "1045  Scoloptoxin          scoloptoxin          scoloptoxin   \n",
       "1046  Scoloptoxin          scoloptoxin          scoloptoxin   \n",
       "\n",
       "         protein_category  \n",
       "0     phospholipase_other  \n",
       "1             scoloptoxin  \n",
       "2     phospholipase_other  \n",
       "3        phospholipase_a2  \n",
       "4       metalloproteinase  \n",
       "...                   ...  \n",
       "1042            conotoxin  \n",
       "1043           neurotoxin  \n",
       "1044           neurotoxin  \n",
       "1045          scoloptoxin  \n",
       "1046          scoloptoxin  \n",
       "\n",
       "[1047 rows x 15 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "      <th>Scoloptoxin</th>\n",
       "      <th>protein_category_x</th>\n",
       "      <th>protein_category_y</th>\n",
       "      <th>protein_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>other</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>8.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>12.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>other</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>8.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase A2 family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>35.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "      <td>phospholipase_a2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>36.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>metalloproteinase</td>\n",
       "      <td>metalloproteinase</td>\n",
       "      <td>metalloproteinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1042</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>746.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>conotoxin</td>\n",
       "      <td>conotoxin</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1043</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>748.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin 10 (Hwtx-1) family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>mature_seq (NO_SP)</td>\n",
       "      <td>581.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1045</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>full_seq (SP)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1047 rows Ã— 15 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 234
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-23T10:03:07.810425Z",
     "start_time": "2025-03-23T10:03:07.805030Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df.to_csv(\"data/scolop.csv\", columns=[\"Entry\", \"Protein families\", \"Fragment\", \"Scoloptoxin\", \"protein_category\"], index=False, header=[\"identifier\", \"Protein families\", \"Fragment\", \"Scoloptoxin\", \"protein_category\"])",
   "id": "2186967bd37501aa",
   "outputs": [],
   "execution_count": 77
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:01:55.418665Z",
     "start_time": "2025-03-26T20:01:55.409677Z"
    }
   },
   "cell_type": "code",
   "source": "no_na.to_csv(\"data/toxins_no_na.tsv\", sep=\"\\t\", index=False)",
   "id": "6524f466ad8dca57",
   "outputs": [],
   "execution_count": 235
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "",
   "id": "ba2463a3300be816",
   "outputs": [],
   "execution_count": null
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
