{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-03-05T09:53:52.015795Z",
     "start_time": "2025-03-05T09:53:52.013757Z"
    }
   },
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n",
    "from sklearn.utils import shuffle"
   ],
   "outputs": [],
   "execution_count": 55
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:53:52.298937Z",
     "start_time": "2025-03-05T09:53:52.280820Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df = pd.read_csv('data/toxins1.tsv', sep='\\t')\n",
    "df"
   ],
   "id": "8a325e392f034880",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                                       Protein families  \\\n",
       "0                               Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family, P-III s...   \n",
       "2                                        Insulin family   \n",
       "3                                        Insulin family   \n",
       "4     Arthropod phospholipase D family, Class II sub...   \n",
       "...                                                 ...   \n",
       "6423                                                NaN   \n",
       "6424                                                NaN   \n",
       "6425                                                NaN   \n",
       "6426                                                NaN   \n",
       "6427                                                NaN   \n",
       "\n",
       "                                               Sequence  Fragment  \n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment  \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN  \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN  \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN  \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment  \n",
       "...                                                 ...       ...  \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN  \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN  \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN  \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment  \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment  \n",
       "\n",
       "[6428 rows x 6 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family, P-III s...</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family, Class II sub...</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 6 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 56
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:53:53.822657Z",
     "start_time": "2025-03-05T09:53:53.813928Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df['Protein families'] = df['Protein families'].str.split(',').str[0]\n",
    "df['Protein families'] = df['Protein families'].str.split(';').str[0]\n",
    "\n",
    "df[\"Protein families\"].value_counts()"
   ],
   "id": "614d78cba278e28f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Snake three-finger toxin family            532\n",
       "Long (4 C-C) scorpion toxin superfamily    418\n",
       "Phospholipase A2 family                    414\n",
       "Neurotoxin 10 (Hwtx-1) family              289\n",
       "Venom metalloproteinase (M12B) family      281\n",
       "                                          ... \n",
       "Scoloptoxin family                           1\n",
       "Scolopendra neurotoxin 8 family              1\n",
       "Scolopendra toxin 6 family                   1\n",
       "Scolopendra toxin 7 family                   1\n",
       "SLPTX(15) family                             1\n",
       "Name: count, Length: 235, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 57
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:53:59.663979Z",
     "start_time": "2025-03-05T09:53:59.658567Z"
    }
   },
   "cell_type": "code",
   "source": [
    "filtered = df[df['Protein families'].str.contains('superfamily', na=False)]['Protein families'].unique()\n",
    "filtered"
   ],
   "id": "e6959ed706b8aaa2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Conotoxin A superfamily', 'Conotoxin O1 superfamily',\n",
       "       'Conotoxin M superfamily',\n",
       "       'Long (4 C-C) scorpion toxin superfamily',\n",
       "       'Short scorpion toxin superfamily', 'Conotoxin B superfamily',\n",
       "       'Formicidae venom precursor-01 superfamily',\n",
       "       'Conotoxin D superfamily', 'O2 superfamily',\n",
       "       'AB hydrolase superfamily', 'Conotoxin T superfamily',\n",
       "       'Non-disulfide-bridged peptide (NDBP) superfamily',\n",
       "       'Long (3 C-C) scorpion toxin superfamily',\n",
       "       'Conotoxin J superfamily', 'Conotoxin O2 superfamily',\n",
       "       'Conotoxin S superfamily', 'Conotoxin C superfamily',\n",
       "       'Conotoxin L superfamily', 'Conotoxin I2 superfamily',\n",
       "       'Conotoxin I1 superfamily', 'Conotoxin P superfamily',\n",
       "       'Conotoxin U superfamily', 'Conotoxin V superfamily',\n",
       "       'Conotoxin Y superfamily', 'Conotoxin K superfamily',\n",
       "       'Conotoxin I3 superfamily', 'Pg turripeptide superfamily',\n",
       "       'Conopeptide P-like superfamily', 'O1 superfamily',\n",
       "       'Conotoxin H superfamily', 'Conotoxin NSf-1 superfamily',\n",
       "       'Conotoxin G superfamily', 'Conotoxin R superfamily',\n",
       "       'Conotoxin Q superfamily', 'Conotoxin E superfamily',\n",
       "       'Conotoxin N superfamily', 'Conopeptide I2-like superfamily',\n",
       "       'E superfamily', 'F superfamily', 'Teretoxin C (TC) superfamily',\n",
       "       'Teretoxin A (TA) superfamily', 'Teretoxin M (TM) superfamily',\n",
       "       'Teretoxin H (TH) superfamily', 'Teretoxin N (TN) superfamily',\n",
       "       'Teretoxin J (TJ) superfamily', 'Conotoxin O3 superfamily',\n",
       "       'I1 superfamily'], dtype=object)"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 58
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:00.655701Z",
     "start_time": "2025-03-05T09:54:00.648595Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df['Protein families'] = df['Protein families'].replace('I1 superfamily', 'Conotoxin I1 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('O1 superfamily', 'Conotoxin 01 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('O2 superfamily', 'Conotoxin 02 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('E superfamily', 'Conotoxin E superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('F superfamily', 'Conotoxin F superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin M family', 'Conotoxin M superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin B2 family', 'Conotoxin B2 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin O1 family', 'Conotoxin O1 superfamily')\n",
    "df['Protein families'] = df['Protein families'].replace('Conotoxin O2 family', 'Conotoxin O2 superfamily')\n"
   ],
   "id": "92750c3a996def3e",
   "outputs": [],
   "execution_count": 59
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:01.611251Z",
     "start_time": "2025-03-05T09:54:01.607538Z"
    }
   },
   "cell_type": "code",
   "source": "df[\"Protein families\"].value_counts()",
   "id": "7918b6efed4306c4",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Snake three-finger toxin family            532\n",
       "Long (4 C-C) scorpion toxin superfamily    418\n",
       "Phospholipase A2 family                    414\n",
       "Neurotoxin 10 (Hwtx-1) family              289\n",
       "Venom metalloproteinase (M12B) family      281\n",
       "                                          ... \n",
       "Teretoxin N (TN) superfamily                 1\n",
       "Teretoxin J (TJ) superfamily                 1\n",
       "Conotoxin B2 superfamily                     1\n",
       "Scutigerotoxin-01 family                     1\n",
       "SLPTX(15) family                             1\n",
       "Name: count, Length: 230, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 60
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:03.361848Z",
     "start_time": "2025-03-05T09:54:03.356740Z"
    }
   },
   "cell_type": "code",
   "source": [
    "scolop = df[df[\"Entry\"].isin([\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"])]\n",
    "scolop"
   ],
   "id": "de40e54c5b02400d",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Entry                                           Organism  \\\n",
       "2440  I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "5337  P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "5339  P0DPX7  Scolopendra morsitans (Tanzanian blue ringleg ...   \n",
       "5340  P0DPX8            Ethmostigmus rubripes (Giant centipede)   \n",
       "5341  P0DPX9            Ethmostigmus rubripes (Giant centipede)   \n",
       "5342  P0DPY0            Ethmostigmus rubripes (Giant centipede)   \n",
       "5343  P0DPY1  Cormocephalus westwoodi (Westwood's green cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "2440  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...   \n",
       "5337          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "5339          U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)   \n",
       "5340          U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)   \n",
       "5341          U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)   \n",
       "5342          U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)   \n",
       "5343          U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)   \n",
       "\n",
       "           Protein families  \\\n",
       "2440  Scoloptoxin-05 family   \n",
       "5337  Scoloptoxin-05 family   \n",
       "5339  Scoloptoxin-05 family   \n",
       "5340  Scoloptoxin-05 family   \n",
       "5341  Scoloptoxin-05 family   \n",
       "5342  Scoloptoxin-05 family   \n",
       "5343  Scoloptoxin-05 family   \n",
       "\n",
       "                                               Sequence Fragment  \n",
       "2440  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...      NaN  \n",
       "5337  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN  \n",
       "5339  MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...      NaN  \n",
       "5340  MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...      NaN  \n",
       "5341  MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...      NaN  \n",
       "5342  MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...      NaN  \n",
       "5343  MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...      NaN  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2440</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5337</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>P0DPX7</td>\n",
       "      <td>Scolopendra morsitans (Tanzanian blue ringleg ...</td>\n",
       "      <td>U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5340</th>\n",
       "      <td>P0DPX8</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5341</th>\n",
       "      <td>P0DPX9</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5342</th>\n",
       "      <td>P0DPY0</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5343</th>\n",
       "      <td>P0DPY1</td>\n",
       "      <td>Cormocephalus westwoodi (Westwood's green cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 61
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:05.536035Z",
     "start_time": "2025-03-05T09:54:05.459630Z"
    }
   },
   "cell_type": "code",
   "source": [
    "def write_fasta(df, filename):\n",
    "    \"\"\"Writes a DataFrame to a FASTA file.\"\"\"\n",
    "    with open(filename, \"w\") as f:\n",
    "        for _, row in df.iterrows():\n",
    "            f.write(f\">{row['Entry']}\\n{row['Sequence']}\\n\")\n",
    "\n",
    "write_fasta(df, \"data/toxins.fasta\")"
   ],
   "id": "50bc1a6313d9e4ac",
   "outputs": [],
   "execution_count": 62
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-03T13:09:02.948675Z",
     "start_time": "2025-03-03T13:08:59.930072Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins.fasta data/mmseqs_0.3/toxins_cluster data/mmseqs_0.3/tmp --min-seq-id 0.3",
   "id": "e5c8c71cdab57aaa",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory data/mmseqs2/tmp\r\n",
      "easy-cluster data/toxins.fasta data/mmseqs2/toxins_cluster data/mmseqs2/tmp --min-seq-id 0.3 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.3\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins.fasta data/mmseqs2/tmp/8096837924678325838/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[6364] 0s 2ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "Create directory data/mmseqs2/tmp/8096837924678325838/clu_tmp\r\n",
      "cluster data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu data/mmseqs2/tmp/8096837924678325838/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.3 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 5.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.3 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 6428 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 6.43K 0s 16ms     \r\n",
      "Sort kmer 0h 0m 0s 1ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 1ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 24ms\r\n",
      "rescorediagonal data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 6.43K 0s 9ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms=================> ] 98.10% 6.31K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "clust data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_rescore1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 6.43K 0s 6ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 10232 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 6.43K 0s 7ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 6.43K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 16ms\r\n",
      "Total time: 0h 0m 0s 17ms\r\n",
      "\r\n",
      "Size of the sequence database: 6428\r\n",
      "Size of the alignment database: 6428\r\n",
      "Number of clusters: 2743\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 19ms\r\n",
      "createsubdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/order_redundancy data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/order_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_filter1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_filter2 --filter-file data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 2.74K 0s 5ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rescorediagonal data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_filter2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 2.74K 0s 5ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 1ms===============>   ] 94.31% 2.59K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "align data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_rescore2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 2743 type: Aminoacid\r\n",
      "Target database size: 2743 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 2.74K 0s 32ms                                        ] 40.85% 1.12K eta 0s       \r\n",
      "Time for merging to aln: 0h 0m 0s 1ms\r\n",
      "3285 alignments calculated\r\n",
      "3200 sequence pairs passed the thresholds (0.974125 of overall calculated)\r\n",
      "1.166606 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 37ms\r\n",
      "clust data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/aln data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 2.74K 0s 4ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 457 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 2.74K 0s 4ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 2.74K 0a 0s       s 1ms\r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 11ms\r\n",
      "Total time: 0h 0m 0s 11ms\r\n",
      "\r\n",
      "Size of the sequence database: 2743\r\n",
      "Size of the alignment database: 2743\r\n",
      "Number of clusters: 2423\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 13ms\r\n",
      "mergeclusters data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pre_clust data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 2.74K 0s 4ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 2.42K 0s 8ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 6.43K 0s 14ms     \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 17ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/linclust/9665814252750331155/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_redundancy data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 2423 type: Aminoacid\r\n",
      "Estimated memory consumption: 980M\r\n",
      "Target database size: 2423 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 2.42K 0s 9ms      \r\n",
      "Index table: Masked residues: 3870\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 2.42K 0s 8ms      \r\n",
      "Index statistics\r\n",
      "Entries:          98514\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.001539\r\n",
      "Top 10 k-mers\r\n",
      "    TICDYT\t32\r\n",
      "    KCNKCY\t24\r\n",
      "    NPCACK\t23\r\n",
      "    ECIEFL\t20\r\n",
      "    PCDAKL\t20\r\n",
      "    GNCCLP\t20\r\n",
      "    CNNHLV\t20\r\n",
      "    AHLHGM\t16\r\n",
      "    EDMYKV\t16\r\n",
      "    GLCCLP\t15\r\n",
      "Time for index table init: 0h 0m 0s 116ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 2423\r\n",
      "Target db start 1 to 2423\r\n",
      "[=================================================================] 100.00% 2.42K 0s 13ms     \r\n",
      "\r\n",
      "7.202269 k-mers per position\r\n",
      "96 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "6 sequences passed prefiltering per query sequence\r\n",
      "3 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 309ms\r\n",
      "align data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step0 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 2423 type: Aminoacid\r\n",
      "Target database size: 2423 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 2.42K 0s 41ms     \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\r\n",
      "8816 alignments calculated\r\n",
      "7776 sequence pairs passed the thresholds (0.882033 of overall calculated)\r\n",
      "3.209245 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 46ms\r\n",
      "clust data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step0 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 2.42K 0s 4ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 537 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 2.42K 0s 5ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 2.42K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 11ms\r\n",
      "Total time: 0h 0m 0s 12ms\r\n",
      "\r\n",
      "Size of the sequence database: 2423\r\n",
      "Size of the alignment database: 2423\r\n",
      "Number of clusters: 1570\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "createsubdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step0 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1570 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1570 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.57K 0s 8ms      \r\n",
      "Index table: Masked residues: 3077\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.57K 0s 6ms      \r\n",
      "Index statistics\r\n",
      "Entries:          81064\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.001267\r\n",
      "Top 10 k-mers\r\n",
      "    MKTVVA\t13\r\n",
      "    MAEGLG\t10\r\n",
      "    GGENEH\t10\r\n",
      "    VALLCQ\t10\r\n",
      "    NPCACK\t9\r\n",
      "    VIGEIN\t8\r\n",
      "    YCGSCP\t7\r\n",
      "    QCELDQ\t7\r\n",
      "    KLCLAV\t7\r\n",
      "    CNNHLV\t7\r\n",
      "Time for index table init: 0h 0m 0s 112ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1570\r\n",
      "Target db start 1 to 1570\r\n",
      "[=================================================================] 100.00% 1.57K 0s 12ms     \r\n",
      "\r\n",
      "31.309548 k-mers per position\r\n",
      "69 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "3 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 310ms\r\n",
      "align data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1570 type: Aminoacid\r\n",
      "Target database size: 1570 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.57K 0s 9ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "2342 alignments calculated\r\n",
      "1854 sequence pairs passed the thresholds (0.791631 of overall calculated)\r\n",
      "1.180892 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "clust data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.57K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 70 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.57K 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.57K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 9ms\r\n",
      "Total time: 0h 0m 0s 9ms\r\n",
      "\r\n",
      "Size of the sequence database: 1570\r\n",
      "Size of the alignment database: 1570\r\n",
      "Number of clusters: 1419\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "createsubdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 5 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1419 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1419 type: Aminoacid\r\n",
      "Index table k-mer threshold: 118 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.42K 0s 7ms      \r\n",
      "Index table: Masked residues: 3007\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.42K 0s 8ms      \r\n",
      "Index statistics\r\n",
      "Entries:          76108\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.001189\r\n",
      "Top 10 k-mers\r\n",
      "    EEFEYE\t9\r\n",
      "    MAEGLG\t9\r\n",
      "    GCGKRI\t9\r\n",
      "    NPCACK\t8\r\n",
      "    DDNLCF\t7\r\n",
      "    AVMHGH\t7\r\n",
      "    LTVIVL\t7\r\n",
      "    VIGEIN\t7\r\n",
      "    QCELDQ\t7\r\n",
      "    KLRGSG\t6\r\n",
      "Time for index table init: 0h 0m 0s 112ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 118\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1419\r\n",
      "Target db start 1 to 1419\r\n",
      "[=================================================================] 100.00% 1.42K 0s 31ms     \r\n",
      "\r\n",
      "190.565581 k-mers per position\r\n",
      "103 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "4 sequences passed prefiltering per query sequence\r\n",
      "3 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 324ms\r\n",
      "align data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.3 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1419 type: Aminoacid\r\n",
      "Target database size: 1419 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.42K 0s 10ms     ===================================>        ] 86.53% 1.23K eta 0s       \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "2651 alignments calculated\r\n",
      "1628 sequence pairs passed the thresholds (0.614108 of overall calculated)\r\n",
      "1.147287 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "clust data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step2 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.42K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 57 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.42K 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.42K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 9ms\r\n",
      "\r\n",
      "Size of the sequence database: 1419\r\n",
      "Size of the alignment database: 1419\r\n",
      "Number of clusters: 1301\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "mergeclusters data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_redundancy data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step0 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step1 data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 2.42K 0s 4ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 1.57K 0s 7ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 1.42K 0s 10ms     \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 1.30K 0s 14ms     \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 6.43K 0s 18ms     \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 21ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_tmp/17343105067443434066/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu data/mmseqs2/tmp/8096837924678325838/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2repseq data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu data/mmseqs2/tmp/8096837924678325838/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.30K 0s 3ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "result2flat data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_rep data/mmseqs2/tmp/8096837924678325838/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createseqfiledb data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu data/mmseqs2/tmp/8096837924678325838/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.30K 0s 4ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "result2flat data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/input data/mmseqs2/tmp/8096837924678325838/clu_seqs data/mmseqs2/tmp/8096837924678325838/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs2/tmp/8096837924678325838/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 40
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-03T13:29:54.402619Z",
     "start_time": "2025-03-03T13:29:54.400531Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# rep_data = []\n",
    "# with open(\"data/mmseqs_0.3/toxins_rep.fasta\") as f:\n",
    "#     for record in SeqIO.parse(f, \"fasta\"):\n",
    "#         rep_data.append([record.id, str(record.seq)])\n",
    "#\n",
    "# toxins_rep = pd.DataFrame(rep_data, columns=[\"Entry\", \"Sequence\"])\n",
    "#\n",
    "# toxins_rep = toxins_rep.merge(df[[\"Entry\", \"Protein families\"]], on=\"Entry\", how=\"left\")\n",
    "# toxins_rep"
   ],
   "id": "489dcf49987c9f15",
   "outputs": [],
   "execution_count": 42
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-03T13:29:54.900952Z",
     "start_time": "2025-03-03T13:29:54.899337Z"
    }
   },
   "cell_type": "code",
   "source": "# toxins_rep[\"Protein families\"].value_counts()",
   "id": "56f9fa13754606f4",
   "outputs": [],
   "execution_count": 43
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:07.861496Z",
     "start_time": "2025-03-05T09:54:07.854883Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cluster_df = pd.read_csv(\"data/mmseqs_0.3/toxins_cluster_cluster.tsv\", sep='\\t', header=None, names=['Entry', 'cluster_member'])\n",
    "cluster_df"
   ],
   "id": "4c6b68af0330835b",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry cluster_member\n",
       "0     A0A0N7CSQ4     A0A0N7CSQ4\n",
       "1     A0A1Z0YU59     A0A1Z0YU59\n",
       "2     A0A1Z0YU59         C0HLA5\n",
       "3     A0A1Z0YU59         C0HLA6\n",
       "4     A0A1Z0YU59         C0HLA7\n",
       "...          ...            ...\n",
       "6423      D6C4M1         D2Y3T3\n",
       "6424      D6C4M1         D6C4M0\n",
       "6425      D6C4M1         D6C4M2\n",
       "6426      D6C4M3         D6C4M3\n",
       "6427      D6C4M3         B6V6L0\n",
       "\n",
       "[6428 rows x 2 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>cluster_member</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>C0HLA5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>C0HLA6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>C0HLA7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>D6C4M1</td>\n",
       "      <td>D2Y3T3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>D6C4M1</td>\n",
       "      <td>D6C4M0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>D6C4M1</td>\n",
       "      <td>D6C4M2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>D6C4M3</td>\n",
       "      <td>D6C4M3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>D6C4M3</td>\n",
       "      <td>B6V6L0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 2 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 63
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### noise",
   "id": "1a4a5d6acf552c25"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:10.466629Z",
     "start_time": "2025-03-05T09:54:10.463860Z"
    }
   },
   "cell_type": "code",
   "source": "print((cluster_df['Entry'].value_counts() != 1).sum())",
   "id": "d63aed4259f44e3f",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "534\n"
     ]
    }
   ],
   "execution_count": 64
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:11.088717Z",
     "start_time": "2025-03-05T09:54:11.080110Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cluster_df['Cluster'] = cluster_df['Entry'].rank(method='dense').astype(int) - 1\n",
    "cluster_df"
   ],
   "id": "617014e66017d591",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry cluster_member  Cluster\n",
       "0     A0A0N7CSQ4     A0A0N7CSQ4       15\n",
       "1     A0A1Z0YU59     A0A1Z0YU59       38\n",
       "2     A0A1Z0YU59         C0HLA5       38\n",
       "3     A0A1Z0YU59         C0HLA6       38\n",
       "4     A0A1Z0YU59         C0HLA7       38\n",
       "...          ...            ...      ...\n",
       "6423      D6C4M1         D2Y3T3      264\n",
       "6424      D6C4M1         D6C4M0      264\n",
       "6425      D6C4M1         D6C4M2      264\n",
       "6426      D6C4M3         D6C4M3      265\n",
       "6427      D6C4M3         B6V6L0      265\n",
       "\n",
       "[6428 rows x 3 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>cluster_member</th>\n",
       "      <th>Cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>C0HLA5</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>C0HLA6</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A1Z0YU59</td>\n",
       "      <td>C0HLA7</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>D6C4M1</td>\n",
       "      <td>D2Y3T3</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>D6C4M1</td>\n",
       "      <td>D6C4M0</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>D6C4M1</td>\n",
       "      <td>D6C4M2</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>D6C4M3</td>\n",
       "      <td>D6C4M3</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>D6C4M3</td>\n",
       "      <td>B6V6L0</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 3 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 65
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:12.366479Z",
     "start_time": "2025-03-05T09:54:12.357746Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df = df.merge(cluster_df[['cluster_member', 'Cluster']], left_on='Entry', right_on='cluster_member', how='left').drop(columns='cluster_member')\n",
    "df"
   ],
   "id": "2d40abcc6e1a0868",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                   Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family   \n",
       "2                            Insulin family   \n",
       "3                            Insulin family   \n",
       "4          Arthropod phospholipase D family   \n",
       "...                                     ...   \n",
       "6423                                    NaN   \n",
       "6424                                    NaN   \n",
       "6425                                    NaN   \n",
       "6426                                    NaN   \n",
       "6427                                    NaN   \n",
       "\n",
       "                                               Sequence  Fragment  Cluster  \n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment      702  \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN       83  \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN       11  \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN       11  \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment      200  \n",
       "...                                                 ...       ...      ...  \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN      496  \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN      497  \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN      499  \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment     1199  \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment     1262  \n",
       "\n",
       "[6428 rows x 7 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "      <td>702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1262</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 7 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 66
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:13.335632Z",
     "start_time": "2025-03-05T09:54:13.331423Z"
    }
   },
   "cell_type": "code",
   "source": "df[\"Protein families\"].value_counts()",
   "id": "65234b78f7a048a5",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Snake three-finger toxin family            532\n",
       "Long (4 C-C) scorpion toxin superfamily    418\n",
       "Phospholipase A2 family                    414\n",
       "Neurotoxin 10 (Hwtx-1) family              289\n",
       "Venom metalloproteinase (M12B) family      281\n",
       "                                          ... \n",
       "Teretoxin N (TN) superfamily                 1\n",
       "Teretoxin J (TJ) superfamily                 1\n",
       "Conotoxin B2 superfamily                     1\n",
       "Scutigerotoxin-01 family                     1\n",
       "SLPTX(15) family                             1\n",
       "Name: count, Length: 230, dtype: int64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 67
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:14.817084Z",
     "start_time": "2025-03-05T09:54:14.809079Z"
    }
   },
   "cell_type": "code",
   "source": [
    "rep_seq_ids = {record.id for record in SeqIO.parse(\"data/mmseqs_0.3/toxins_cluster_rep_seq.fasta\", \"fasta\")}\n",
    "df['Cluster_Rep'] = df['Entry'].isin(rep_seq_ids)\n",
    "df"
   ],
   "id": "d90235f0a5ac4ecd",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                   Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family   \n",
       "2                            Insulin family   \n",
       "3                            Insulin family   \n",
       "4          Arthropod phospholipase D family   \n",
       "...                                     ...   \n",
       "6423                                    NaN   \n",
       "6424                                    NaN   \n",
       "6425                                    NaN   \n",
       "6426                                    NaN   \n",
       "6427                                    NaN   \n",
       "\n",
       "                                               Sequence  Fragment  Cluster  \\\n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment      702   \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN       83   \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN       11   \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN       11   \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment      200   \n",
       "...                                                 ...       ...      ...   \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN      496   \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN      497   \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN      499   \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment     1199   \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment     1262   \n",
       "\n",
       "      Cluster_Rep  \n",
       "0           False  \n",
       "1           False  \n",
       "2           False  \n",
       "3           False  \n",
       "4           False  \n",
       "...           ...  \n",
       "6423         True  \n",
       "6424         True  \n",
       "6425         True  \n",
       "6426         True  \n",
       "6427         True  \n",
       "\n",
       "[6428 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "      <td>702</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>83</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "      <td>200</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>496</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>497</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>499</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1199</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1262</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 68
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:15.378888Z",
     "start_time": "2025-03-05T09:54:15.374275Z"
    }
   },
   "cell_type": "code",
   "source": "df[\"Protein families\"].value_counts()",
   "id": "1f4d2b4b40700da0",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Snake three-finger toxin family            532\n",
       "Long (4 C-C) scorpion toxin superfamily    418\n",
       "Phospholipase A2 family                    414\n",
       "Neurotoxin 10 (Hwtx-1) family              289\n",
       "Venom metalloproteinase (M12B) family      281\n",
       "                                          ... \n",
       "Teretoxin N (TN) superfamily                 1\n",
       "Teretoxin J (TJ) superfamily                 1\n",
       "Conotoxin B2 superfamily                     1\n",
       "Scutigerotoxin-01 family                     1\n",
       "SLPTX(15) family                             1\n",
       "Name: count, Length: 230, dtype: int64"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 69
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Train-Val-Test sets with 70:15:15 split",
   "id": "4965221a60e0c49"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:18.432380Z",
     "start_time": "2025-03-05T09:54:18.333402Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Get unique clusters and shuffle them\n",
    "clusters = df['Cluster'].unique()\n",
    "np.random.seed(42)\n",
    "np.random.shuffle(clusters)\n",
    "\n",
    "# Calculate total number of entries\n",
    "total_entries = len(df)\n",
    "\n",
    "# Define target sizes\n",
    "train_target = 0.7 * total_entries\n",
    "val_target = 0.15 * total_entries\n",
    "test_target = 0.15 * total_entries\n",
    "\n",
    "# Initialize cluster lists\n",
    "train_clusters, val_clusters, test_clusters = [], [], []\n",
    "\n",
    "# Initialize counters\n",
    "train_count, val_count, test_count = 0, 0, 0\n",
    "\n",
    "# Assign clusters while considering their sizes\n",
    "for cluster in clusters:\n",
    "    cluster_size = len(df[df['Cluster'] == cluster])\n",
    "\n",
    "    if train_count + cluster_size <= train_target:\n",
    "        train_clusters.append(cluster)\n",
    "        train_count += cluster_size\n",
    "    elif val_count + cluster_size <= val_target:\n",
    "        val_clusters.append(cluster)\n",
    "        val_count += cluster_size\n",
    "    else:\n",
    "        test_clusters.append(cluster)\n",
    "        test_count += cluster_size\n",
    "\n",
    "# Filter original df by cluster IDs\n",
    "train_df = df[df['Cluster'].isin(train_clusters)]\n",
    "val_df = df[df['Cluster'].isin(val_clusters)]\n",
    "test_df = df[df['Cluster'].isin(test_clusters)]\n",
    "\n",
    "# Shuffle each dataset\n",
    "train_df = shuffle(train_df, random_state=42).reset_index(drop=True)\n",
    "val_df = shuffle(val_df, random_state=42).reset_index(drop=True)\n",
    "test_df = shuffle(test_df, random_state=42).reset_index(drop=True)\n",
    "\n",
    "# Print results\n",
    "print(f\"Total entries: {total_entries}\")\n",
    "print(f\"Train entries: {len(train_df)} ({len(train_df)/total_entries:.2%})\")\n",
    "print(f\"Validation entries: {len(val_df)} ({len(val_df)/total_entries:.2%})\")\n",
    "print(f\"Test entries: {len(test_df)} ({len(test_df)/total_entries:.2%})\")\n",
    "\n",
    "# Output dataframes\n",
    "train_df, val_df, test_df"
   ],
   "id": "f6c7a3b9113fbb92",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total entries: 6428\n",
      "Train entries: 4499 (69.99%)\n",
      "Validation entries: 964 (15.00%)\n",
      "Test entries: 965 (15.01%)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(       Entry                                           Organism  \\\n",
       " 0     D2Y241  Cyriopagopus hainanus (Chinese bird spider) (H...   \n",
       " 1     P0DJC6                         Conus quercinus (Oak cone)   \n",
       " 2     B3EWT8       Cupiennius salei (American wandering spider)   \n",
       " 3     B3FIS1  Cyriopagopus schmidti (Chinese bird spider) (H...   \n",
       " 4     X2JCV5            Cerastes cerastes (Horned desert viper)   \n",
       " ...      ...                                                ...   \n",
       " 4494  Q9XZL0                 Conus textile (Cloth-of-gold cone)   \n",
       " 4495  Q9Y0X6  Olivierus martensii (Manchurian scorpion) (Mes...   \n",
       " 4496  B6DCR3  Lycosa singoriensis (Wolf spider) (Aranea sing...   \n",
       " 4497  P0DQA0  Scolopendra morsitans (Tanzanian blue ringleg ...   \n",
       " 4498  P0C837                     Conus striatus (Striated cone)   \n",
       " \n",
       "                                           Protein names  \\\n",
       " 0     U3-theraphotoxin-Hhn1a 2 (U3-TRTX-Hhn1a) (Hain...   \n",
       " 1                                       Conotoxin Qc6.1   \n",
       " 2                                  Cupiennin-2c (Cu-2c)   \n",
       " 3     U5-theraphotoxin-Hs1d (U5-TRTX-Hs1d) (Lectin S...   \n",
       " 4     L-amino acid oxidase (CC-LAAO) (LAO) (EC 1.4.3.2)   \n",
       " ...                                                 ...   \n",
       " 4494                          Omega-conotoxin-like TxO3   \n",
       " 4495  BmK-YA precursor [Cleaved into: BmK-YA 1 (Enke...   \n",
       " 4496  U3-lycotoxin-Ls1a (Toxin-like structure LSTX-B...   \n",
       " 4497          U-scoloptoxin(11)-Sm5a (U-SLPTX(11)-Sm5a)   \n",
       " 4498  Contryphan-S (St990) [Cleaved into: Contryphan...   \n",
       " \n",
       "                             Protein families  \\\n",
       " 0              Neurotoxin 10 (Hwtx-1) family   \n",
       " 1                   Conotoxin O1 superfamily   \n",
       " 2     Cationic peptide 04 (cupiennin) family   \n",
       " 3              Neurotoxin 10 (Hwtx-1) family   \n",
       " 4            Flavin monoamine oxidase family   \n",
       " ...                                      ...   \n",
       " 4494                Conotoxin O1 superfamily   \n",
       " 4495                                     NaN   \n",
       " 4496             Neurotoxin 19 (CSTX) family   \n",
       " 4497                   Scoloptoxin-11 family   \n",
       " 4498                Conotoxin 02 superfamily   \n",
       " \n",
       "                                                Sequence  Fragment  Cluster  \\\n",
       " 0     MVNMKASMFLTFAGLVLLFVVCYASESEEKEFPKEMLSSIFAVDDD...       NaN      983   \n",
       " 1     MKLTCMMIVALLFLTAWTFVTAVDSKNELENRGGWGQAGGWGKLFP...       NaN     1265   \n",
       " 2                   AFGTILKALAKIAGKVVKKLATKPGATYMLKENLQ       NaN      129   \n",
       " 3     MKTSMFLTLTGLVLLFVVCYASESEEKEFPKELLSSIFAADSDFKV...       NaN      983   \n",
       " 4     MNVFFMFSLLFLAALESCADDKNPLEEEFFEADYEEFLLIAKNGLQ...       NaN     1006   \n",
       " ...                                                 ...       ...      ...   \n",
       " 4494  MKLTCVVIVAVLFLTAWTFVTAVPHSSNALENLYLKAHHEMNNPEA...       NaN     1265   \n",
       " 4495  MIFHQFYSILILCLIFPNQVVQSDKERQDWIPSDYGGYMNPAGRSD...       NaN     1288   \n",
       " 4496  MKFVLLFGVFLVTLFSYSSAEMLDDFDQADEDELLSLIEKEEARAK...       NaN      152   \n",
       " 4497  RRVFTLTFLFLAAEAFHLSDRIETRESETGKGLATCKKDSICAYLQ...  fragment      729   \n",
       " 4498  MEKLTILVLVAAVLLSTQVMVQGDADQPADRDAVPRDDNAGGTDGK...       NaN       35   \n",
       " \n",
       "       Cluster_Rep  \n",
       " 0           False  \n",
       " 1           False  \n",
       " 2           False  \n",
       " 3           False  \n",
       " 4           False  \n",
       " ...           ...  \n",
       " 4494        False  \n",
       " 4495         True  \n",
       " 4496        False  \n",
       " 4497        False  \n",
       " 4498        False  \n",
       " \n",
       " [4499 rows x 8 columns],\n",
       "           Entry                                           Organism  \\\n",
       " 0        P0DN62  Terebra subulata (Chocolate spotted auger) (Bu...   \n",
       " 1        Q91126                          Naja atra (Chinese cobra)   \n",
       " 2        Q9U6Z5                   Conus imperialis (Imperial cone)   \n",
       " 3        Q9BPF5                 Conus tessulatus (Tessellate cone)   \n",
       " 4        A8HDJ6  Tropidechis carinatus (Australian rough-scaled...   \n",
       " ..          ...                                                ...   \n",
       " 959  A0A1L1QJU3  Hadronyche infensa (Fraser island funnel-web s...   \n",
       " 960  A0A3G3C7V7                         Conus amadis (Amadis cone)   \n",
       " 961      Q69CJ8      Ophiophagus hannah (King cobra) (Naja hannah)   \n",
       " 962      P19959                 Aipysurus laevis (Olive sea snake)   \n",
       " 963      Q9YI19  Gloydius brevicauda (Korean slamosa snake) (Ag...   \n",
       " \n",
       "                                          Protein names  \\\n",
       " 0                                    Teretoxin Tsu15.4   \n",
       " 1                                       Cardiotoxin 7a   \n",
       " 2                                      Conotoxin Im5.1   \n",
       " 3                                  Conotoxin TsMRCL-04   \n",
       " 4                          Short neurotoxin 2 (SNTX-2)   \n",
       " ..                                                 ...   \n",
       " 959  Pi-hexatoxin-Hi1a (Pi-HXTX-Hi1a) (Double-knot ...   \n",
       " 960                            Mu-conotoxin-like Am3.3   \n",
       " 961     Weak toxin DE-1 homolog 1 (WTX DE-1 homolog 1)   \n",
       " 962                                 Short neurotoxin B   \n",
       " 963  Zinc metalloproteinase/disintegrin [Cleaved in...   \n",
       " \n",
       "                           Protein families  \\\n",
       " 0                                      NaN   \n",
       " 1          Snake three-finger toxin family   \n",
       " 2                  Conotoxin T superfamily   \n",
       " 3                  Conotoxin T superfamily   \n",
       " 4          Snake three-finger toxin family   \n",
       " ..                                     ...   \n",
       " 959                   Psalmotoxin-1 family   \n",
       " 960                Conotoxin M superfamily   \n",
       " 961        Snake three-finger toxin family   \n",
       " 962        Snake three-finger toxin family   \n",
       " 963  Venom metalloproteinase (M12B) family   \n",
       " \n",
       "                                               Sequence Fragment  Cluster  \\\n",
       " 0    MTKLTVLLLAILVLLPLATSNSAADEALASLSGLLRRAKRCVPQNS...      NaN      667   \n",
       " 1    MKTLLLTLVVVTIVCLDLGYTLKCHNTQLPFIYKTCPEGKNLCFKA...      NaN      280   \n",
       " 2    MYCLPVFIILLLLISSAPSTPPQPRNKDRVHLISLLDNHKQILQRD...      NaN      290   \n",
       " 3    MRCLPVFIILLLLIPSAASAAQPETKDDAALASFYDNAKRTLQRHW...      NaN      290   \n",
       " 4    MKTLLLTLVVVTIVCLDLGYTMTCCNQQSSQPKTTTPCAESSCYKK...      NaN      280   \n",
       " ..                                                 ...      ...      ...   \n",
       " 959  NECIRKWLSCVDRKNDCCEGLECYKRRHSFEVCVPIPGFCLVKWKQ...      NaN       33   \n",
       " 960  MMSKLGVLLTICLLLFPLTAVPLDGDQPADRPAERMQDDISSENHP...      NaN      336   \n",
       " 961  MKPVLLTLVVVTIVCLDLGYTRICLKQEPFQPETTTTCPEGEDACY...      NaN      280   \n",
       " 962  MKTLLLTLVVVTIVCLDLGYTLTCCNQQSSQPKTTTDCADNSCYKM...      NaN      280   \n",
       " 963  MIQVLLVIICLADFPYQGTSIILESGNVNDYEVVYPRKVTALPKGA...      NaN      302   \n",
       " \n",
       "      Cluster_Rep  \n",
       " 0           True  \n",
       " 1          False  \n",
       " 2          False  \n",
       " 3          False  \n",
       " 4          False  \n",
       " ..           ...  \n",
       " 959         True  \n",
       " 960        False  \n",
       " 961        False  \n",
       " 962        False  \n",
       " 963        False  \n",
       " \n",
       " [964 rows x 8 columns],\n",
       "           Entry                                           Organism  \\\n",
       " 0        Q2I2Q4                   Conus litteratus (Lettered cone)   \n",
       " 1        P0DPV0  Scolopendra dehaani (Thai centipede) (Scolopen...   \n",
       " 2        P0DJM9  Crotalus basiliscus (Mexican west-coast rattle...   \n",
       " 3        P0CI45      Lychas mucronatus (Chinese swimming scorpion)   \n",
       " 4        Q9BH73                 Conus textile (Cloth-of-gold cone)   \n",
       " ..          ...                                                ...   \n",
       " 960      D2Y204  Cyriopagopus hainanus (Chinese bird spider) (H...   \n",
       " 961  A0A0B4J184          Terebra variegata (Variegate auger snail)   \n",
       " 962      P12028  Crotalus concolor (Midget faded rattlesnake) (...   \n",
       " 963      P0DJP9  Trimeresurus stejnegeri (Chinese green tree vi...   \n",
       " 964      C0HLA3               Lachesis muta rhombeata (Bushmaster)   \n",
       " \n",
       "                                          Protein names  \\\n",
       " 0                               Conotoxin Lt3.2 (Lt3b)   \n",
       " 1    Omega-scoloptoxin(10)-Ssd1b (Omega-SLPTX(10)-S...   \n",
       " 2    Basic phospholipase A2 CB1 (svPLA2) (EC 3.1.1....   \n",
       " 3    Lipolysis-activating peptide 1-alpha chain (LV...   \n",
       " 4    Conotoxin elongated-tx3a-a [Cleaved into: Cono...   \n",
       " ..                                                 ...   \n",
       " 960  U4-theraphotoxin-Hhn1a (U4-TRTX-Hhn1a) (Hainan...   \n",
       " 961                  Venom peptide Tv1 (Teretoxin v3a)   \n",
       " 962                            Myotoxin-1 (Myotoxin I)   \n",
       " 963  Acidic phospholipase A2 CTs-A2 (svPLA2) (EC 3....   \n",
       " 964  Snake venom serine protease LmrSP-4 (SVSP) (EC...   \n",
       " \n",
       "                             Protein families  \\\n",
       " 0                    Conotoxin M superfamily   \n",
       " 1                      Scoloptoxin-10 family   \n",
       " 2                    Phospholipase A2 family   \n",
       " 3    Long (3 C-C) scorpion toxin superfamily   \n",
       " 4                    Conotoxin M superfamily   \n",
       " ..                                       ...   \n",
       " 960            Neurotoxin 12 (Hwtx-2) family   \n",
       " 961                                      NaN   \n",
       " 962                Crotamine-myotoxin family   \n",
       " 963                  Phospholipase A2 family   \n",
       " 964                      Peptidase S1 family   \n",
       " \n",
       "                                               Sequence  Fragment  Cluster  \\\n",
       " 0    MLKIGVVLFTFLVLFPLATLQLDADQPVERYAENKQDLNPNERMKM...       NaN      262   \n",
       " 1    MNKLTIIFFTILLLTYIIVEKEALKIEDLPEPESYKKAKQLAVKDA...       NaN      707   \n",
       " 2                              HLLQFNKMIKFETRKNAIPFYAF  fragment      571   \n",
       " 3    MNIMLFCSVFILVSLTGLSVSDDVPGNYPMSLYGNKYSCGVLGENE...       NaN      268   \n",
       " 4    MLKMGVVLFIFLVLFPLATLQLDADQPVERYAENKQLLSPDERREI...       NaN      262   \n",
       " ..                                                 ...       ...      ...   \n",
       " 960  MKVTLIAILTCAAVLVLHTTAAEELEAESQLMEVGMPDTELAAVDE...       NaN      992   \n",
       " 961                              TRICCGCYWNGSKDVCSQSCC       NaN       10   \n",
       " 962        YKRCHKKEGHCFPKTVICLPPSSDFGKMDCRWKWKCCKKGSVN       NaN     1143   \n",
       " 963                            NLMQFELLIMKVAGRSGIVWYSD  fragment      571   \n",
       " 964  VFGGDECNINEHRSLVVLFDSDGFLCAGTLINKEWVLTAAHCDSEN...  fragment     1166   \n",
       " \n",
       "      Cluster_Rep  \n",
       " 0          False  \n",
       " 1           True  \n",
       " 2          False  \n",
       " 3          False  \n",
       " 4          False  \n",
       " ..           ...  \n",
       " 960        False  \n",
       " 961         True  \n",
       " 962        False  \n",
       " 963        False  \n",
       " 964        False  \n",
       " \n",
       " [965 rows x 8 columns])"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 70
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:21.350913Z",
     "start_time": "2025-03-05T09:54:21.268608Z"
    }
   },
   "cell_type": "code",
   "source": [
    "write_fasta(train_df, \"data/toxins_train.fasta\")\n",
    "write_fasta(val_df, \"data/toxins_val.fasta\")\n",
    "write_fasta(test_df, \"data/toxins_test.fasta\")"
   ],
   "id": "7ce4d9c75bcc450d",
   "outputs": [],
   "execution_count": 71
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-03T14:50:07.557258Z",
     "start_time": "2025-03-03T14:50:05.310783Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins_train.fasta data/mmseqs_0.5/train_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "85b55d3f72a81043",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create directory data/mmseqs_0.5/tmp\r\n",
      "easy-cluster data/toxins_train.fasta data/mmseqs_0.5/train_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_train.fasta data/mmseqs_0.5/tmp/15403511805714754484/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[4445] 0s 1ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 4499 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 4.50K 0s 14ms     \r\n",
      "Sort kmer 0h 0m 0s 1ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 24ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 4.50K 0s 8ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms=================> ] 98.04% 4.41K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "clust data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_rescore1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 4.50K 0s 5ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 7963 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 4.50K 0s 6ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 4.50K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 14ms\r\n",
      "Total time: 0h 0m 0s 15ms\r\n",
      "\r\n",
      "Size of the sequence database: 4499\r\n",
      "Size of the alignment database: 4499\r\n",
      "Number of clusters: 1806\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 17ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/order_redundancy data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/order_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_filter1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_filter2 --filter-file data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 1.81K 0s 4ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_filter2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.81K 0s 4ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 1ms==================>] 99.61% 1.80K eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "align data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_rescore2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1806 type: Aminoacid\r\n",
      "Target database size: 1806 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.81K 0s 23ms     \r\n",
      "Time for merging to aln: 0h 0m 0s 1ms\r\n",
      "2167 alignments calculated\r\n",
      "2066 sequence pairs passed the thresholds (0.953392 of overall calculated)\r\n",
      "1.143965 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 28ms\r\n",
      "clust data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/aln data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.81K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 260 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.81K 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.81K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 9ms\r\n",
      "Total time: 0h 0m 0s 10ms\r\n",
      "\r\n",
      "Size of the sequence database: 1806\r\n",
      "Size of the alignment database: 1806\r\n",
      "Number of clusters: 1618\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pre_clust data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 1.81K 0s 4ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 1.62K 0s 7ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 4.50K 0s 12ms     \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 15ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/linclust/13147724460293524184/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_redundancy data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1618 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1618 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.62K 0s 7ms      \r\n",
      "Index table: Masked residues: 2821\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.62K 0s 7ms      \r\n",
      "Index statistics\r\n",
      "Entries:          65976\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.001031\r\n",
      "Top 10 k-mers\r\n",
      "    GGENEH\t22\r\n",
      "    GNCCLP\t19\r\n",
      "    KCNKCY\t18\r\n",
      "    RCFHCY\t14\r\n",
      "    CTNAGY\t12\r\n",
      "    NPCACK\t10\r\n",
      "    INHFAL\t10\r\n",
      "    YCWGGN\t10\r\n",
      "    CDAAFR\t10\r\n",
      "    CYKWHR\t10\r\n",
      "Time for index table init: 0h 0m 0s 106ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1618\r\n",
      "Target db start 1 to 1618\r\n",
      "[=================================================================] 100.00% 1.62K 0s 11ms     \r\n",
      "[===============================================================> ] 97.46% 1.58K eta 0s       \r\n",
      "7.110539 k-mers per position\r\n",
      "72 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "5 sequences passed prefiltering per query sequence\r\n",
      "3 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 302ms\r\n",
      "align data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step0 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1618 type: Aminoacid\r\n",
      "Target database size: 1618 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.62K 0s 34ms     \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\r\n",
      "6068 alignments calculated\r\n",
      "4016 sequence pairs passed the thresholds (0.661833 of overall calculated)\r\n",
      "2.482077 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 39ms\r\n",
      "clust data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step0 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.62K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 112 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.62K 0s 4ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.62K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 9ms\r\n",
      "Total time: 0h 0m 0s 10ms\r\n",
      "\r\n",
      "Size of the sequence database: 1618\r\n",
      "Size of the alignment database: 1618\r\n",
      "Number of clusters: 1156\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step0 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1156 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1156 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.16K 0s 6ms      \r\n",
      "Index table: Masked residues: 2403\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.16K 0s 6ms      \r\n",
      "Index statistics\r\n",
      "Entries:          54646\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000854\r\n",
      "Top 10 k-mers\r\n",
      "    VALLCQ\t13\r\n",
      "    DDNLCF\t7\r\n",
      "    CVIAIL\t7\r\n",
      "    KCNKCY\t7\r\n",
      "    REDELE\t5\r\n",
      "    CFEDEF\t5\r\n",
      "    TCLVLF\t5\r\n",
      "    LEGECG\t5\r\n",
      "    CFLLIG\t5\r\n",
      "    EELING\t5\r\n",
      "Time for index table init: 0h 0m 0s 109ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1156\r\n",
      "Target db start 1 to 1156\r\n",
      "[=================================================================] 100.00% 1.16K 0s 8ms      \r\n",
      "\r\n",
      "12.492633 k-mers per position\r\n",
      "54 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 293ms\r\n",
      "align data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1156 type: Aminoacid\r\n",
      "Target database size: 1156 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.16K 0s 13ms     \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "1781 alignments calculated\r\n",
      "1206 sequence pairs passed the thresholds (0.677148 of overall calculated)\r\n",
      "1.043253 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 17ms\r\n",
      "clust data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.16K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 6 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.16K 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.16K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 9ms\r\n",
      "\r\n",
      "Size of the sequence database: 1156\r\n",
      "Size of the alignment database: 1156\r\n",
      "Number of clusters: 1128\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 1128 type: Aminoacid\r\n",
      "Estimated memory consumption: 979M\r\n",
      "Target database size: 1128 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 1.13K 0s 6ms      \r\n",
      "Index table: Masked residues: 2374\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 1.13K 0s 5ms      \r\n",
      "Index statistics\r\n",
      "Entries:          65041\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.001016\r\n",
      "Top 10 k-mers\r\n",
      "    MKTVVA\t15\r\n",
      "    VALLCQ\t12\r\n",
      "    KLCLAV\t11\r\n",
      "    KCNKCY\t7\r\n",
      "    AVFTQL\t6\r\n",
      "    YGCGNN\t6\r\n",
      "    REDELE\t5\r\n",
      "    CFEDEF\t5\r\n",
      "    LEGECG\t5\r\n",
      "    CFLLIG\t5\r\n",
      "Time for index table init: 0h 0m 0s 107ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 1128\r\n",
      "Target db start 1 to 1128\r\n",
      "[=================================================================] 100.00% 1.13K 0s 11ms     2.81% 1.05K eta 0s       \r\n",
      "[================================================================>] 99.73% 1.12K eta 0s       \r\n",
      "33.132925 k-mers per position\r\n",
      "71 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "3 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 290ms\r\n",
      "align data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 1128 type: Aminoacid\r\n",
      "Target database size: 1128 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 1.13K 0s 7ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 1ms\r\n",
      "2022 alignments calculated\r\n",
      "1173 sequence pairs passed the thresholds (0.580119 of overall calculated)\r\n",
      "1.039894 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 16ms\r\n",
      "clust data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step2 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 1.13K 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 17 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 1.13K 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 1.13K 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 1128\r\n",
      "Size of the alignment database: 1128\r\n",
      "Number of clusters: 1098\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_redundancy data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step0 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step1 data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 1.62K 0s 3ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 1.16K 0s 7ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 1.13K 0s 9ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 1.10K 0s 12ms     \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 4.50K 0s 17ms     \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 20ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_tmp/608004905528923415/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu data/mmseqs_0.5/tmp/15403511805714754484/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu data/mmseqs_0.5/tmp/15403511805714754484/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.10K 0s 3ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 2ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_rep data/mmseqs_0.5/tmp/15403511805714754484/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu data/mmseqs_0.5/tmp/15403511805714754484/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 1.10K 0s 4ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/input data/mmseqs_0.5/tmp/15403511805714754484/clu_seqs data/mmseqs_0.5/tmp/15403511805714754484/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/15403511805714754484/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 127
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-03T14:50:13.099348Z",
     "start_time": "2025-03-03T14:50:11.290177Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins_val.fasta data/mmseqs_0.5/val_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "4e88e675eaece40c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_val.fasta data/mmseqs_0.5/val_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_val.fasta data/mmseqs_0.5/tmp/752296466822456242/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[910] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/752296466822456242/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu data/mmseqs_0.5/tmp/752296466822456242/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 964 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 964 0s 6ms      \r\n",
      "Sort kmer 0h 0m 0s 0ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 12ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 964 0s 4ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms==================>] 98.75% 952 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "clust data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_rescore1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 964 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1562 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 964 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 964 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 964\r\n",
      "Size of the alignment database: 964\r\n",
      "Number of clusters: 435\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/order_redundancy data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/order_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_filter1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_filter2 --filter-file data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 435 0s 2ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_filter2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 435 0s 2ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms=============>     ] 92.17% 401 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "align data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_rescore2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 435 type: Aminoacid\r\n",
      "Target database size: 435 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 435 0s 11ms     \r\n",
      "Time for merging to aln: 0h 0m 0s 0ms\r\n",
      "556 alignments calculated\r\n",
      "525 sequence pairs passed the thresholds (0.944245 of overall calculated)\r\n",
      "1.206897 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 17ms\r\n",
      "clust data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/aln data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 435 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 90 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 435 0s 2ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 435 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 6ms\r\n",
      "\r\n",
      "Size of the sequence database: 435\r\n",
      "Size of the alignment database: 435\r\n",
      "Number of clusters: 377\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pre_clust data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 435 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 377 0s 3ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 964 0s 7ms      \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/linclust/18445507043006036693/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_redundancy data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 377 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 377 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 377 0s 2ms      \r\n",
      "Index table: Masked residues: 899\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 377 0s 3ms      =====================>                                     ] 42.02% 159 eta 0s       \r\n",
      "Index statistics\r\n",
      "Entries:          13327\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000208\r\n",
      "Top 10 k-mers\r\n",
      "    TICDYT\t27\r\n",
      "    RCPFLL\t10\r\n",
      "    LDGTCY\t8\r\n",
      "    LDGTCH\t7\r\n",
      "    GCGKRI\t7\r\n",
      "    VCDGLK\t7\r\n",
      "    LTCLPL\t7\r\n",
      "    CSPCNN\t7\r\n",
      "    TAPDQP\t6\r\n",
      "    GYYGPI\t5\r\n",
      "Time for index table init: 0h 0m 0s 96ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 377\r\n",
      "Target db start 1 to 377\r\n",
      "[=================================================================] 100.00% 377 0s 6ms      ======================>              ] 78.19% 295 eta 0s       \r\n",
      "[===============================================================> ] 97.87% 369 eta 0s       \r\n",
      "6.901844 k-mers per position\r\n",
      "51 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "4 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 274ms\r\n",
      "align data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step0 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 377 type: Aminoacid\r\n",
      "Target database size: 377 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 377 0s 5ms      \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\r\n",
      "1482 alignments calculated\r\n",
      "936 sequence pairs passed the thresholds (0.631579 of overall calculated)\r\n",
      "2.482759 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "clust data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step0 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 377 0s 2ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 47 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 377 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 377 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 6ms\r\n",
      "Total time: 0h 0m 0s 6ms\r\n",
      "\r\n",
      "Size of the sequence database: 377\r\n",
      "Size of the alignment database: 377\r\n",
      "Number of clusters: 271\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step0 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 271 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 271 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 271 0s 2ms      \r\n",
      "Index table: Masked residues: 807\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 271 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          10127\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000158\r\n",
      "Top 10 k-mers\r\n",
      "    TLVTCL\t11\r\n",
      "    TICDYT\t8\r\n",
      "    CLLYIC\t3\r\n",
      "    CAGCQC\t3\r\n",
      "    ELEGCD\t3\r\n",
      "    CFLLIG\t3\r\n",
      "    MAEGLG\t3\r\n",
      "    DRGMLG\t3\r\n",
      "    KLRGSG\t3\r\n",
      "    LDGTCH\t3\r\n",
      "Time for index table init: 0h 0m 0s 101ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 271\r\n",
      "Target db start 1 to 271\r\n",
      "[=================================================================] 100.00% 271 0s 5ms      \r\n",
      "[=========================================================>       ] 88.15% 239 eta 0s       \r\n",
      "12.040031 k-mers per position\r\n",
      "36 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 284ms\r\n",
      "align data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 271 type: Aminoacid\r\n",
      "Target database size: 271 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 271 0s 2ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "400 alignments calculated\r\n",
      "280 sequence pairs passed the thresholds (0.700000 of overall calculated)\r\n",
      "1.033210 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "clust data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 271 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 3 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 271 0s 2ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 271 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 271\r\n",
      "Size of the alignment database: 271\r\n",
      "Number of clusters: 265\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 265 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 265 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 265 0s 2ms      \r\n",
      "Index table: Masked residues: 807\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 265 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          11721\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000183\r\n",
      "Top 10 k-mers\r\n",
      "    VVICLG\t9\r\n",
      "    TLVTCL\t9\r\n",
      "    MKLLVV\t8\r\n",
      "    VTVLGY\t4\r\n",
      "    CFLLIG\t3\r\n",
      "    MAEGLG\t3\r\n",
      "    DRGMLG\t3\r\n",
      "    KLRGSG\t3\r\n",
      "    NPCACK\t3\r\n",
      "    CLVILL\t3\r\n",
      "Time for index table init: 0h 0m 0s 95ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 265\r\n",
      "Target db start 1 to 265\r\n",
      "[=================================================================] 100.00% 265 0s 5ms      \r\n",
      "[================================================================>] 99.24% 263 eta 0s       \r\n",
      "31.123930 k-mers per position\r\n",
      "44 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 273ms\r\n",
      "align data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 265 type: Aminoacid\r\n",
      "Target database size: 265 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 265 0s 3ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "383 alignments calculated\r\n",
      "276 sequence pairs passed the thresholds (0.720627 of overall calculated)\r\n",
      "1.041509 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "clust data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step2 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 265 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 265 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 265 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 265\r\n",
      "Size of the alignment database: 265\r\n",
      "Number of clusters: 259\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_redundancy data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step0 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step1 data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 377 0s 2ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 271 0s 3ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 265 0s 5ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 259 0s 7ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 964 0s 10ms     \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 13ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_tmp/5282437003029201426/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu data/mmseqs_0.5/tmp/752296466822456242/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu data/mmseqs_0.5/tmp/752296466822456242/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 259 0s 1ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_rep data/mmseqs_0.5/tmp/752296466822456242/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu data/mmseqs_0.5/tmp/752296466822456242/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 259 0s 1ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/input data/mmseqs_0.5/tmp/752296466822456242/clu_seqs data/mmseqs_0.5/tmp/752296466822456242/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/752296466822456242/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 128
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-03T14:50:15.346168Z",
     "start_time": "2025-03-03T14:50:13.528797Z"
    }
   },
   "cell_type": "code",
   "source": "!mmseqs easy-cluster data/toxins_test.fasta data/mmseqs_0.5/test_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "37b6452492119dab",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_test.fasta data/mmseqs_0.5/test_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_test.fasta data/mmseqs_0.5/tmp/9239027741616835087/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[910] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 2ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 965 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 965 0s 5ms      \r\n",
      "Sort kmer 0h 0m 0s 0ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 965 0s 3ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms================>  ] 96.68% 933 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_rescore1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 965 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1236 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 965 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 965 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 965\r\n",
      "Size of the alignment database: 965\r\n",
      "Number of clusters: 444\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/order_redundancy data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/order_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_filter1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_filter2 --filter-file data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 444 0s 3ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_filter2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 444 0s 2ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms==================>] 99.77% 443 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "align data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_rescore2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 444 type: Aminoacid\r\n",
      "Target database size: 444 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 444 0s 16ms     \r\n",
      "Time for merging to aln: 0h 0m 0s 0ms\r\n",
      "523 alignments calculated\r\n",
      "512 sequence pairs passed the thresholds (0.978967 of overall calculated)\r\n",
      "1.153153 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 20ms\r\n",
      "clust data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/aln data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 444 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 68 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 444 0s 2ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 444 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 6ms\r\n",
      "Total time: 0h 0m 0s 6ms\r\n",
      "\r\n",
      "Size of the sequence database: 444\r\n",
      "Size of the alignment database: 444\r\n",
      "Number of clusters: 398\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pre_clust data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 444 0s 2ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 398 0s 4ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 965 0s 7ms      \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/linclust/10678250117009581158/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_redundancy data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 398 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 398 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 398 0s 2ms      \r\n",
      "Index table: Masked residues: 437\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 398 0s 2ms      \r\n",
      "Index statistics\r\n",
      "Entries:          15996\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000250\r\n",
      "Top 10 k-mers\r\n",
      "    YYGIDA\t6\r\n",
      "    TTPVHC\t6\r\n",
      "    MIVLIC\t6\r\n",
      "    KYNEDE\t6\r\n",
      "    EPVHKN\t6\r\n",
      "    YSDRTT\t6\r\n",
      "    SANLHF\t5\r\n",
      "    LHENLF\t5\r\n",
      "    ICAFQG\t5\r\n",
      "    ISCGGH\t5\r\n",
      "Time for index table init: 0h 0m 0s 97ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 398\r\n",
      "Target db start 1 to 398\r\n",
      "[=================================================================] 100.00% 398 0s 6ms      \r\n",
      "[================================================================>] 99.24% 395 eta 0s       \r\n",
      "7.253466 k-mers per position\r\n",
      "50 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "2 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 281ms\r\n",
      "align data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step0 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 398 type: Aminoacid\r\n",
      "Target database size: 398 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 398 0s 5ms      \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\r\n",
      "844 alignments calculated\r\n",
      "662 sequence pairs passed the thresholds (0.784360 of overall calculated)\r\n",
      "1.663317 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "clust data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step0 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 398 0s 2ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 2 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 398 0s 2ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 398 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 6ms\r\n",
      "Total time: 0h 0m 0s 6ms\r\n",
      "\r\n",
      "Size of the sequence database: 398\r\n",
      "Size of the alignment database: 398\r\n",
      "Number of clusters: 307\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step0 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 307 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 307 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 307 0s 1ms      \r\n",
      "Index table: Masked residues: 395\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 307 0s 2ms       44.12% 136 eta 0s       \r\n",
      "Index statistics\r\n",
      "Entries:          15573\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000243\r\n",
      "Top 10 k-mers\r\n",
      "    YYGIDA\t4\r\n",
      "    GGENEH\t4\r\n",
      "    IQLVCL\t4\r\n",
      "    SFERDL\t4\r\n",
      "    EPVHKN\t4\r\n",
      "    LVILFP\t4\r\n",
      "    CYHRND\t3\r\n",
      "    KYNEDE\t3\r\n",
      "    LKHKGE\t3\r\n",
      "    SANLHF\t3\r\n",
      "Time for index table init: 0h 0m 0s 97ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 307\r\n",
      "Target db start 1 to 307\r\n",
      "[=================================================================] 100.00% 307 0s 5ms      =============================================>              ] 77.12% 237 eta 0s       \r\n",
      "[===============================================================> ] 97.06% 298 eta 0s       \r\n",
      "11.925174 k-mers per position\r\n",
      "51 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 282ms\r\n",
      "align data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 307 type: Aminoacid\r\n",
      "Target database size: 307 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 307 0s 3ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "402 alignments calculated\r\n",
      "315 sequence pairs passed the thresholds (0.783582 of overall calculated)\r\n",
      "1.026059 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 307 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 2 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 307 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 307 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 4ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 307\r\n",
      "Size of the alignment database: 307\r\n",
      "Number of clusters: 302\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 302 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 302 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 302 0s 1ms      \r\n",
      "Index table: Masked residues: 395\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 302 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          18454\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000288\r\n",
      "Top 10 k-mers\r\n",
      "    YYGIDA\t4\r\n",
      "    SIANKG\t4\r\n",
      "    STSSNG\t4\r\n",
      "    GEVLEK\t4\r\n",
      "    ASSCLK\t4\r\n",
      "    PVLLNK\t4\r\n",
      "    IQLVCL\t4\r\n",
      "    SFERDL\t4\r\n",
      "    TAIAGL\t4\r\n",
      "    EPVHKN\t4\r\n",
      "Time for index table init: 0h 0m 0s 94ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 302\r\n",
      "Target db start 1 to 302\r\n",
      "[=================================================================] 100.00% 302 0s 6ms      \r\n",
      "[================================================================>] 99.67% 301 eta 0s       \r\n",
      "32.097107 k-mers per position\r\n",
      "64 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 283ms\r\n",
      "align data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 302 type: Aminoacid\r\n",
      "Target database size: 302 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 302 0s 3ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "447 alignments calculated\r\n",
      "313 sequence pairs passed the thresholds (0.700224 of overall calculated)\r\n",
      "1.036424 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "clust data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step2 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 302 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 3 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 302 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 302 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 4ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 302\r\n",
      "Size of the alignment database: 302\r\n",
      "Number of clusters: 295\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_redundancy data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step0 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step1 data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 398 0s 2ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 307 0s 4ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 302 0s 5ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 295 0s 7ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 965 0s 11ms     \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 14ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_tmp/3764798205673116255/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu data/mmseqs_0.5/tmp/9239027741616835087/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu data/mmseqs_0.5/tmp/9239027741616835087/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 295 0s 1ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_rep data/mmseqs_0.5/tmp/9239027741616835087/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu data/mmseqs_0.5/tmp/9239027741616835087/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 295 0s 2ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/input data/mmseqs_0.5/tmp/9239027741616835087/clu_seqs data/mmseqs_0.5/tmp/9239027741616835087/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/9239027741616835087/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 129
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:27.358408Z",
     "start_time": "2025-03-05T09:54:27.347620Z"
    }
   },
   "cell_type": "code",
   "source": [
    "rep_seq_files = {\n",
    "    \"Train_Cluster_Rep\": \"data/mmseqs_0.5/train_cluster_rep_seq.fasta\",\n",
    "    \"Val_Cluster_Rep\": \"data/mmseqs_0.5/val_cluster_rep_seq.fasta\",\n",
    "    \"Test_Cluster_Rep\": \"data/mmseqs_0.5/test_cluster_rep_seq.fasta\",\n",
    "}\n",
    "\n",
    "for col, file in rep_seq_files.items():\n",
    "    rep_seq_ids = {record.id for record in SeqIO.parse(file, \"fasta\")}\n",
    "    df[col] = df['Entry'].isin(rep_seq_ids)\n",
    "\n",
    "df"
   ],
   "id": "3b891d1bb61c3473",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                   Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family   \n",
       "2                            Insulin family   \n",
       "3                            Insulin family   \n",
       "4          Arthropod phospholipase D family   \n",
       "...                                     ...   \n",
       "6423                                    NaN   \n",
       "6424                                    NaN   \n",
       "6425                                    NaN   \n",
       "6426                                    NaN   \n",
       "6427                                    NaN   \n",
       "\n",
       "                                               Sequence  Fragment  Cluster  \\\n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment      702   \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN       83   \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN       11   \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN       11   \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment      200   \n",
       "...                                                 ...       ...      ...   \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN      496   \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN      497   \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN      499   \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment     1199   \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment     1262   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "0           False              False            False             False  \n",
       "1           False              False            False             False  \n",
       "2           False              False            False             False  \n",
       "3           False              False            False             False  \n",
       "4           False              False            False             False  \n",
       "...           ...                ...              ...               ...  \n",
       "6423         True               True            False             False  \n",
       "6424         True               True            False             False  \n",
       "6425         True               True            False             False  \n",
       "6426         True              False             True             False  \n",
       "6427         True               True            False             False  \n",
       "\n",
       "[6428 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "      <td>702</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>83</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "      <td>200</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>496</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>497</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>499</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1199</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1262</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 72
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:28.949034Z",
     "start_time": "2025-03-05T09:54:28.945638Z"
    }
   },
   "cell_type": "code",
   "source": [
    "all_50_pct_seq = (\n",
    "    df[\"Train_Cluster_Rep\"].sum().item() +\n",
    "    df[\"Val_Cluster_Rep\"].sum().item() +\n",
    "    df[\"Test_Cluster_Rep\"].sum().item()\n",
    ")\n",
    "print(\"All representative sequences after 50% redundancy reduction:\", all_50_pct_seq)\n",
    "print(f\"Train entries: {df[\"Train_Cluster_Rep\"].sum().item()} ({df[\"Train_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")\n",
    "\n",
    "print(f\"Validation entries: {df[\"Val_Cluster_Rep\"].sum().item()} ({df[\"Val_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")\n",
    "print(f\"Test entries: {df[\"Test_Cluster_Rep\"].sum().item()} ({df[\"Test_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")"
   ],
   "id": "5c3f48f7dcfebde8",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All representative sequences after 50% redundancy reduction: 1652\n",
      "Train entries: 1098 (66.46%)\n",
      "Validation entries: 259 (15.68%)\n",
      "Test entries: 295 (17.86%)\n"
     ]
    }
   ],
   "execution_count": 73
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:29.916650Z",
     "start_time": "2025-03-05T09:54:29.909615Z"
    }
   },
   "cell_type": "code",
   "source": [
    "train_val_test_data = df[df[[\"Train_Cluster_Rep\", \"Val_Cluster_Rep\", \"Test_Cluster_Rep\"]].any(axis=1)]\n",
    "train_val_test_data"
   ],
   "id": "fb55be7b5bb58242",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "6     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "9     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "18    A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "19        A0S864               Boiga irregularis (Brown tree snake)   \n",
       "33        A8YPR6       Echis ocellatus (Ocellated saw-scaled viper)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "6     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "9     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "18    Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "19                          Irditoxin subunit A (IrTxA)   \n",
       "33    Snake venom metalloprotease inhibitor 02D01 (0...   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                           Protein families  \\\n",
       "6                     Scoloptoxin-04 family   \n",
       "9          Arthropod phospholipase D family   \n",
       "18    Venom metalloproteinase (M12B) family   \n",
       "19          Snake three-finger toxin family   \n",
       "33               Natriuretic peptide family   \n",
       "...                                     ...   \n",
       "6423                                    NaN   \n",
       "6424                                    NaN   \n",
       "6425                                    NaN   \n",
       "6426                                    NaN   \n",
       "6427                                    NaN   \n",
       "\n",
       "                                               Sequence  Fragment  Cluster  \\\n",
       "6     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...       NaN       15   \n",
       "9     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...       NaN      200   \n",
       "18    MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...       NaN       55   \n",
       "19    MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...       NaN       44   \n",
       "33    MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...       NaN       88   \n",
       "...                                                 ...       ...      ...   \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN      496   \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN      497   \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN      499   \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment     1199   \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment     1262   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "6            True              False             True             False  \n",
       "9           False               True            False             False  \n",
       "18           True              False            False              True  \n",
       "19          False               True            False             False  \n",
       "33          False              False             True             False  \n",
       "...           ...                ...              ...               ...  \n",
       "6423         True               True            False             False  \n",
       "6424         True               True            False             False  \n",
       "6425         True               True            False             False  \n",
       "6426         True              False             True             False  \n",
       "6427         True               True            False             False  \n",
       "\n",
       "[1652 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin-04 family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>A0S864</td>\n",
       "      <td>Boiga irregularis (Brown tree snake)</td>\n",
       "      <td>Irditoxin subunit A (IrTxA)</td>\n",
       "      <td>Snake three-finger toxin family</td>\n",
       "      <td>MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>A8YPR6</td>\n",
       "      <td>Echis ocellatus (Ocellated saw-scaled viper)</td>\n",
       "      <td>Snake venom metalloprotease inhibitor 02D01 (0...</td>\n",
       "      <td>Natriuretic peptide family</td>\n",
       "      <td>MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>496</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>497</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>499</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1199</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "      <td>1262</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1652 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 74
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:32.284095Z",
     "start_time": "2025-03-05T09:54:32.280480Z"
    }
   },
   "cell_type": "code",
   "source": [
    "no_na = train_val_test_data.dropna(subset=[\"Protein families\"]).copy()\n",
    "\n",
    "print(\"All representative sequences after 50% redundancy reduction:\", len(no_na))\n",
    "print(f\"Train entries: {no_na['Train_Cluster_Rep'].sum()} ({no_na['Train_Cluster_Rep'].sum()/len(no_na):.2%})\")\n",
    "print(f\"Validation entries: {no_na['Val_Cluster_Rep'].sum()} ({no_na['Val_Cluster_Rep'].sum()/len(no_na):.2%})\")\n",
    "print(f\"Test entries: {no_na['Test_Cluster_Rep'].sum()} ({no_na['Test_Cluster_Rep'].sum()/len(no_na):.2%})\")"
   ],
   "id": "bb801831b12fcb5c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All representative sequences after 50% redundancy reduction: 1392\n",
      "Train entries: 940 (67.53%)\n",
      "Validation entries: 210 (15.09%)\n",
      "Test entries: 242 (17.39%)\n"
     ]
    }
   ],
   "execution_count": 75
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:35.640850Z",
     "start_time": "2025-03-05T09:54:35.637088Z"
    }
   },
   "cell_type": "code",
   "source": "no_na[\"Protein families\"].value_counts()",
   "id": "78840daa8b5029e0",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Short scorpion toxin superfamily           81\n",
       "Snake three-finger toxin family            64\n",
       "Venom metalloproteinase (M12B) family      63\n",
       "Long (4 C-C) scorpion toxin superfamily    61\n",
       "Conotoxin O1 superfamily                   48\n",
       "                                           ..\n",
       "Conotoxin U superfamily                     1\n",
       "Neurotoxin 36 family                        1\n",
       "Conotoxin V superfamily                     1\n",
       "Conotoxin K superfamily                     1\n",
       "Scolopendra toxin 2 family                  1\n",
       "Name: count, Length: 224, dtype: int64"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 76
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:35.925350Z",
     "start_time": "2025-03-05T09:54:35.921253Z"
    }
   },
   "cell_type": "code",
   "source": [
    "print(no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() <= 10].sum())\n",
    "no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() <= 10]"
   ],
   "id": "7f3c8089b0715c9d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "445\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Scoloptoxin-11 family                      10\n",
       "Vasopressin/oxytocin family                 8\n",
       "Scoloptoxin-16 family                       8\n",
       "FARP (FMRFamide related peptide) family     8\n",
       "Cationic peptide 01 (latrotoxin) family     7\n",
       "                                           ..\n",
       "Conotoxin U superfamily                     1\n",
       "Neurotoxin 36 family                        1\n",
       "Conotoxin V superfamily                     1\n",
       "Conotoxin K superfamily                     1\n",
       "Scolopendra toxin 2 family                  1\n",
       "Name: count, Length: 191, dtype: int64"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 77
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:54:36.904138Z",
     "start_time": "2025-03-05T09:54:36.899250Z"
    }
   },
   "cell_type": "code",
   "source": [
    "print(no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() > 10].sum())\n",
    "no_na[\"Protein families\"].value_counts()[no_na[\"Protein families\"].value_counts() > 10]"
   ],
   "id": "3623adb8c4ef986d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "947\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Short scorpion toxin superfamily                    81\n",
       "Snake three-finger toxin family                     64\n",
       "Venom metalloproteinase (M12B) family               63\n",
       "Long (4 C-C) scorpion toxin superfamily             61\n",
       "Conotoxin O1 superfamily                            48\n",
       "Conotoxin M superfamily                             44\n",
       "Neurotoxin 10 (Hwtx-1) family                       43\n",
       "Conotoxin A superfamily                             42\n",
       "MCD family                                          42\n",
       "Phospholipase A2 family                             40\n",
       "Bradykinin-potentiating peptide family              38\n",
       "Conotoxin T superfamily                             28\n",
       "Neurotoxin 02 (plectoxin) family                    26\n",
       "Snaclec family                                      24\n",
       "Venom Kunitz-type family                            23\n",
       "Non-disulfide-bridged peptide (NDBP) superfamily    20\n",
       "Peptidase S1 family                                 20\n",
       "CRISP family                                        19\n",
       "Cationic peptide 04 (cupiennin) family              19\n",
       "Long (3 C-C) scorpion toxin superfamily             18\n",
       "Conotoxin 02 superfamily                            18\n",
       "Neurotoxin 03 (Tx2) family                          17\n",
       "Flavin monoamine oxidase family                     16\n",
       "Neurotoxin 14 (magi-1) family                       16\n",
       "Neurotoxin 04 (omega-agtx) family                   16\n",
       "Natriuretic peptide family                          15\n",
       "Conotoxin I1 superfamily                            14\n",
       "Arthropod phospholipase D family                    14\n",
       "Long chain scorpion toxin family                    12\n",
       "Formicidae venom precursor-01 superfamily           12\n",
       "Bradykinin-related peptide family                   12\n",
       "Conotoxin O2 superfamily                            11\n",
       "Neurotoxin 19 (CSTX) family                         11\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 78
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### interesting scoloptoxins are still in the data, nice",
   "id": "e0017834632f7562"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T10:03:59.216709Z",
     "start_time": "2025-03-05T10:03:59.212157Z"
    }
   },
   "cell_type": "code",
   "source": "no_na[no_na[\"Entry\"].isin([\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"])]",
   "id": "566e17256a3a9d55",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Entry                                           Organism  \\\n",
       "2440  I6R1R5  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "5339  P0DPX7  Scolopendra morsitans (Tanzanian blue ringleg ...   \n",
       "5340  P0DPX8            Ethmostigmus rubripes (Giant centipede)   \n",
       "5341  P0DPX9            Ethmostigmus rubripes (Giant centipede)   \n",
       "5342  P0DPY0            Ethmostigmus rubripes (Giant centipede)   \n",
       "5343  P0DPY1  Cormocephalus westwoodi (Westwood's green cent...   \n",
       "\n",
       "                                          Protein names Protein families  \\\n",
       "2440  Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...            other   \n",
       "5339          U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)            other   \n",
       "5340          U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)            other   \n",
       "5341          U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)            other   \n",
       "5342          U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)            other   \n",
       "5343          U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)            other   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "2440  MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...      NaN      720   \n",
       "5339  MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...      NaN      720   \n",
       "5340  MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...      NaN      721   \n",
       "5341  MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...      NaN      720   \n",
       "5342  MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...      NaN      720   \n",
       "5343  MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...      NaN      720   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "2440        False              False            False              True  \n",
       "5339         True              False            False              True  \n",
       "5340         True               True            False             False  \n",
       "5341        False              False            False              True  \n",
       "5342        False              False            False              True  \n",
       "5343        False              False            False              True  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2440</th>\n",
       "      <td>I6R1R5</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Omega-scoloptoxin(05)-Ssm1a (Omega-SLPTX(05)-S...</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVRCDGPMSNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>720</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5339</th>\n",
       "      <td>P0DPX7</td>\n",
       "      <td>Scolopendra morsitans (Tanzanian blue ringleg ...</td>\n",
       "      <td>U-scoloptoxin(05)-Sm1a (U-SLPTX(05)-Sm1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNVLYTKIFFILILTRTSSALKCVQCFTGYGNIVQDCNKTSYSEPK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>720</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5340</th>\n",
       "      <td>P0DPX8</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er1a (U-SLPTX(05)-Er1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MLSLGVSIFLLVFLIPENSGLECYQCTWMKNSQSPDNCYKNLPNAT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>721</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5341</th>\n",
       "      <td>P0DPX9</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er2a (U-SLPTX(05)-Er2a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MTFVVAAVVLLTVVPLATPLKCVQCDGPLTEFDCKTTVPEAKDCPQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>720</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5342</th>\n",
       "      <td>P0DPY0</td>\n",
       "      <td>Ethmostigmus rubripes (Giant centipede)</td>\n",
       "      <td>U-scoloptoxin(05)-Er3a (U-SLPTX(05)-Er3a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRSWFVFVALLAVVFLPSSLDALKCIQCDSQPNRDECKTTLPEARD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>720</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5343</th>\n",
       "      <td>P0DPY1</td>\n",
       "      <td>Cormocephalus westwoodi (Westwood's green cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Cw1a (U-SLPTX(05)-Cw1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNPLNLSTFIVFTLFAASATTALTCFQCTTSEGSDYCVSSFPKPSQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>720</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 93
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### replace Protein families with \"other\" if counts <= 10 and save to tsv for model",
   "id": "e423f25794a1ec4e"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T10:04:01.027554Z",
     "start_time": "2025-03-05T10:04:00.914686Z"
    }
   },
   "cell_type": "code",
   "source": [
    "no_na[\"Protein families\"] = no_na[\"Protein families\"].apply(lambda x: \"other\" if no_na[\"Protein families\"].value_counts()[x] <= 10 else x)\n",
    "no_na"
   ],
   "id": "8df294feeae090d5",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "6     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "9     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "18    A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "19        A0S864               Boiga irregularis (Brown tree snake)   \n",
       "33        A8YPR6       Echis ocellatus (Ocellated saw-scaled viper)   \n",
       "...          ...                                                ...   \n",
       "6386      W4VS32       Trittame loki (Brush-footed trapdoor spider)   \n",
       "6395      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "6397      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "6399      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "6404      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "6     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "9     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "18    Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "19                          Irditoxin subunit A (IrTxA)   \n",
       "33    Snake venom metalloprotease inhibitor 02D01 (0...   \n",
       "...                                                 ...   \n",
       "6386                                       Toxin ICK-15   \n",
       "6395                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "6397                                        Toxin ICK-9   \n",
       "6399                              Conotoxin Vc1 (H_Vc1)   \n",
       "6404                                       Toxin ICK-18   \n",
       "\n",
       "                           Protein families  \\\n",
       "6                                     other   \n",
       "9          Arthropod phospholipase D family   \n",
       "18    Venom metalloproteinase (M12B) family   \n",
       "19          Snake three-finger toxin family   \n",
       "33               Natriuretic peptide family   \n",
       "...                                     ...   \n",
       "6386                                  other   \n",
       "6395                                  other   \n",
       "6397                                  other   \n",
       "6399                                  other   \n",
       "6404                                  other   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "6     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       15   \n",
       "9     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN      200   \n",
       "18    MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN       55   \n",
       "19    MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...      NaN       44   \n",
       "33    MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...      NaN       88   \n",
       "...                                                 ...      ...      ...   \n",
       "6386  MKPIVSILIFCALAVVIMGHPLDSGYGIPHIVEKLPNGQWCKTPGD...      NaN     1298   \n",
       "6395  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN     1295   \n",
       "6397  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN       82   \n",
       "6399  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN     1296   \n",
       "6404  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN     1298   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "6            True              False             True             False  \n",
       "9           False               True            False             False  \n",
       "18           True              False            False              True  \n",
       "19          False               True            False             False  \n",
       "33          False              False             True             False  \n",
       "...           ...                ...              ...               ...  \n",
       "6386        False               True            False             False  \n",
       "6395         True               True            False             False  \n",
       "6397        False               True            False             False  \n",
       "6399         True               True            False             False  \n",
       "6404         True               True            False             False  \n",
       "\n",
       "[1392 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>A0S864</td>\n",
       "      <td>Boiga irregularis (Brown tree snake)</td>\n",
       "      <td>Irditoxin subunit A (IrTxA)</td>\n",
       "      <td>Snake three-finger toxin family</td>\n",
       "      <td>MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>A8YPR6</td>\n",
       "      <td>Echis ocellatus (Ocellated saw-scaled viper)</td>\n",
       "      <td>Snake venom metalloprotease inhibitor 02D01 (0...</td>\n",
       "      <td>Natriuretic peptide family</td>\n",
       "      <td>MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6386</th>\n",
       "      <td>W4VS32</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-15</td>\n",
       "      <td>other</td>\n",
       "      <td>MKPIVSILIFCALAVVIMGHPLDSGYGIPHIVEKLPNGQWCKTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1298</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6395</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1295</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6397</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>other</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>82</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6399</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1296</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6404</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1298</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1392 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 94
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T10:04:04.212838Z",
     "start_time": "2025-03-05T10:04:04.208891Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Get all unique protein families\n",
    "unique_protein_families = set(no_na[\"Protein families\"].unique())\n",
    "\n",
    "# Find unique protein families present in each cluster column separately\n",
    "train_families = set(no_na.loc[no_na[\"Train_Cluster_Rep\"] > 0, \"Protein families\"].unique())\n",
    "val_families = set(no_na.loc[no_na[\"Val_Cluster_Rep\"] > 0, \"Protein families\"].unique())\n",
    "test_families = set(no_na.loc[no_na[\"Test_Cluster_Rep\"] > 0, \"Protein families\"].unique())\n",
    "\n",
    "# Find missing families in each set\n",
    "missing_in_train = unique_protein_families - train_families\n",
    "missing_in_val = unique_protein_families - val_families\n",
    "missing_in_test = unique_protein_families - test_families\n",
    "\n",
    "# Check if all unique protein families are present in all three\n",
    "all_present_in_all = not (missing_in_train or missing_in_val or missing_in_test)\n",
    "\n",
    "# Print results\n",
    "print(f\"All unique protein families are present in all three columns: {all_present_in_all}\")\n",
    "\n",
    "if not all_present_in_all:\n",
    "    if missing_in_train:\n",
    "        print(f\"Missing in Train: {missing_in_train}\")\n",
    "    if missing_in_val:\n",
    "        print(f\"Missing in Val: {missing_in_val}\")\n",
    "    if missing_in_test:\n",
    "        print(f\"Missing in Test: {missing_in_test}\")"
   ],
   "id": "a35f8251fc75be06",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All unique protein families are present in all three columns: False\n",
      "Missing in Val: {'Long chain scorpion toxin family', 'Conotoxin I1 superfamily', 'Venom Kunitz-type family', 'Conotoxin O2 superfamily'}\n",
      "Missing in Test: {'Neurotoxin 04 (omega-agtx) family', 'Neurotoxin 14 (magi-1) family'}\n"
     ]
    }
   ],
   "execution_count": 96
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T10:04:52.520067Z",
     "start_time": "2025-03-05T10:04:52.512121Z"
    }
   },
   "cell_type": "code",
   "source": [
    "duplicate_entries = set(no_na['Entry']).intersection(set(scolop['Entry']))\n",
    "no_na_scolop = no_na[~no_na['Entry'].isin(duplicate_entries)].copy()\n",
    "\n",
    "scolop = scolop[~scolop['Entry'].isin(duplicate_entries)]\n",
    "\n",
    "duplicates_no_na = no_na[no_na['Entry'].isin(duplicate_entries)]\n",
    "duplicates_scolop = scolop[scolop['Entry'].isin(duplicate_entries)]\n",
    "\n",
    "merged_df = pd.concat([no_na_scolop, scolop], ignore_index=True)\n",
    "merged_df"
   ],
   "id": "3c1ddd89cf359ef7",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "2     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "3         A0S864               Boiga irregularis (Brown tree snake)   \n",
       "4         A8YPR6       Echis ocellatus (Ocellated saw-scaled viper)   \n",
       "...          ...                                                ...   \n",
       "1388      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "1389      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1390      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1391      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1392      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "1     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "2     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "3                           Irditoxin subunit A (IrTxA)   \n",
       "4     Snake venom metalloprotease inhibitor 02D01 (0...   \n",
       "...                                                 ...   \n",
       "1388                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "1389                                        Toxin ICK-9   \n",
       "1390                              Conotoxin Vc1 (H_Vc1)   \n",
       "1391                                       Toxin ICK-18   \n",
       "1392          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1          Arthropod phospholipase D family   \n",
       "2     Venom metalloproteinase (M12B) family   \n",
       "3           Snake three-finger toxin family   \n",
       "4                Natriuretic peptide family   \n",
       "...                                     ...   \n",
       "1388                                  other   \n",
       "1389                                  other   \n",
       "1390                                  other   \n",
       "1391                                  other   \n",
       "1392                  Scoloptoxin-05 family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN     15.0   \n",
       "1     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN    200.0   \n",
       "2     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN     55.0   \n",
       "3     MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...      NaN     44.0   \n",
       "4     MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...      NaN     88.0   \n",
       "...                                                 ...      ...      ...   \n",
       "1388  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN   1295.0   \n",
       "1389  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN     82.0   \n",
       "1390  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN   1296.0   \n",
       "1391  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN   1298.0   \n",
       "1392  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN      NaN   \n",
       "\n",
       "     Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \n",
       "0           True             False            True            False  \n",
       "1          False              True           False            False  \n",
       "2           True             False           False             True  \n",
       "3          False              True           False            False  \n",
       "4          False             False            True            False  \n",
       "...          ...               ...             ...              ...  \n",
       "1388        True              True           False            False  \n",
       "1389       False              True           False            False  \n",
       "1390        True              True           False            False  \n",
       "1391        True              True           False            False  \n",
       "1392         NaN               NaN             NaN              NaN  \n",
       "\n",
       "[1393 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0S864</td>\n",
       "      <td>Boiga irregularis (Brown tree snake)</td>\n",
       "      <td>Irditoxin subunit A (IrTxA)</td>\n",
       "      <td>Snake three-finger toxin family</td>\n",
       "      <td>MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A8YPR6</td>\n",
       "      <td>Echis ocellatus (Ocellated saw-scaled viper)</td>\n",
       "      <td>Snake venom metalloprotease inhibitor 02D01 (0...</td>\n",
       "      <td>Natriuretic peptide family</td>\n",
       "      <td>MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1388</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1295.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1389</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>other</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>82.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1390</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1296.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1391</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1298.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1392</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>Scoloptoxin-05 family</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1393 rows Ã— 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 103
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:29:32.451941Z",
     "start_time": "2025-03-05T13:29:32.449550Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df.loc[merged_df[\"Protein families\"] == \"Scoloptoxin-05 family\", \"Protein families\"] = \"other\"",
   "id": "60cb67463e138fe5",
   "outputs": [],
   "execution_count": 106
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:29:39.098614Z",
     "start_time": "2025-03-05T13:29:39.095442Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df[\"Protein families\"].value_counts()",
   "id": "53f84dcfd80a0440",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "other                                               446\n",
       "Short scorpion toxin superfamily                     81\n",
       "Snake three-finger toxin family                      64\n",
       "Venom metalloproteinase (M12B) family                63\n",
       "Long (4 C-C) scorpion toxin superfamily              61\n",
       "Conotoxin O1 superfamily                             48\n",
       "Conotoxin M superfamily                              44\n",
       "Neurotoxin 10 (Hwtx-1) family                        43\n",
       "MCD family                                           42\n",
       "Conotoxin A superfamily                              42\n",
       "Phospholipase A2 family                              40\n",
       "Bradykinin-potentiating peptide family               38\n",
       "Conotoxin T superfamily                              28\n",
       "Neurotoxin 02 (plectoxin) family                     26\n",
       "Snaclec family                                       24\n",
       "Venom Kunitz-type family                             23\n",
       "Peptidase S1 family                                  20\n",
       "Non-disulfide-bridged peptide (NDBP) superfamily     20\n",
       "Cationic peptide 04 (cupiennin) family               19\n",
       "CRISP family                                         19\n",
       "Long (3 C-C) scorpion toxin superfamily              18\n",
       "Conotoxin 02 superfamily                             18\n",
       "Neurotoxin 03 (Tx2) family                           17\n",
       "Neurotoxin 04 (omega-agtx) family                    16\n",
       "Flavin monoamine oxidase family                      16\n",
       "Neurotoxin 14 (magi-1) family                        16\n",
       "Natriuretic peptide family                           15\n",
       "Arthropod phospholipase D family                     14\n",
       "Conotoxin I1 superfamily                             14\n",
       "Formicidae venom precursor-01 superfamily            12\n",
       "Long chain scorpion toxin family                     12\n",
       "Bradykinin-related peptide family                    12\n",
       "Conotoxin O2 superfamily                             11\n",
       "Neurotoxin 19 (CSTX) family                          11\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 107
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:29:58.914618Z",
     "start_time": "2025-03-05T13:29:58.879121Z"
    }
   },
   "cell_type": "code",
   "source": [
    "write_fasta(merged_df, \"data/toxins_no_na_scolop.fasta\")\n",
    "write_fasta(no_na, \"data/toxins_no_na.fasta\")"
   ],
   "id": "8d9e91ecd8353902",
   "outputs": [],
   "execution_count": 108
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:30:00.077743Z",
     "start_time": "2025-03-05T13:30:00.074814Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df[\"Fragment\"].value_counts()",
   "id": "9df49c759f0209ab",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Fragment\n",
       "fragment    368\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 109
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:30:03.623407Z",
     "start_time": "2025-03-05T13:30:03.617912Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df",
   "id": "d5342ea48bbcdb3",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "2     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "3         A0S864               Boiga irregularis (Brown tree snake)   \n",
       "4         A8YPR6       Echis ocellatus (Ocellated saw-scaled viper)   \n",
       "...          ...                                                ...   \n",
       "1388      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "1389      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1390      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1391      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1392      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "1     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "2     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "3                           Irditoxin subunit A (IrTxA)   \n",
       "4     Snake venom metalloprotease inhibitor 02D01 (0...   \n",
       "...                                                 ...   \n",
       "1388                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "1389                                        Toxin ICK-9   \n",
       "1390                              Conotoxin Vc1 (H_Vc1)   \n",
       "1391                                       Toxin ICK-18   \n",
       "1392          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1          Arthropod phospholipase D family   \n",
       "2     Venom metalloproteinase (M12B) family   \n",
       "3           Snake three-finger toxin family   \n",
       "4                Natriuretic peptide family   \n",
       "...                                     ...   \n",
       "1388                                  other   \n",
       "1389                                  other   \n",
       "1390                                  other   \n",
       "1391                                  other   \n",
       "1392                                  other   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN     15.0   \n",
       "1     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN    200.0   \n",
       "2     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN     55.0   \n",
       "3     MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...      NaN     44.0   \n",
       "4     MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...      NaN     88.0   \n",
       "...                                                 ...      ...      ...   \n",
       "1388  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN   1295.0   \n",
       "1389  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN     82.0   \n",
       "1390  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN   1296.0   \n",
       "1391  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN   1298.0   \n",
       "1392  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN      NaN   \n",
       "\n",
       "     Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \\\n",
       "0           True             False            True            False   \n",
       "1          False              True           False            False   \n",
       "2           True             False           False             True   \n",
       "3          False              True           False            False   \n",
       "4          False             False            True            False   \n",
       "...          ...               ...             ...              ...   \n",
       "1388        True              True           False            False   \n",
       "1389       False              True           False            False   \n",
       "1390        True              True           False            False   \n",
       "1391        True              True           False            False   \n",
       "1392         NaN               NaN             NaN              NaN   \n",
       "\n",
       "      Scoloptoxin  \n",
       "0           other  \n",
       "1           other  \n",
       "2           other  \n",
       "3            3FTX  \n",
       "4           other  \n",
       "...           ...  \n",
       "1388        other  \n",
       "1389        other  \n",
       "1390        other  \n",
       "1391        other  \n",
       "1392  Scoloptoxin  \n",
       "\n",
       "[1393 rows x 12 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "      <th>Scoloptoxin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0S864</td>\n",
       "      <td>Boiga irregularis (Brown tree snake)</td>\n",
       "      <td>Irditoxin subunit A (IrTxA)</td>\n",
       "      <td>Snake three-finger toxin family</td>\n",
       "      <td>MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>3FTX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A8YPR6</td>\n",
       "      <td>Echis ocellatus (Ocellated saw-scaled viper)</td>\n",
       "      <td>Snake venom metalloprotease inhibitor 02D01 (0...</td>\n",
       "      <td>Natriuretic peptide family</td>\n",
       "      <td>MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1388</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1295.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1389</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>other</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>82.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1390</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1296.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1391</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1298.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1392</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1393 rows Ã— 12 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 110
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:30:05.359449Z",
     "start_time": "2025-03-05T13:30:05.349614Z"
    }
   },
   "cell_type": "code",
   "source": [
    "merged_df[\"Scoloptoxin\"] = merged_df.apply(\n",
    "    lambda row: \"Scoloptoxin\" if row[\"Entry\"] in [\"I6R1R5\", \"P0DPX5\", \"P0DPX9\", \"P0DPY0\", \"P0DPY1\", \"P0DPX7\", \"P0DPX8\"]\n",
    "    else \"3FTX\" if row[\"Protein families\"] == \"Snake three-finger toxin family\"\n",
    "    else \"other\",\n",
    "    axis=1\n",
    ")\n",
    "\n",
    "#merged_df.to_csv(\"data/scolop.csv\", columns=[\"Entry\", \"Protein families\", \"Fragment\", \"Scoloptoxin\"], index=False, header=[\"identifier\", \"Protein families\", \"Fragment\", \"Scoloptoxin\"])"
   ],
   "id": "87994bfc3f14fe6a",
   "outputs": [],
   "execution_count": 111
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T09:55:48.822579Z",
     "start_time": "2025-03-05T09:55:48.814158Z"
    }
   },
   "cell_type": "code",
   "source": "no_na.to_csv(\"data/toxins_no_na.tsv\", sep=\"\\t\", index=False)",
   "id": "1a29d7c311569a51",
   "outputs": [],
   "execution_count": 89
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### add regex protein_category",
   "id": "f1e8a4e8e4a1dd27"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:38:04.940619Z",
     "start_time": "2025-03-05T13:38:04.922624Z"
    }
   },
   "cell_type": "code",
   "source": [
    "regex_tox = pd.read_csv(\"data/toxins_regex.csv\")\n",
    "regex_tox"
   ],
   "id": "745af5ab2073cdb",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A068B6Q6                       Conus betulinus (Beech cone)   \n",
       "1     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "2     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "4     A0A0D4WTV1        Loxosceles arizonica (Arizona brown spider)   \n",
       "...          ...                                                ...   \n",
       "6423      P0CI18      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6424      P0CI19      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6425      P0CI21      Hastula hectica (Sea snail) (Impages hectica)   \n",
       "6426      Q45RU8                     Conus striatus (Striated cone)   \n",
       "6427      Q9BP49                 Conus tessulatus (Tessellate cone)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0                                       Conotoxin Bt1.8   \n",
       "1     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "2     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "3     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "4     Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....   \n",
       "...                                                 ...   \n",
       "6423                                Augerpeptide hhe6.3   \n",
       "6424                                Augerpeptide hhe6.4   \n",
       "6425                                 Augerpeptide hhe53   \n",
       "6426                                     Conotoxin S5.1   \n",
       "6427                                Conotoxin TsMEKL-02   \n",
       "\n",
       "                                       Protein families  \\\n",
       "0                               Conotoxin A superfamily   \n",
       "1     Venom metalloproteinase (M12B) family, P-III s...   \n",
       "2                                        Insulin family   \n",
       "3                                        Insulin family   \n",
       "4     Arthropod phospholipase D family, Class II sub...   \n",
       "...                                                 ...   \n",
       "6423                                                NaN   \n",
       "6424                                                NaN   \n",
       "6425                                                NaN   \n",
       "6426                                                NaN   \n",
       "6427                                                NaN   \n",
       "\n",
       "                                               Sequence  Fragment  \\\n",
       "0                 PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG  fragment   \n",
       "1     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...       NaN   \n",
       "2     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...       NaN   \n",
       "3     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...       NaN   \n",
       "4     EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...  fragment   \n",
       "...                                                 ...       ...   \n",
       "6423                    VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD       NaN   \n",
       "6424               SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP       NaN   \n",
       "6425             GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI       NaN   \n",
       "6426       HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ  fragment   \n",
       "6427                               NNPPCCSGYVCEGVYCAVDV  fragment   \n",
       "\n",
       "         protein_category  \n",
       "0               conotoxin  \n",
       "1       metalloproteinase  \n",
       "2         hormone_related  \n",
       "3         hormone_related  \n",
       "4     phospholipase_other  \n",
       "...                   ...  \n",
       "6423                  NaN  \n",
       "6424                  NaN  \n",
       "6425                  NaN  \n",
       "6426                  NaN  \n",
       "6427                  NaN  \n",
       "\n",
       "[6428 rows x 7 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>protein_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A068B6Q6</td>\n",
       "      <td>Conus betulinus (Beech cone)</td>\n",
       "      <td>Conotoxin Bt1.8</td>\n",
       "      <td>Conotoxin A superfamily</td>\n",
       "      <td>PDGRNAAAKAFDLITPTVRKGCCSNPACILNNPNQCG</td>\n",
       "      <td>fragment</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family, P-III s...</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metalloproteinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hormone_related</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hormone_related</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0D4WTV1</td>\n",
       "      <td>Loxosceles arizonica (Arizona brown spider)</td>\n",
       "      <td>Dermonecrotic toxin LarSicTox-betaID1 (EC 4.6....</td>\n",
       "      <td>Arthropod phospholipase D family, Class II sub...</td>\n",
       "      <td>EGAEQDGSERTDGGRPIWNIAHMVNNKQAIDKYLDKGANSVESDVS...</td>\n",
       "      <td>fragment</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6423</th>\n",
       "      <td>P0CI18</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLFTPPELLGCGNRCSDDCCKWGRCQPGCTD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6424</th>\n",
       "      <td>P0CI19</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SSLHCGDDPWCPTGCCENEDCDIGCKRDWEKSRSQP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6425</th>\n",
       "      <td>P0CI21</td>\n",
       "      <td>Hastula hectica (Sea snail) (Impages hectica)</td>\n",
       "      <td>Augerpeptide hhe53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GLSQSGCQAFTGRWCVGCERLRSRVVWECSPKRVVNSI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6426</th>\n",
       "      <td>Q45RU8</td>\n",
       "      <td>Conus striatus (Striated cone)</td>\n",
       "      <td>Conotoxin S5.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HNDLIRAGLTVCLSENRKRLTCSGLLNMAGSVCCKVDTSCCSSQ</td>\n",
       "      <td>fragment</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6427</th>\n",
       "      <td>Q9BP49</td>\n",
       "      <td>Conus tessulatus (Tessellate cone)</td>\n",
       "      <td>Conotoxin TsMEKL-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NNPPCCSGYVCEGVYCAVDV</td>\n",
       "      <td>fragment</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6428 rows Ã— 7 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 112
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:42:03.386242Z",
     "start_time": "2025-03-05T13:42:03.377709Z"
    }
   },
   "cell_type": "code",
   "source": [
    "merged_df = merged_df.merge(regex_tox[['Entry', 'protein_category']], on='Entry', how='left')\n",
    "merged_df"
   ],
   "id": "977b1322c6f129ff",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "1     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "2     A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "3         A0S864               Boiga irregularis (Brown tree snake)   \n",
       "4         A8YPR6       Echis ocellatus (Ocellated saw-scaled viper)   \n",
       "...          ...                                                ...   \n",
       "1388      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "1389      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1390      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "1391      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "1392      P0DPX5  Scolopendra alternans (Florida Keys giant cent...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "1     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "2     Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "3                           Irditoxin subunit A (IrTxA)   \n",
       "4     Snake venom metalloprotease inhibitor 02D01 (0...   \n",
       "...                                                 ...   \n",
       "1388                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "1389                                        Toxin ICK-9   \n",
       "1390                              Conotoxin Vc1 (H_Vc1)   \n",
       "1391                                       Toxin ICK-18   \n",
       "1392          U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)   \n",
       "\n",
       "                           Protein families  \\\n",
       "0                                     other   \n",
       "1          Arthropod phospholipase D family   \n",
       "2     Venom metalloproteinase (M12B) family   \n",
       "3           Snake three-finger toxin family   \n",
       "4                Natriuretic peptide family   \n",
       "...                                     ...   \n",
       "1388                                  other   \n",
       "1389                                  other   \n",
       "1390                                  other   \n",
       "1391                                  other   \n",
       "1392                                  other   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN     15.0   \n",
       "1     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN    200.0   \n",
       "2     MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN     55.0   \n",
       "3     MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...      NaN     44.0   \n",
       "4     MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...      NaN     88.0   \n",
       "...                                                 ...      ...      ...   \n",
       "1388  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN   1295.0   \n",
       "1389  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN     82.0   \n",
       "1390  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN   1296.0   \n",
       "1391  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN   1298.0   \n",
       "1392  MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...      NaN      NaN   \n",
       "\n",
       "     Cluster_Rep Train_Cluster_Rep Val_Cluster_Rep Test_Cluster_Rep  \\\n",
       "0           True             False            True            False   \n",
       "1          False              True           False            False   \n",
       "2           True             False           False             True   \n",
       "3          False              True           False            False   \n",
       "4          False             False            True            False   \n",
       "...          ...               ...             ...              ...   \n",
       "1388        True              True           False            False   \n",
       "1389       False              True           False            False   \n",
       "1390        True              True           False            False   \n",
       "1391        True              True           False            False   \n",
       "1392         NaN               NaN             NaN              NaN   \n",
       "\n",
       "      Scoloptoxin     protein_category  \n",
       "0           other          scoloptoxin  \n",
       "1           other  phospholipase_other  \n",
       "2           other    metalloproteinase  \n",
       "3            3FTX   three_finger_toxin  \n",
       "4           other                  NaN  \n",
       "...           ...                  ...  \n",
       "1388        other            conotoxin  \n",
       "1389        other           neurotoxin  \n",
       "1390        other            conotoxin  \n",
       "1391        other           neurotoxin  \n",
       "1392  Scoloptoxin          scoloptoxin  \n",
       "\n",
       "[1393 rows x 13 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "      <th>Scoloptoxin</th>\n",
       "      <th>protein_category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>other</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Arthropod phospholipase D family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>200.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>phospholipase_other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.0</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>metalloproteinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0S864</td>\n",
       "      <td>Boiga irregularis (Brown tree snake)</td>\n",
       "      <td>Irditoxin subunit A (IrTxA)</td>\n",
       "      <td>Snake three-finger toxin family</td>\n",
       "      <td>MKTLLLAVAVVAFVCLGSADQLGLGRQQIDWGQGQAVGPPYTLCFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>3FTX</td>\n",
       "      <td>three_finger_toxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A8YPR6</td>\n",
       "      <td>Echis ocellatus (Ocellated saw-scaled viper)</td>\n",
       "      <td>Snake venom metalloprotease inhibitor 02D01 (0...</td>\n",
       "      <td>Natriuretic peptide family</td>\n",
       "      <td>MFVSRLAASGLLLLSLLALSLDGKPLPQRQPHHIQPMEQKWLAPDA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1388</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1295.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1389</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>other</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>82.0</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1390</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>other</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1296.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>conotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1391</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>other</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1298.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>neurotoxin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1392</th>\n",
       "      <td>P0DPX5</td>\n",
       "      <td>Scolopendra alternans (Florida Keys giant cent...</td>\n",
       "      <td>U-scoloptoxin(05)-Sa1a (U-SLPTX(05)-Sa1a)</td>\n",
       "      <td>other</td>\n",
       "      <td>MPSLCIIALFGTLTFYTLIPSIHTLKCVICDSPMGNYDCKTTYPAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Scoloptoxin</td>\n",
       "      <td>scoloptoxin</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1393 rows Ã— 13 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 116
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-05T13:44:29.039898Z",
     "start_time": "2025-03-05T13:44:29.035260Z"
    }
   },
   "cell_type": "code",
   "source": "merged_df.to_csv(\"data/scolop.csv\", columns=[\"Entry\", \"Protein families\", \"Fragment\", \"Scoloptoxin\", \"protein_category\"], index=False, header=[\"identifier\", \"Protein families\", \"Fragment\", \"Scoloptoxin\", \"protein_category\"])",
   "id": "19e2b93a29fb2730",
   "outputs": [],
   "execution_count": 119
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### cut off SP and then see which Entry is full_seq, mature or fragment",
   "id": "9a49267cb1167227"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-07T12:25:28.884791Z",
     "start_time": "2025-03-07T12:25:28.860040Z"
    }
   },
   "cell_type": "code",
   "source": "write_fasta(merged_df, \"data/toxins_no_na_scolop.fasta\")",
   "id": "c1bb5d14e430b550",
   "outputs": [],
   "execution_count": 121
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "!signalp6",
   "id": "5d276eba84c1e33c"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
