{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "initial_id",
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n",
    "from sklearn.utils import shuffle"
   ]
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "# Toxins",
   "id": "4cc7c0697a550d73"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": [
    "tox = pd.read_csv('./data/toxins.tsv', sep='\\t')\n",
    "tox = tox.dropna(subset=[\"Protein families\"])\n",
    "\n",
    "tox"
   ],
   "id": "9f80e78addf01d6"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": [
    "tox['Protein families'] = tox['Protein families'].str.split(',').str[0]\n",
    "tox['Protein families'] = tox['Protein families'].str.split(';').str[0]"
   ],
   "id": "5d228fcac1547583"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": [
    "tox['Protein families'] = tox['Protein families'].replace('I1 superfamily', 'Conotoxin I1 superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('O1 superfamily', 'Conotoxin O1 superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('O2 superfamily', 'Conotoxin O2 superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('E superfamily', 'Conotoxin E superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('F superfamily', 'Conotoxin F superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('Conotoxin M family', 'Conotoxin M superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('Conotoxin B2 family', 'Conotoxin B2 superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('Conotoxin O1 family', 'Conotoxin O1 superfamily')\n",
    "tox['Protein families'] = tox['Protein families'].replace('Conotoxin O2 family', 'Conotoxin O2 superfamily')"
   ],
   "id": "787745e580d7645e"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "source": [
    "mapping = {\n",
    "    r'Conotoxin.*': 'Conotoxin family',\n",
    "    r'Neurotoxin.*': 'Neurotoxin family',\n",
    "    r'Scoloptoxin.*|Scolopendra.*': 'Scoloptoxin family',\n",
    "    r'Caterpillar.*': 'Caterpillar family',\n",
    "    r'Teretoxin.*': 'Teretoxin family',\n",
    "    r'Limacoditoxin.*': 'Limacoditoxin family',\n",
    "    r'Scutigerotoxin.*': 'Scutigerotoxin family',\n",
    "    r'Cationic peptide.*': 'Cationic peptide family',\n",
    "    r'Formicidae venom.*': 'Formicidae venom family',\n",
    "    r'Bradykinin-potentiating peptide family|Natriuretic peptide family': 'Natriuretic, Bradykinin potentiating peptide family',\n",
    "    r'.*phospholipase.*|.*Phospholipase.*': 'Phospholipase family'\n",
    "}\n",
    "\n",
    "# Apply mapping\n",
    "for pattern, replacement in mapping.items():\n",
    "    tox['Protein families'] = tox['Protein families'].str.replace(pattern, replacement, regex=True)\n",
    "\n",
    "tox['Protein families'].value_counts()"
   ],
   "id": "a5e8ed17a43c3aa7",
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## Clustering\n",
    "### Fasta Generation"
   ],
   "id": "2d482cbf29bcb65b"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "source": [
    "def write_fasta(df, filename):\n",
    "    \"\"\"Writes a DataFrame to a FASTA file.\"\"\"\n",
    "    with open(filename, \"w\") as f:\n",
    "        for _, row in df.iterrows():\n",
    "            f.write(f\">{row['Entry']}\\n{row['Sequence']}\\n\")\n",
    "\n",
    "write_fasta(tox, \"./data/toxins.fasta\")"
   ],
   "id": "f5fca1dfa7e75549",
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### run mmseqs2 30% sequence similarity clustering",
   "id": "5ad11e4ab908383c"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "!mmseqs easy-cluster data/toxins.fasta data/mmseqs_0.3/toxins_cluster data/mmseqs_0.3/tmp --min-seq-id 0.3",
   "id": "de696b630452ec73"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "source": [
    "rep_data = []\n",
    "with open(\"./data/mmseqs_0.3/toxins_cluster_rep_seq.fasta\") as f:\n",
    "    for record in SeqIO.parse(f, \"fasta\"):\n",
    "        rep_data.append([record.id, str(record.seq)])\n",
    "\n",
    "toxins_rep = pd.DataFrame(rep_data, columns=[\"Entry\", \"Sequence\"])\n",
    "\n",
    "toxins_rep = toxins_rep.merge(tox[[\"Entry\", \"Protein families\"]], on=\"Entry\", how=\"left\")\n",
    "toxins_rep"
   ],
   "id": "309a620714199d5d",
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "source": [
    "cluster_df = pd.read_csv(\"./data/mmseqs_0.3/toxins_cluster_cluster.tsv\", sep='\\t', header=None, names=['Entry', 'cluster_member'])\n",
    "cluster_df"
   ],
   "id": "f216e7ac42d5aed8",
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### noise",
   "id": "eb592b42e4c59a15"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T19:59:37.561182Z",
     "start_time": "2025-03-26T19:59:37.558255Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "403\n"
     ]
    }
   ],
   "execution_count": 212,
   "source": "print((cluster_df['Entry'].value_counts() != 1).sum())",
   "id": "49eca55b0d53227"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-05-09T09:24:18.287593Z",
     "start_time": "2025-05-09T09:24:18.219562Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'cluster_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001B[0;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[0;31mNameError\u001B[0m                                 Traceback (most recent call last)",
      "Cell \u001B[0;32mIn[1], line 1\u001B[0m\n\u001B[0;32m----> 1\u001B[0m cluster_df[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mCluster\u001B[39m\u001B[38;5;124m'\u001B[39m] \u001B[38;5;241m=\u001B[39m \u001B[43mcluster_df\u001B[49m[\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mEntry\u001B[39m\u001B[38;5;124m'\u001B[39m]\u001B[38;5;241m.\u001B[39mrank(method\u001B[38;5;241m=\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mdense\u001B[39m\u001B[38;5;124m'\u001B[39m)\u001B[38;5;241m.\u001B[39mastype(\u001B[38;5;28mint\u001B[39m) \u001B[38;5;241m-\u001B[39m \u001B[38;5;241m1\u001B[39m\n\u001B[1;32m      2\u001B[0m cluster_df\n",
      "\u001B[0;31mNameError\u001B[0m: name 'cluster_df' is not defined"
     ]
    }
   ],
   "execution_count": 1,
   "source": [
    "cluster_df['Cluster'] = cluster_df['Entry'].rank(method='dense').astype(int) - 1\n",
    "cluster_df"
   ],
   "id": "8a941fd222896ea3"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-05-09T09:24:18.295589Z",
     "start_time": "2025-03-26T19:59:50.170426Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "1     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "2     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "5034      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5035      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5037      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5038      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "1     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "2     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "5034                                       Toxin ICK-13   \n",
       "5035                                        Toxin ICK-8   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "5037                                       Contryphan-G   \n",
       "5038  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "0     Venom metalloproteinase (M12B) family   \n",
       "1                            Insulin family   \n",
       "2                            Insulin family   \n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "5034                      Neurotoxin family   \n",
       "5035                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "5037                       Conotoxin family   \n",
       "5038                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \n",
       "0     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN       60  \n",
       "1     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN        7  \n",
       "2     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7  \n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8  \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12  \n",
       "...                                                 ...      ...      ...  \n",
       "5034  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN      748  \n",
       "5035  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN       19  \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581  \n",
       "5037  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN       24  \n",
       "5038  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN      751  \n",
       "\n",
       "[5039 rows x 7 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>751</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows × 7 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 214,
   "source": [
    "tox = tox.merge(cluster_df[['cluster_member', 'Cluster']], left_on='Entry', right_on='cluster_member', how='left').drop(columns='cluster_member')\n",
    "tox"
   ],
   "id": "92c29f7d19e33c88"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-05-09T09:24:18.297534Z",
     "start_time": "2025-03-26T19:59:52.270542Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "1     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "2     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "5034      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5035      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5037      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5038      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "1     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "2     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "5034                                       Toxin ICK-13   \n",
       "5035                                        Toxin ICK-8   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "5037                                       Contryphan-G   \n",
       "5038  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "0     Venom metalloproteinase (M12B) family   \n",
       "1                            Insulin family   \n",
       "2                            Insulin family   \n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "5034                      Neurotoxin family   \n",
       "5035                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "5037                       Conotoxin family   \n",
       "5038                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN       60   \n",
       "1     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN        7   \n",
       "2     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7   \n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "...                                                 ...      ...      ...   \n",
       "5034  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN      748   \n",
       "5035  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN       19   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "5037  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN       24   \n",
       "5038  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN      751   \n",
       "\n",
       "      Cluster_Rep  \n",
       "0           False  \n",
       "1           False  \n",
       "2           False  \n",
       "3            True  \n",
       "4            True  \n",
       "...           ...  \n",
       "5034        False  \n",
       "5035        False  \n",
       "5036        False  \n",
       "5037        False  \n",
       "5038         True  \n",
       "\n",
       "[5039 rows x 8 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>751</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows × 8 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 215,
   "source": [
    "rep_seq_ids = {record.id for record in SeqIO.parse(\"data/mmseqs_0.3/toxins_cluster_rep_seq.fasta\", \"fasta\")}\n",
    "tox['Cluster_Rep'] = tox['Entry'].isin(rep_seq_ids)\n",
    "tox"
   ],
   "id": "787f1d177f91fcd6"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Train-Val-Test sets with 70:15:15 split",
   "id": "5f0d4f045230dbb1"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-05-09T09:24:18.298485Z",
     "start_time": "2025-03-26T19:59:55.343607Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total entries: 5039\n",
      "Train entries: 3527 (69.99%)\n",
      "Validation entries: 755 (14.98%)\n",
      "Test entries: 757 (15.02%)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(           Entry                                           Organism  \\\n",
       " 0         Q4VM07  Macrovipera lebetinus (Levantine viper) (Viper...   \n",
       " 1         B6DD05  Lycosa singoriensis (Wolf spider) (Aranea sing...   \n",
       " 2         P0CI57      Lychas mucronatus (Chinese swimming scorpion)   \n",
       " 3         P10116  Laticauda colubrina (Yellow-lipped sea krait) ...   \n",
       " 4         Q1ELU5                       Lachesana tarabaevi (Spider)   \n",
       " ...          ...                                                ...   \n",
       " 3522  A0A2P1BRQ0          Scorpaena plumieri (Spotted scorpionfish)   \n",
       " 3523      C1IC52        Walterinnesia aegyptia (Desert black snake)   \n",
       " 3524      P82942    Naja kaouthia (Monocled cobra) (Naja siamensis)   \n",
       " 3525      W4VS70              Conus victoriae (Queen Victoria cone)   \n",
       " 3526      Q3YEE2                    Conus capitaneus (Captain cone)   \n",
       " \n",
       "                                           Protein names  \\\n",
       " 0     Zinc metalloproteinase-disintegrin-like VLAIP-...   \n",
       " 1     U10-lycotoxin-Ls1b (Toxin-like structure LSTX-J3)   \n",
       " 2                                 Neurotoxin LmNaTx11.6   \n",
       " 3     Basic phospholipase A2 2 (svPLA2) (EC 3.1.1.4)...   \n",
       " 4     M-zodatoxin-Lt4a (M-ZDTX-Lt4a) [Cleaved into: ...   \n",
       " ...                                                 ...   \n",
       " 3522               Cytolytic toxin-alpha (Sp-CTx-alpha)   \n",
       " 3523  Kunitz-type serine protease inhibitor spermati...   \n",
       " 3524  Hemorrhagic metalloproteinase-disintegrin-like...   \n",
       " 3525                           Conotoxin Vc22.1 (E_Vc1)   \n",
       " 3526                                   Conotoxin CaHr91   \n",
       " \n",
       "                              Protein families  \\\n",
       " 0       Venom metalloproteinase (M12B) family   \n",
       " 1                           Neurotoxin family   \n",
       " 2     Long (4 C-C) scorpion toxin superfamily   \n",
       " 3                        Phospholipase family   \n",
       " 4                     Cationic peptide family   \n",
       " ...                                       ...   \n",
       " 3522                    SNTX/VTX toxin family   \n",
       " 3523                 Venom Kunitz-type family   \n",
       " 3524    Venom metalloproteinase (M12B) family   \n",
       " 3525                         Conotoxin family   \n",
       " 3526                         Conotoxin family   \n",
       " \n",
       "                                                Sequence Fragment  Cluster  \\\n",
       " 0     MMQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKITALPKGA...      NaN       60   \n",
       " 1     MKLIIFTGLVLFAIVSLIEAEEESGRVCIPLNGECSKSPDKCCDNI...      NaN      108   \n",
       " 2     MKIVIIFFIAMMAVGVYSKDGYLVKKNGCAYDCTVLFGDSNCNGEC...      NaN      682   \n",
       " 3     NLIQFSELIQCANKGKRATYYYMDYGCYCGKGGSGTPVDDLDRCCK...      NaN      199   \n",
       " 4     MKFSIIALALAVAFVCVAESRSEEEGYDVSEEIQAEELEEAERGGI...      NaN      684   \n",
       " ...                                                 ...      ...      ...   \n",
       " 3522  MSSDIIMAGLGRPFTLGFLYDARREKLIPGFSLFGDETLQKYATST...      NaN       26   \n",
       " 3523  MSSGCLLLLLGLLTLWAELTPVSGRPRLCELPAESGLCNAYIPSFY...      NaN      163   \n",
       " 3524  TNTPEQDRYLQAEKYIEFYVIVDNRMYRYYNYDKPAIKIRVYEMIN...      NaN      609   \n",
       " 3525  MMTRVFLAMFFLLVLTKGWPRLYDGDCTRGPNMHITCFKDQSCGLI...      NaN       13   \n",
       " 3526  MKLTCALIITVLFLSITADDSRGKQGYRALKSIAGMLNSKTVRECR...      NaN      731   \n",
       " \n",
       "       Cluster_Rep  \n",
       " 0           False  \n",
       " 1           False  \n",
       " 2           False  \n",
       " 3           False  \n",
       " 4           False  \n",
       " ...           ...  \n",
       " 3522         True  \n",
       " 3523        False  \n",
       " 3524         True  \n",
       " 3525        False  \n",
       " 3526        False  \n",
       " \n",
       " [3527 rows x 8 columns],\n",
       "           Entry                                           Organism  \\\n",
       " 0        D2Y2I6  Cyriopagopus hainanus (Chinese bird spider) (H...   \n",
       " 1    A0A7S8RFI7             Tityus serrulatus (Brazilian scorpion)   \n",
       " 2        P0DM21                 Conus textile (Cloth-of-gold cone)   \n",
       " 3        Q7YT39  Grammostola rosea (Chilean rose tarantula) (Gr...   \n",
       " 4        Q8JI38  Laticauda semifasciata (Black-banded sea krait...   \n",
       " ..          ...                                                ...   \n",
       " 750      Q2PQJ3                 Bothrops jararacussu (Jararacussu)   \n",
       " 751      B1P1E6  Chilobrachys guangxiensis (Chinese earth tiger...   \n",
       " 752      D2Y1Z6  Cyriopagopus hainanus (Chinese bird spider) (H...   \n",
       " 753      P82424  Neoponera goeldii (Ponerine ant) (Pachycondyla...   \n",
       " 754      B0FXM3  Crotalus durissus terrificus (South American r...   \n",
       " \n",
       "                                          Protein names  \\\n",
       " 0    Mu-theraphotoxin-Hhn2i (Mu-TRTX-Hhn2i) (Hainan...   \n",
       " 1    Putative sodium channel toxin Ts37 (Tityustoxi...   \n",
       " 2                Alpha-conotoxin TxIA (Conotoxin tx1a)   \n",
       " 3    M-theraphotoxin-Gr1a (M-TRTX-Gr1a) (GsMTx-4) (...   \n",
       " 4         Cysteine-rich venom protein latisemin (CRVP)   \n",
       " ..                                                 ...   \n",
       " 750  Thrombin-like enzyme BjussuSP-1 (SVTLE BjussuS...   \n",
       " 751  U14-theraphotoxin-Cg1a 1 (U14-TRTX-Cg1a) (Jing...   \n",
       " 752  Mu-theraphotoxin-Hhn2j 4 (Mu-TRTX-Hhn2j) (Hain...   \n",
       " 753  U1-poneritoxin-Ng1b (U1-PONTX-Ng1b) (Poneratox...   \n",
       " 754  Thrombin-like enzyme gyroxin B1.7 (SVTLE gyrox...   \n",
       " \n",
       "                                      Protein families  \\\n",
       " 0                                   Neurotoxin family   \n",
       " 1             Long (4 C-C) scorpion toxin superfamily   \n",
       " 2                                    Conotoxin family   \n",
       " 3                                   Neurotoxin family   \n",
       " 4                                        CRISP family   \n",
       " ..                                                ...   \n",
       " 750                               Peptidase S1 family   \n",
       " 751                                 Neurotoxin family   \n",
       " 752                                 Neurotoxin family   \n",
       " 753  Non-disulfide-bridged peptide (NDBP) superfamily   \n",
       " 754                               Peptidase S1 family   \n",
       " \n",
       "                                               Sequence Fragment  Cluster  \\\n",
       " 0    MKASMFLALAGLVLLFVVGYASESEEKEFPRELLSKIFAVDDFKGE...      NaN      338   \n",
       " 1    MAGEWACLLVSLVLLWGAAGSRDGFLLDRNFCRIKCSFLGSNSMCA...      NaN       51   \n",
       " 2    MFTVFLLVVLATAVVSFTSDRASDDGKAAASDLITLTIKGCCSRPP...      NaN       31   \n",
       " 3    MKTSVVFVIAGLALLSVVCYASELKEQSSVNEVLSTIFHFEQPEER...      NaN      338   \n",
       " 4    MIAFIVLLSLAAVLQQSSGTVDFASESSNKRENQKEIVDKHNALRR...      NaN      725   \n",
       " ..                                                 ...      ...      ...   \n",
       " 750  VLGGDECDINEHPFLAFLYSHGYFCGLTLINQEWVVTAAHCDSTNF...      NaN      700   \n",
       " 751  MKTSVLLVILGIAAITVQCTASESVEQDSLRTFVDAVLGWNAEMAS...      NaN       64   \n",
       " 752  MKASMFLALAGLVLLFVVGYASESEEKEFPIELLSKIFAVDVFKGE...      NaN      338   \n",
       " 753                          WLGSALKIGAKLLPSVVGLFQKKKK      NaN      604   \n",
       " 754  MVLIRVLANLLILQLSYAQKSSELVIGGDECNINEHRLLAIVYTNS...      NaN      700   \n",
       " \n",
       "      Cluster_Rep  \n",
       " 0          False  \n",
       " 1           True  \n",
       " 2          False  \n",
       " 3          False  \n",
       " 4          False  \n",
       " ..           ...  \n",
       " 750        False  \n",
       " 751        False  \n",
       " 752        False  \n",
       " 753        False  \n",
       " 754        False  \n",
       " \n",
       " [755 rows x 8 columns],\n",
       "           Entry                                           Organism  \\\n",
       " 0        Q91136                          Naja atra (Chinese cobra)   \n",
       " 1    A0A0B5A7P2  Conus geographus (Geography cone) (Nubecula ge...   \n",
       " 2        P0DUS5            Doratifera vulnerans (Mottled cup moth)   \n",
       " 3        F1CJ67  Hottentotta judaicus (Black scorpion) (Buthotu...   \n",
       " 4    A0A2P1BST7       Micrurus mipartitus (Red-tailed coral snake)   \n",
       " ..          ...                                                ...   \n",
       " 752      P83312                  Parabuthus schlechteri (Scorpion)   \n",
       " 753  A0A0B5AC90  Conus tulipa (Fish-hunting cone snail) (Tulip ...   \n",
       " 754      P0DPU7  Scolopendra dehaani (Thai centipede) (Scolopen...   \n",
       " 755  A0A125S9E4                   Conus imperialis (Imperial cone)   \n",
       " 756  A0A0B5ABD9  Conus geographus (Geography cone) (Nubecula ge...   \n",
       " \n",
       "                                          Protein names  \\\n",
       " 0      Cytotoxin I-like T-15 (Cardiotoxin I-like T-15)   \n",
       " 1    Con-Ins G1c (Insulin 1c) [Cleaved into: Con-In...   \n",
       " 2    DELTA-limacoditoxin(2)-Dv11 (DELTA-LCTX(2)-Dv1...   \n",
       " 3                                 U10-hottentoxin-Hj2a   \n",
       " 4    Mipartoxin-1A (Mipartoxin-IA) (Three-finger to...   \n",
       " ..                                                 ...   \n",
       " 752  Parabutoporin (PP) (Non-disulfide-bridged pept...   \n",
       " 753  Con-Ins T1B (Con-Ins T2) (Insulin 2) [Cleaved ...   \n",
       " 754                                 Scoloptoxin SSD996   \n",
       " 755                Conotoxin Im23a (Conopeptide im010)   \n",
       " 756  Con-Ins G2 (Insulin 2) [Cleaved into: Con-Ins ...   \n",
       " \n",
       "                                      Protein families  \\\n",
       " 0                     Snake three-finger toxin family   \n",
       " 1                                      Insulin family   \n",
       " 2                                Limacoditoxin family   \n",
       " 3                    Short scorpion toxin superfamily   \n",
       " 4                     Snake three-finger toxin family   \n",
       " ..                                                ...   \n",
       " 752  Non-disulfide-bridged peptide (NDBP) superfamily   \n",
       " 753                                    Insulin family   \n",
       " 754                                Scoloptoxin family   \n",
       " 755                                  Conotoxin family   \n",
       " 756                                    Insulin family   \n",
       " \n",
       "                                               Sequence Fragment  Cluster  \\\n",
       " 0    MKTLLLTLVVVTIVCLDLGYTLKCNKLIPIASKTCTAGKNLCYKMF...      NaN      179   \n",
       " 1    MTTSFYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7   \n",
       " 2    MKFAKTFLLLFVVLLLLSIVMAEPKRGFGKLLRKVFKVGRRVAGSA...      NaN      477   \n",
       " 3    MQKLLIILILFCILKFNVDVEGRTAFPCNQSKCQERCKKEIKKGKC...      NaN      178   \n",
       " 4    MKTLLLTLVVVTIVCLDLGNSLKCYVSREGETQTCPEGEKLCEKYA...      NaN      179   \n",
       " ..                                                 ...      ...      ...   \n",
       " 752      FKLGSFLKKAWKSKLAKKLRAKGKEMLKDYAKGLLEGGSEEVPGQ      NaN      618   \n",
       " 753  MTTSFYFLLMALGLLLYVCQSSFGNQHTRNSDTPKYRCGSDIPNSY...      NaN        7   \n",
       " 754  MKVFIAVLSIAIVLIIIVSIQETTANTGEAETTFAVPNKKFTREDC...      NaN      381   \n",
       " 755  MIMRMTLTLFVLVVMTAASASGDALTEAKRIPYCGQTGAECYSWCI...      NaN      147   \n",
       " 756  MTTSSYFLLVALGLLLYVRQSFSTHEHTCQLDDPAHPQGKCGSDLV...      NaN        7   \n",
       " \n",
       "      Cluster_Rep  \n",
       " 0          False  \n",
       " 1          False  \n",
       " 2           True  \n",
       " 3          False  \n",
       " 4          False  \n",
       " ..           ...  \n",
       " 752         True  \n",
       " 753        False  \n",
       " 754         True  \n",
       " 755        False  \n",
       " 756        False  \n",
       " \n",
       " [757 rows x 8 columns])"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 216,
   "source": [
    "# Get unique clusters and shuffle them\n",
    "clusters = tox['Cluster'].unique()\n",
    "np.random.seed(42)\n",
    "np.random.shuffle(clusters)\n",
    "\n",
    "# Calculate total number of entries\n",
    "total_entries = len(tox)\n",
    "\n",
    "# Define target sizes\n",
    "train_target = 0.7 * total_entries\n",
    "val_target = 0.15 * total_entries\n",
    "test_target = 0.15 * total_entries\n",
    "\n",
    "# Initialize cluster lists\n",
    "train_clusters, val_clusters, test_clusters = [], [], []\n",
    "\n",
    "# Initialize counters\n",
    "train_count, val_count, test_count = 0, 0, 0\n",
    "\n",
    "# Assign clusters while considering their sizes\n",
    "for cluster in clusters:\n",
    "    cluster_size = len(tox[tox['Cluster'] == cluster])\n",
    "\n",
    "    if train_count + cluster_size <= train_target:\n",
    "        train_clusters.append(cluster)\n",
    "        train_count += cluster_size\n",
    "    elif val_count + cluster_size <= val_target:\n",
    "        val_clusters.append(cluster)\n",
    "        val_count += cluster_size\n",
    "    else:\n",
    "        test_clusters.append(cluster)\n",
    "        test_count += cluster_size\n",
    "\n",
    "# Filter original df by cluster IDs\n",
    "train_df = tox[tox['Cluster'].isin(train_clusters)]\n",
    "val_df = tox[tox['Cluster'].isin(val_clusters)]\n",
    "test_df = tox[tox['Cluster'].isin(test_clusters)]\n",
    "\n",
    "# Shuffle each dataset\n",
    "train_df = shuffle(train_df, random_state=42).reset_index(drop=True)\n",
    "val_df = shuffle(val_df, random_state=42).reset_index(drop=True)\n",
    "test_df = shuffle(test_df, random_state=42).reset_index(drop=True)\n",
    "\n",
    "# Print results\n",
    "print(f\"Total entries: {total_entries}\")\n",
    "print(f\"Train entries: {len(train_df)} ({len(train_df)/total_entries:.2%})\")\n",
    "print(f\"Validation entries: {len(val_df)} ({len(val_df)/total_entries:.2%})\")\n",
    "print(f\"Test entries: {len(test_df)} ({len(test_df)/total_entries:.2%})\")\n",
    "\n",
    "# Output dataframes\n",
    "train_df, val_df, test_df"
   ],
   "id": "d9831b253165a068"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": [
    "write_fasta(train_df, \"./data/toxins_train.fasta\")\n",
    "write_fasta(val_df, \"./data/toxins_val.fasta\")\n",
    "write_fasta(test_df, \"./data/toxins_test.fasta\")"
   ],
   "id": "62e3734a9a4dae34"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### actual redundancy reduction, 50% sequence similarity for train, val and test",
   "id": "32df9f62b08d48e"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "!mmseqs easy-cluster data/toxins_train.fasta data/mmseqs_0.5/train_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "e65e382168685a51"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:04.332771Z",
     "start_time": "2025-03-26T20:00:02.505296Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_val.fasta data/mmseqs_0.5/val_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_val.fasta data/mmseqs_0.5/tmp/8382699393235466370/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[708] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 755 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 755 0s 6ms      \r\n",
      "Sort kmer 0h 0m 0s 0ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 13ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 755 0s 3ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms===============>   ] 95.36% 720 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 755 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1425 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 755 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 755 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 9ms\r\n",
      "\r\n",
      "Size of the sequence database: 755\r\n",
      "Size of the alignment database: 755\r\n",
      "Number of clusters: 266\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/order_redundancy data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/order_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter2 --filter-file data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 266 0s 2ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 266 0s 2ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms==================>] 99.62% 265 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 266 type: Aminoacid\r\n",
      "Target database size: 266 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 266 0s 6ms      \r\n",
      "Time for merging to aln: 0h 0m 0s 0ms\r\n",
      "343 alignments calculated\r\n",
      "337 sequence pairs passed the thresholds (0.982507 of overall calculated)\r\n",
      "1.266917 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 11ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/aln data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 266 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 71 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 266 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 266 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 6ms\r\n",
      "\r\n",
      "Size of the sequence database: 266\r\n",
      "Size of the alignment database: 266\r\n",
      "Number of clusters: 220\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pre_clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 266 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 220 0s 2ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 755 0s 6ms      \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/linclust/11714160101988923731/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 220 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 220 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Index table: Masked residues: 271\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          8662\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000135\r\n",
      "Top 10 k-mers\r\n",
      "    CSPCNN\t7\r\n",
      "    APSPPP\t6\r\n",
      "    GQQIWG\t5\r\n",
      "    GMMFFL\t5\r\n",
      "    GDGPCN\t5\r\n",
      "    SGPIGQ\t5\r\n",
      "    IGVGCN\t4\r\n",
      "    DWQIGN\t4\r\n",
      "    HIPSSN\t4\r\n",
      "    CRMWIT\t4\r\n",
      "Time for index table init: 0h 0m 0s 109ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 220\r\n",
      "Target db start 1 to 220\r\n",
      "[=================================================================] 100.00% 220 0s 3ms      \r\n",
      "[==============================================================>  ] 95.43% 210 eta 0s       \r\n",
      "6.218910 k-mers per position\r\n",
      "47 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 292ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 220 type: Aminoacid\r\n",
      "Target database size: 220 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 220 0s 3ms      \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\r\n",
      "459 alignments calculated\r\n",
      "379 sequence pairs passed the thresholds (0.825708 of overall calculated)\r\n",
      "1.722727 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 220 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 4ms\r\n",
      "\r\n",
      "Size of the sequence database: 220\r\n",
      "Size of the alignment database: 220\r\n",
      "Number of clusters: 171\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 171 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 171 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Index table: Masked residues: 259\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          8201\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000128\r\n",
      "Top 10 k-mers\r\n",
      "    GCGKRI\t3\r\n",
      "    TLVTCL\t3\r\n",
      "    MRMTLL\t3\r\n",
      "    IGVGCN\t3\r\n",
      "    TTPACP\t3\r\n",
      "    SCPNCC\t2\r\n",
      "    GQVPCC\t2\r\n",
      "    GQCDDC\t2\r\n",
      "    AVNHDC\t2\r\n",
      "    GFRPFC\t2\r\n",
      "Time for index table init: 0h 0m 0s 101ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 171\r\n",
      "Target db start 1 to 171\r\n",
      "[=================================================================] 100.00% 171 0s 5ms      \r\n",
      "\r\n",
      "11.075385 k-mers per position\r\n",
      "44 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 280ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 171 type: Aminoacid\r\n",
      "Target database size: 171 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 171 0s 2ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 0ms\r\n",
      "211 alignments calculated\r\n",
      "181 sequence pairs passed the thresholds (0.857820 of overall calculated)\r\n",
      "1.058480 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 0 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 171 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 171 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 4ms\r\n",
      "\r\n",
      "Size of the sequence database: 171\r\n",
      "Size of the alignment database: 171\r\n",
      "Number of clusters: 166\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 166 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 166 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 166 0s 1ms      \r\n",
      "Index table: Masked residues: 259\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 166 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          9822\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000153\r\n",
      "Top 10 k-mers\r\n",
      "    LALLDG\t3\r\n",
      "    VGSLGG\t3\r\n",
      "    GCGKRI\t3\r\n",
      "    ALPAGK\t3\r\n",
      "    TLVTCL\t3\r\n",
      "    MRMTLL\t3\r\n",
      "    MFVLVL\t3\r\n",
      "    IGVGCN\t3\r\n",
      "    LLLLKP\t3\r\n",
      "    VLTVFT\t3\r\n",
      "Time for index table init: 0h 0m 0s 112ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 166\r\n",
      "Target db start 1 to 166\r\n",
      "[=================================================================] 100.00% 166 0s 4ms      \r\n",
      "[================================================================>] 99.39% 165 eta 0s       \r\n",
      "29.999293 k-mers per position\r\n",
      "56 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "1 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 290ms\r\n",
      "align data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 166 type: Aminoacid\r\n",
      "Target database size: 166 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 166 0s 2ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "220 alignments calculated\r\n",
      "169 sequence pairs passed the thresholds (0.768182 of overall calculated)\r\n",
      "1.018072 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "clust data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step2 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 166 0s 0ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 166 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 166 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 4ms\r\n",
      "\r\n",
      "Size of the sequence database: 166\r\n",
      "Size of the alignment database: 166\r\n",
      "Number of clusters: 164\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 220 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 171 0s 2ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 166 0s 3ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 164 0s 4ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 755 0s 8ms      \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_tmp/12967349430691163054/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 164 0s 1ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_rep data/mmseqs_0.5/tmp/8382699393235466370/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu data/mmseqs_0.5/tmp/8382699393235466370/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 164 0s 1ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/input data/mmseqs_0.5/tmp/8382699393235466370/clu_seqs data/mmseqs_0.5/tmp/8382699393235466370/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/8382699393235466370/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 219,
   "source": "!mmseqs easy-cluster data/toxins_val.fasta data/mmseqs_0.5/val_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "d5e2d83cf34133e7"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.118386Z",
     "start_time": "2025-03-26T20:00:04.355232Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "easy-cluster data/toxins_test.fasta data/mmseqs_0.5/test_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5 \r\n",
      "\r\n",
      "MMseqs Version:                     \t17.b804f\r\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\r\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\r\n",
      "Sensitivity                         \t4\r\n",
      "k-mer length                        \t0\r\n",
      "Target search mode                  \t0\r\n",
      "k-score                             \tseq:2147483647,prof:2147483647\r\n",
      "Alphabet size                       \taa:21,nucl:5\r\n",
      "Max sequence length                 \t65535\r\n",
      "Max results per query               \t20\r\n",
      "Split database                      \t0\r\n",
      "Split mode                          \t2\r\n",
      "Split memory limit                  \t0\r\n",
      "Coverage threshold                  \t0.8\r\n",
      "Coverage mode                       \t0\r\n",
      "Compositional bias                  \t1\r\n",
      "Compositional bias                  \t1\r\n",
      "Diagonal scoring                    \ttrue\r\n",
      "Exact k-mer matching                \t0\r\n",
      "Mask residues                       \t1\r\n",
      "Mask residues probability           \t0.9\r\n",
      "Mask lower case residues            \t0\r\n",
      "Mask lower letter repeating N times \t0\r\n",
      "Minimum diagonal score              \t15\r\n",
      "Selected taxa                       \t\r\n",
      "Include identical seq. id.          \tfalse\r\n",
      "Spaced k-mers                       \t1\r\n",
      "Preload mode                        \t0\r\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\r\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\r\n",
      "Spaced k-mer pattern                \t\r\n",
      "Local temporary path                \t\r\n",
      "Threads                             \t14\r\n",
      "Compressed                          \t0\r\n",
      "Verbosity                           \t3\r\n",
      "Add backtrace                       \tfalse\r\n",
      "Alignment mode                      \t3\r\n",
      "Alignment mode                      \t0\r\n",
      "Allow wrapped scoring               \tfalse\r\n",
      "E-value threshold                   \t0.001\r\n",
      "Seq. id. threshold                  \t0.5\r\n",
      "Min alignment length                \t0\r\n",
      "Seq. id. mode                       \t0\r\n",
      "Alternative alignments              \t0\r\n",
      "Max reject                          \t2147483647\r\n",
      "Max accept                          \t2147483647\r\n",
      "Score bias                          \t0\r\n",
      "Realign hits                        \tfalse\r\n",
      "Realign score bias                  \t-0.2\r\n",
      "Realign max seqs                    \t2147483647\r\n",
      "Correlation score weight            \t0\r\n",
      "Gap open cost                       \taa:11,nucl:5\r\n",
      "Gap extension cost                  \taa:1,nucl:2\r\n",
      "Zdrop                               \t40\r\n",
      "Rescore mode                        \t0\r\n",
      "Remove hits by seq. id. and coverage\tfalse\r\n",
      "Sort results                        \t0\r\n",
      "Cluster mode                        \t0\r\n",
      "Max connected component depth       \t1000\r\n",
      "Similarity type                     \t2\r\n",
      "Weight file name                    \t\r\n",
      "Cluster Weight threshold            \t0.9\r\n",
      "Single step clustering              \tfalse\r\n",
      "Cascaded clustering steps           \t3\r\n",
      "Cluster reassign                    \tfalse\r\n",
      "Remove temporary files              \ttrue\r\n",
      "Force restart with latest tmp       \tfalse\r\n",
      "MPI runner                          \t\r\n",
      "k-mers per sequence                 \t21\r\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\r\n",
      "Adjust k-mer length                 \tfalse\r\n",
      "Shift hash                          \t67\r\n",
      "Include only extendable             \tfalse\r\n",
      "Skip repeating k-mers               \tfalse\r\n",
      "Database type                       \t0\r\n",
      "Shuffle input database              \ttrue\r\n",
      "Createdb mode                       \t1\r\n",
      "Write lookup file                   \t0\r\n",
      "Offset of numeric ids               \t0\r\n",
      "\r\n",
      "createdb data/toxins_test.fasta data/mmseqs_0.5/tmp/4163972790619549465/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \r\n",
      "\r\n",
      "\u001B[33mShuffle database cannot be combined with --createdb-mode 0\r\n",
      "\u001B[39m\u001B[33mWe recompute with --shuffle 0\r\n",
      "\u001B[39mConverting sequences\r\n",
      "[708] 0s 0ms\r\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\r\n",
      "Time for merging to input: 0h 0m 0s 0ms\r\n",
      "Database type: Aminoacid\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "Create directory data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp\r\n",
      "cluster data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.5 --remove-tmp-files 1 \r\n",
      "\r\n",
      "Set cluster sensitivity to -s 3.000000\r\n",
      "Set cluster mode SET COVER\r\n",
      "Set cluster iterations to 3\r\n",
      "linclust data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "kmermatcher data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Database size: 757 type: Aminoacid\r\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \r\n",
      "\r\n",
      "Generate k-mers list for 1 split\r\n",
      "[=================================================================] 100.00% 757 0s 5ms      \r\n",
      "Sort kmer 0h 0m 0s 0ms\r\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\r\n",
      "Time for fill: 0h 0m 0s 0ms\r\n",
      "Time for merging to pref: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 757 0s 2ms      \r\n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms==================>] 99.07% 750 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 757 0s 3ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1197 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 757 0s 3ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 757 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 8ms\r\n",
      "Total time: 0h 0m 0s 8ms\r\n",
      "\r\n",
      "Size of the sequence database: 757\r\n",
      "Size of the alignment database: 757\r\n",
      "Number of clusters: 275\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 10ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/order_redundancy data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/order_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "filterdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter2 --filter-file data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/order_redundancy --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Filtering using file(s)\r\n",
      "[=================================================================] 100.00% 275 0s 2ms      \r\n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "rescorediagonal data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 275 0s 2ms      \r\n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms================>  ] 96.72% 266 eta 0s       \r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 275 type: Aminoacid\r\n",
      "Target database size: 275 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 275 0s 2ms      \r\n",
      "Time for merging to aln: 0h 0m 0s 0ms\r\n",
      "321 alignments calculated\r\n",
      "303 sequence pairs passed the thresholds (0.943925 of overall calculated)\r\n",
      "1.101818 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/aln data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 28 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 5ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 275\r\n",
      "Size of the alignment database: 275\r\n",
      "Number of clusters: 251\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clust: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 7ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pre_clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/clust --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 275 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 251 0s 3ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 757 0s 6ms      \r\n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pre_clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_filter2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/pref_rescore2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/aln -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/linclust/16066917079072446161/clust -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 251 type: Aminoacid\r\n",
      "Estimated memory consumption: 978M\r\n",
      "Target database size: 251 type: Aminoacid\r\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Index table: Masked residues: 80\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Index statistics\r\n",
      "Entries:          9817\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000153\r\n",
      "Top 10 k-mers\r\n",
      "    TICDYT\t32\r\n",
      "    CGPVGI\t13\r\n",
      "    CYKWHR\t13\r\n",
      "    RGGPKP\t12\r\n",
      "    LDGTCY\t12\r\n",
      "    CPGNFK\t9\r\n",
      "    KRCDPK\t9\r\n",
      "    VKGICP\t9\r\n",
      "    CSDCFP\t9\r\n",
      "    VCDGLK\t7\r\n",
      "Time for index table init: 0h 0m 0s 99ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 154\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 251\r\n",
      "Target db start 1 to 251\r\n",
      "[=================================================================] 100.00% 251 0s 5ms      \r\n",
      "[===========================================================>     ] 92.00% 231 eta 0s       \r\n",
      "6.921199 k-mers per position\r\n",
      "82 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "8 sequences passed prefiltering per query sequence\r\n",
      "4 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step0: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 276ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 251 type: Aminoacid\r\n",
      "Target database size: 251 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 251 0s 3ms      \r\n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\r\n",
      "1675 alignments calculated\r\n",
      "1034 sequence pairs passed the thresholds (0.617313 of overall calculated)\r\n",
      "4.119522 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 8ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 49 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 251 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 4ms\r\n",
      "Total time: 0h 0m 0s 5ms\r\n",
      "\r\n",
      "Size of the sequence database: 251\r\n",
      "Size of the alignment database: 251\r\n",
      "Number of clusters: 132\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 2 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 132 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 132 type: Aminoacid\r\n",
      "Index table k-mer threshold: 145 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 132 0s 1ms      \r\n",
      "Index table: Masked residues: 59\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "Index statistics\r\n",
      "Entries:          5718\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000089\r\n",
      "Top 10 k-mers\r\n",
      "    TLVTCL\t9\r\n",
      "    TICDYT\t7\r\n",
      "    LNVLAC\t2\r\n",
      "    WCKECC\t2\r\n",
      "    SSLKCC\t2\r\n",
      "    VKGNCC\t2\r\n",
      "    KCEIGC\t2\r\n",
      "    GTIRGC\t2\r\n",
      "    CLLYIC\t2\r\n",
      "    TECCKC\t2\r\n",
      "Time for index table init: 0h 0m 0s 106ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 145\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 132\r\n",
      "Target db start 1 to 132\r\n",
      "[=================================================================] 100.00% 132 0s 3ms      \r\n",
      "[=============================================================>   ] 94.66% 125 eta 0s       \r\n",
      "13.313164 k-mers per position\r\n",
      "43 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step1: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 283ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 132 type: Aminoacid\r\n",
      "Target database size: 132 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 132 0s 1ms      \r\n",
      "Time for merging to aln_step1: 0h 0m 0s 1ms\r\n",
      "235 alignments calculated\r\n",
      "142 sequence pairs passed the thresholds (0.604255 of overall calculated)\r\n",
      "1.075758 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 6ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 0 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 132 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 3ms\r\n",
      "Total time: 0h 0m 0s 3ms\r\n",
      "\r\n",
      "Size of the sequence database: 132\r\n",
      "Size of the alignment database: 132\r\n",
      "Number of clusters: 127\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step1: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "createsubdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 -v 3 --subdb-mode 1 \r\n",
      "\r\n",
      "Time for merging to input_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 1ms\r\n",
      "prefilter data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 3 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 1 --comp-bias-corr-scale 1 --diag-score 1 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --mask-n-repeat 0 --min-ungapped-score 15 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Query database size: 127 type: Aminoacid\r\n",
      "Estimated memory consumption: 977M\r\n",
      "Target database size: 127 type: Aminoacid\r\n",
      "Index table k-mer threshold: 136 at k-mer size 6 \r\n",
      "Index table: counting k-mers\r\n",
      "[=================================================================] 100.00% 127 0s 1ms      \r\n",
      "Index table: Masked residues: 59\r\n",
      "Index table: fill\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "Index statistics\r\n",
      "Entries:          6491\r\n",
      "DB size:          488 MB\r\n",
      "Avg k-mer size:   0.000101\r\n",
      "Top 10 k-mers\r\n",
      "    TLVTCL\t6\r\n",
      "    MKLLVV\t6\r\n",
      "    CSDCFP\t3\r\n",
      "    HGSGYA\t2\r\n",
      "    LNVLAC\t2\r\n",
      "    WCKECC\t2\r\n",
      "    KCEIGC\t2\r\n",
      "    TECCKC\t2\r\n",
      "    KTPGLC\t2\r\n",
      "    GKFPLC\t2\r\n",
      "Time for index table init: 0h 0m 0s 108ms\r\n",
      "Process prefiltering step 1 of 1\r\n",
      "\r\n",
      "k-mer similarity threshold: 136\r\n",
      "Starting prefiltering scores calculation (step 1 of 1)\r\n",
      "Query db start 1 to 127\r\n",
      "Target db start 1 to 127\r\n",
      "[=================================================================] 100.00% 127 0s 4ms      \r\n",
      "[=============================================================>   ] 94.44% 120 eta 0s       \r\n",
      "35.515875 k-mers per position\r\n",
      "51 DB matches per sequence\r\n",
      "0 overflows\r\n",
      "2 sequences passed prefiltering per query sequence\r\n",
      "1 median result list length\r\n",
      "0 sequences with 0 size result lists\r\n",
      "Time for merging to pref_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 292ms\r\n",
      "align data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.5 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 1 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Compute score, coverage and sequence identity\r\n",
      "Query database size: 127 type: Aminoacid\r\n",
      "Target database size: 127 type: Aminoacid\r\n",
      "Calculation of alignments\r\n",
      "[=================================================================] 100.00% 127 0s 1ms      \r\n",
      "Time for merging to aln_step2: 0h 0m 0s 0ms\r\n",
      "228 alignments calculated\r\n",
      "132 sequence pairs passed the thresholds (0.578947 of overall calculated)\r\n",
      "1.039370 hits per query sequence\r\n",
      "Time for processing: 0h 0m 0s 5ms\r\n",
      "clust data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step2 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step2 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 14 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \r\n",
      "\r\n",
      "Clustering mode: Set Cover\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "Sort entries\r\n",
      "Find missing connections\r\n",
      "Found 1 new connections.\r\n",
      "Reconstruct initial order\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "Add missing connections\r\n",
      "[=================================================================] 100.00% 127 0s 0ms      \r\n",
      "\r\n",
      "Time for read in: 0h 0m 0s 2ms\r\n",
      "Total time: 0h 0m 0s 3ms\r\n",
      "\r\n",
      "Size of the sequence database: 127\r\n",
      "Size of the alignment database: 127\r\n",
      "Number of clusters: 124\r\n",
      "\r\n",
      "Writing results 0h 0m 0s 0ms\r\n",
      "Time for merging to clu_step2: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 4ms\r\n",
      "mergeclusters data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step2 --threads 14 --compressed 0 -v 3 \r\n",
      "\r\n",
      "Clustering step 1\r\n",
      "[=================================================================] 100.00% 251 0s 1ms      \r\n",
      "Clustering step 2\r\n",
      "[=================================================================] 100.00% 132 0s 2ms      \r\n",
      "Clustering step 3\r\n",
      "[=================================================================] 100.00% 127 0s 3ms      \r\n",
      "Clustering step 4\r\n",
      "[=================================================================] 100.00% 124 0s 4ms      \r\n",
      "Write merged clustering\r\n",
      "[=================================================================] 100.00% 757 0s 7ms      \r\n",
      "Time for merging to clu: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 9ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step_redundancy_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step0 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/pref_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/aln_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/clu_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step1_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2 -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_tmp/16191481891507253488/input_step2_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createtsv data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/cluster.tsv --threads 14 -v 3 \r\n",
      "\r\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2repseq data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_rep --db-load-mode 0 --compressed 0 --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 124 0s 0ms      \r\n",
      "Time for merging to clu_rep: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_rep data/mmseqs_0.5/tmp/4163972790619549465/rep_seq.fasta --use-fasta-header -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "createseqfiledb data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu data/mmseqs_0.5/tmp/4163972790619549465/clu_seqs --threads 14 -v 3 \r\n",
      "\r\n",
      "[=================================================================] 100.00% 124 0s 0ms      \r\n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\r\n",
      "Time for processing: 0h 0m 0s 3ms\r\n",
      "result2flat data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/input data/mmseqs_0.5/tmp/4163972790619549465/clu_seqs data/mmseqs_0.5/tmp/4163972790619549465/all_seqs.fasta -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/input -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/input_h -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_seqs -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu_rep -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n",
      "rmdb data/mmseqs_0.5/tmp/4163972790619549465/clu -v 3 \r\n",
      "\r\n",
      "Time for processing: 0h 0m 0s 0ms\r\n"
     ]
    }
   ],
   "execution_count": 220,
   "source": "!mmseqs easy-cluster data/toxins_test.fasta data/mmseqs_0.5/test_cluster data/mmseqs_0.5/tmp --min-seq-id 0.5",
   "id": "892e66ffe955157d"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.164125Z",
     "start_time": "2025-03-26T20:00:06.156660Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "0     A0A0B4U9L8    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "1     A0A0B5A8P4  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "2     A0A0B5AC95  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "...          ...                                                ...   \n",
       "5034      W4VSI7       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5035      W4VSI8       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5037      X5IFY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "5038      X5IXY8  Conus geographus (Geography cone) (Nubecula ge...   \n",
       "\n",
       "                                          Protein names  \\\n",
       "0     Zinc metalloproteinase-disintegrin-like protei...   \n",
       "1     Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...   \n",
       "2     Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...   \n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "...                                                 ...   \n",
       "5034                                       Toxin ICK-13   \n",
       "5035                                        Toxin ICK-8   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "5037                                       Contryphan-G   \n",
       "5038  Consomatin G2 (ConSST G2) (Somatostatin-relate...   \n",
       "\n",
       "                           Protein families  \\\n",
       "0     Venom metalloproteinase (M12B) family   \n",
       "1                            Insulin family   \n",
       "2                            Insulin family   \n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "...                                     ...   \n",
       "5034                      Neurotoxin family   \n",
       "5035                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "5037                       Conotoxin family   \n",
       "5038                       Conotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "0     MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...      NaN       60   \n",
       "1     MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...      NaN        7   \n",
       "2     MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...      NaN        7   \n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "...                                                 ...      ...      ...   \n",
       "5034  MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...      NaN      748   \n",
       "5035  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...      NaN       19   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "5037  MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...      NaN       24   \n",
       "5038  MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...      NaN      751   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "0           False              False            False             False  \n",
       "1           False              False            False             False  \n",
       "2           False              False            False             False  \n",
       "3            True               True            False             False  \n",
       "4            True               True            False             False  \n",
       "...           ...                ...              ...               ...  \n",
       "5034        False              False            False             False  \n",
       "5035        False              False            False             False  \n",
       "5036        False               True            False             False  \n",
       "5037        False              False            False             False  \n",
       "5038         True              False            False             False  \n",
       "\n",
       "[5039 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A0B4U9L8</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Zinc metalloproteinase-disintegrin-like protei...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MLQVLLVTICLAVFPYQGSSIILESGNVNDYEVVYPQKLTALLKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0B5A8P4</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G3 (Insulin 3) [Cleaved into: Con-Ins ...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSFYFLLVALGLLLYVCQSSFGNQHTRNSDTPKHRCGSELADQY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0B5AC95</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Con-Ins G1a (Insulin 1) [Cleaved into: Con-Ins...</td>\n",
       "      <td>Insulin family</td>\n",
       "      <td>MTTSSYFLLMALGLLLYVCQSSFGNQHTRTFDTPKHRCGSEITNSY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5034</th>\n",
       "      <td>W4VSI7</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-13</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKPTISILIFFALAVAIMGHRLNSGYGIPHIVEKLPNGQWCRTPGD...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5035</th>\n",
       "      <td>W4VSI8</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-8</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQEDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5037</th>\n",
       "      <td>X5IFY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Contryphan-G</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MGKLTILVLVAAVLLSTQAMVQGDGDQPAARNAVPRDDNPDGPSAK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5038</th>\n",
       "      <td>X5IXY8</td>\n",
       "      <td>Conus geographus (Geography cone) (Nubecula ge...</td>\n",
       "      <td>Consomatin G2 (ConSST G2) (Somatostatin-relate...</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MQTAYWVMLMMMVCITAPLPEGGKPNSGIRGLVPNDLTPQHTLRSL...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>751</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5039 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 221,
   "source": [
    "rep_seq_files = {\n",
    "    \"Train_Cluster_Rep\": \"data/mmseqs_0.5/train_cluster_rep_seq.fasta\",\n",
    "    \"Val_Cluster_Rep\": \"data/mmseqs_0.5/val_cluster_rep_seq.fasta\",\n",
    "    \"Test_Cluster_Rep\": \"data/mmseqs_0.5/test_cluster_rep_seq.fasta\",\n",
    "}\n",
    "\n",
    "for col, file in rep_seq_files.items():\n",
    "    rep_seq_ids = {record.id for record in SeqIO.parse(file, \"fasta\")}\n",
    "    df[col] = df['Entry'].isin(rep_seq_ids)\n",
    "\n",
    "df"
   ],
   "id": "ff6564be9557a057"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.183955Z",
     "start_time": "2025-03-26T20:00:06.181439Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All representative sequences after 50% redundancy reduction: 1045\n",
      "Train entries: 757 (72.44%)\n",
      "Validation entries: 164 (15.69%)\n",
      "Test entries: 124 (11.87%)\n"
     ]
    }
   ],
   "execution_count": 222,
   "source": [
    "all_50_pct_seq = (\n",
    "    tox[\"Train_Cluster_Rep\"].sum().item() +\n",
    "    tox[\"Val_Cluster_Rep\"].sum().item() +\n",
    "    tox[\"Test_Cluster_Rep\"].sum().item()\n",
    ")\n",
    "print(\"All representative sequences after 50% redundancy reduction:\", all_50_pct_seq)\n",
    "print(f\"Train entries: {tox[\"Train_Cluster_Rep\"].sum().item()} ({tox[\"Train_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")\n",
    "\n",
    "print(f\"Validation entries: {tox[\"Val_Cluster_Rep\"].sum().item()} ({tox[\"Val_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")\n",
    "print(f\"Test entries: {tox[\"Test_Cluster_Rep\"].sum().item()} ({tox[\"Test_Cluster_Rep\"].sum().item()/all_50_pct_seq:.2%})\")"
   ],
   "id": "af4f3eb0590bc5f"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.213984Z",
     "start_time": "2025-03-26T20:00:06.208695Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           Entry                                           Organism  \\\n",
       "3     A0A0D4WV12                    Sicarius terrosus (Cave spider)   \n",
       "4     A0A0N7CSQ4  Scolopendra mutilans (Chinese red-headed centi...   \n",
       "7     A0A1L4BJ98                   Hemiscorpius lepturus (Scorpion)   \n",
       "15    A0A5C2A2T2                   Conus purpurascens (Purple cone)   \n",
       "16    A0A6B7FMR5    Vipera ammodytes ammodytes (Western sand viper)   \n",
       "...          ...                                                ...   \n",
       "5024      W4VSB6              Conus victoriae (Queen Victoria cone)   \n",
       "5026      W4VSB9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5028      W4VSG7              Conus victoriae (Queen Victoria cone)   \n",
       "5033      W4VSI6       Trittame loki (Brush-footed trapdoor spider)   \n",
       "5036      W4VSI9       Trittame loki (Brush-footed trapdoor spider)   \n",
       "\n",
       "                                          Protein names  \\\n",
       "3     Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....   \n",
       "4     Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...   \n",
       "7     Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...   \n",
       "15    Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...   \n",
       "16    Disintegrin-like/cysteine-rich protein MPIII-3...   \n",
       "...                                                 ...   \n",
       "5024                          Conotoxin Vc7.1 (H_Vc7.1)   \n",
       "5026                                        Toxin ICK-9   \n",
       "5028                              Conotoxin Vc1 (H_Vc1)   \n",
       "5033                                       Toxin ICK-18   \n",
       "5036   U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)   \n",
       "\n",
       "                           Protein families  \\\n",
       "3                      Phospholipase family   \n",
       "4                        Scoloptoxin family   \n",
       "7                      Phospholipase family   \n",
       "15                     Phospholipase family   \n",
       "16    Venom metalloproteinase (M12B) family   \n",
       "...                                     ...   \n",
       "5024                       Conotoxin family   \n",
       "5026                      Neurotoxin family   \n",
       "5028                       Conotoxin family   \n",
       "5033                      Neurotoxin family   \n",
       "5036                      Neurotoxin family   \n",
       "\n",
       "                                               Sequence Fragment  Cluster  \\\n",
       "3     GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...      NaN        8   \n",
       "4     MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...      NaN       12   \n",
       "7     MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...      NaN        8   \n",
       "15    MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...      NaN       35   \n",
       "16    MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...      NaN       36   \n",
       "...                                                 ...      ...      ...   \n",
       "5024  MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...      NaN      744   \n",
       "5026  MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...      NaN       19   \n",
       "5028  MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...      NaN      746   \n",
       "5033  MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...      NaN      748   \n",
       "5036  MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...      NaN      581   \n",
       "\n",
       "      Cluster_Rep  Train_Cluster_Rep  Val_Cluster_Rep  Test_Cluster_Rep  \n",
       "3            True               True            False             False  \n",
       "4            True               True            False             False  \n",
       "7           False               True            False             False  \n",
       "15          False               True            False             False  \n",
       "16           True              False            False              True  \n",
       "...           ...                ...              ...               ...  \n",
       "5024        False               True            False             False  \n",
       "5026        False               True            False             False  \n",
       "5028         True               True            False             False  \n",
       "5033         True               True            False             False  \n",
       "5036        False               True            False             False  \n",
       "\n",
       "[1045 rows x 11 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Protein families</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Fragment</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Cluster_Rep</th>\n",
       "      <th>Train_Cluster_Rep</th>\n",
       "      <th>Val_Cluster_Rep</th>\n",
       "      <th>Test_Cluster_Rep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D4WV12</td>\n",
       "      <td>Sicarius terrosus (Cave spider)</td>\n",
       "      <td>Dermonecrotic toxin StSicTox-betaIB1i (EC 4.6....</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>GDSRRPIWNIAHMVNDLDLVDEYLDDGANSLELDVEFSKSGTALRT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0N7CSQ4</td>\n",
       "      <td>Scolopendra mutilans (Chinese red-headed centi...</td>\n",
       "      <td>Tau-scoloptoxin(04)-Ssm1b (Tau-SLPTX(04)-Ssm1b...</td>\n",
       "      <td>Scoloptoxin family</td>\n",
       "      <td>MLKSFCILSVFMVLFLAKFPDLCSGEEISPLKIVVRNSEYLNNPCN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>A0A1L4BJ98</td>\n",
       "      <td>Hemiscorpius lepturus (Scorpion)</td>\n",
       "      <td>Dermonecrotic toxin Hl-PLD1 (EC 4.6.1.-) (Phos...</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>MAHCYYNSKRGCNRVMKTVALVVLISTVMVEESRGDSQEDKKRPIW...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>A0A5C2A2T2</td>\n",
       "      <td>Conus purpurascens (Purple cone)</td>\n",
       "      <td>Conodipine-P1 (Cdpi-P1) (EC 3.1.1.4) (Phosphat...</td>\n",
       "      <td>Phospholipase family</td>\n",
       "      <td>MKLLAPVLWAMAALGVTWLVAVDSKESCTKHSNGCSTPLRLPCQEY...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>A0A6B7FMR5</td>\n",
       "      <td>Vipera ammodytes ammodytes (Western sand viper)</td>\n",
       "      <td>Disintegrin-like/cysteine-rich protein MPIII-3...</td>\n",
       "      <td>Venom metalloproteinase (M12B) family</td>\n",
       "      <td>MIQVLLVIICLAVFPYQVSSIILESGNINNYEVVYPQKVTALPKGA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5024</th>\n",
       "      <td>W4VSB6</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc7.1 (H_Vc7.1)</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MNTAGRLLLLCLALGLVFESLGIPVADDVEAVRDTDPDEKDPSVHN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>744</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5026</th>\n",
       "      <td>W4VSB9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-9</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MMKLYSLVIIATLAAAAFAATSEEISAAVSEIISQHQQDLERYAKI...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5028</th>\n",
       "      <td>W4VSG7</td>\n",
       "      <td>Conus victoriae (Queen Victoria cone)</td>\n",
       "      <td>Conotoxin Vc1 (H_Vc1)</td>\n",
       "      <td>Conotoxin family</td>\n",
       "      <td>MRTSGRLLLLCLAVGLLLESQAHPNADAGDATRDVGSDRTSVELSK...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>746</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5033</th>\n",
       "      <td>W4VSI6</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>Toxin ICK-18</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTIFALVFCCAIAVVVLGFGENEGSTIDHDQNNCKGPGSRCSNKN...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5036</th>\n",
       "      <td>W4VSI9</td>\n",
       "      <td>Trittame loki (Brush-footed trapdoor spider)</td>\n",
       "      <td>U10-barytoxin-Tl1a (U10-BATX-Tl1a) (Toxin ICK-3)</td>\n",
       "      <td>Neurotoxin family</td>\n",
       "      <td>MKTLVLVAVLGVASLYLLSSASEVQQLSPAEEEFRAFVSTFGGLFE...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>581</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1045 rows × 11 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 223,
   "source": [
    "train_val_test_data = tox[tox[[\"Train_Cluster_Rep\", \"Val_Cluster_Rep\", \"Test_Cluster_Rep\"]].any(axis=1)]\n",
    "train_val_test_data"
   ],
   "id": "f8ae1c451c360c01"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.258634Z",
     "start_time": "2025-03-26T20:00:06.254668Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All representative sequences after 50% redundancy reduction: 1045\n",
      "Train entries: 757 (72.44%)\n",
      "Validation entries: 164 (15.69%)\n",
      "Test entries: 124 (11.87%)\n"
     ]
    }
   ],
   "execution_count": 224,
   "source": [
    "print(\"All representative sequences after 50% redundancy reduction:\", len(tox))\n",
    "print(f\"Train entries: {tox['Train_Cluster_Rep'].sum()} ({tox['Train_Cluster_Rep'].sum()/len(tox):.2%})\")\n",
    "print(f\"Validation entries: {tox['Val_Cluster_Rep'].sum()} ({tox['Val_Cluster_Rep'].sum()/len(tox):.2%})\")\n",
    "print(f\"Test entries: {tox['Test_Cluster_Rep'].sum()} ({tox['Test_Cluster_Rep'].sum()/len(tox):.2%})\")"
   ],
   "id": "e50c2c8ac53761d5"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:06.393505Z",
     "start_time": "2025-03-26T20:00:06.390685Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                                       215\n",
       "Neurotoxin family                                      167\n",
       "Scoloptoxin family                                      83\n",
       "Short scorpion toxin superfamily                        69\n",
       "Natriuretic, Bradykinin potentiating peptide family     52\n",
       "                                                      ... \n",
       "Actinoporin family                                       1\n",
       "ConoGAY family                                           1\n",
       "Opioid neuropeptide precursor family                     1\n",
       "Diuretic hormone class 2 family                          1\n",
       "Cystatin family                                          1\n",
       "Name: count, Length: 93, dtype: int64"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 225,
   "source": "tox[\"Protein families\"].value_counts()",
   "id": "d760849f27e80beb"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:10.957297Z",
     "start_time": "2025-03-26T20:00:10.952689Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "199\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Long chain scorpion toxin family           10\n",
       "Bradykinin-related peptide family          10\n",
       "Teretoxin family                            9\n",
       "Vasopressin/oxytocin family                 8\n",
       "FARP (FMRFamide related peptide) family     8\n",
       "                                           ..\n",
       "Actinoporin family                          1\n",
       "ConoGAY family                              1\n",
       "Opioid neuropeptide precursor family        1\n",
       "Diuretic hormone class 2 family             1\n",
       "Cystatin family                             1\n",
       "Name: count, Length: 77, dtype: int64"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 226,
   "source": [
    "print(tox[\"Protein families\"].value_counts()[tox[\"Protein families\"].value_counts() <= 10].sum())\n",
    "tox[\"Protein families\"].value_counts()[tox[\"Protein families\"].value_counts() <= 10]"
   ],
   "id": "931fb73561a16dc7"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-03-26T20:00:15.657297Z",
     "start_time": "2025-03-26T20:00:15.652045Z"
    }
   },
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "846\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Protein families\n",
       "Conotoxin family                                       215\n",
       "Neurotoxin family                                      167\n",
       "Scoloptoxin family                                      83\n",
       "Short scorpion toxin superfamily                        69\n",
       "Natriuretic, Bradykinin potentiating peptide family     52\n",
       "MCD family                                              41\n",
       "Long (4 C-C) scorpion toxin superfamily                 39\n",
       "Snake three-finger toxin family                         38\n",
       "Cationic peptide family                                 31\n",
       "Non-disulfide-bridged peptide (NDBP) superfamily        19\n",
       "Venom Kunitz-type family                                18\n",
       "Formicidae venom family                                 16\n",
       "Phospholipase family                                    16\n",
       "Long (3 C-C) scorpion toxin superfamily                 15\n",
       "Venom metalloproteinase (M12B) family                   15\n",
       "Limacoditoxin family                                    12\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 227,
   "source": [
    "print(tox[\"Protein families\"].value_counts()[tox[\"Protein families\"].value_counts() > 10].sum())\n",
    "tox[\"Protein families\"].value_counts()[tox[\"Protein families\"].value_counts() > 10]"
   ],
   "id": "9311e22a0d646ef3"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "# Non-Toxins",
   "id": "573633d365ca1d46"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "4b926a752ecbc1e0"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
